diff --git a/yast2-autofs.spec.in b/yast2-autofs.spec.in
index e157043..298370e 100644
--- a/yast2-autofs.spec.in
+++ b/yast2-autofs.spec.in
@@ -9,6 +9,8 @@ BuildRequires:	perl-XML-Writer update-desktop-files yast2 yast2-devtools yast2-t
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Module to Create and Manage autofs Entries in LDAP
 
 %description
@@ -29,8 +31,8 @@ under ou=AUTOFS,$LDAPBASE.
 %defattr(-,root,root)
 %dir @yncludedir@/autofs
 @yncludedir@/autofs/*
-@clientdir@/autofs.ycp
-#@clientdir@/autofs_*.ycp
+@clientdir@/autofs.rb
+#@clientdir@/autofs_*.rb
 @moduledir@/Autofs*
 @desktopdir@/autofs.desktop
 %doc @docdir@
