diff --git a/agents/Makefile.am b/agents/Makefile.am
index c09e509..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,7 +0,0 @@
-#
-# Makefile.am for backup/agents
-#
-
-agent_SCRIPTS =		ag_file_append
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-backup.spec.in b/yast2-backup.spec.in
index 3f5b8fc..6be0439 100644
--- a/yast2-backup.spec.in
+++ b/yast2-backup.spec.in
@@ -4,6 +4,8 @@
 Group:          System/YaST
 License:        GPL-2.0
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:		YaST2 - System Backup
 BuildArchitectures:	noarch
 
@@ -65,10 +67,9 @@ backs them up.
 @agentdir@/ag_file_append
 @scrconfdir@/cfg_backup.scr
 @yncludedir@/backup/*
-@clientdir@/backup*.ycp
+@clientdir@/backup*.rb
 @ybindir@/backup_*
-@moduledir@/Backup.ycp
-@moduledir@/Backup.ybc
+@moduledir@/Backup.rb
 @desktopdir@/backup.desktop
 %doc @docdir@
 
