diff --git a/agents/Makefile.am b/agents/Makefile.am
index 012844e..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,7 +0,0 @@
-# Makefile.am for boot-server/agents
-
-agent_SCRIPTS =
-
-scrconf_DATA =
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-boot-server.spec.in b/yast2-boot-server.spec.in
index 4c884ee..e04ae98 100644
--- a/yast2-boot-server.spec.in
+++ b/yast2-boot-server.spec.in
@@ -8,6 +8,8 @@ Requires:	yast2
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Network Booting and Wake-On-Lan Configuration
 
 %description
@@ -24,7 +26,7 @@ YaST2 module for network booting and Wake-On-Lan.
 
 %files
 %defattr(-,root,root)
-@clientdir@/wol.ycp
+@clientdir@/wol.rb
 @moduledir@/WOL.*
 %doc @docdir@
 @desktopdir@/wol.desktop
