diff --git a/agents/Makefile.am b/agents/Makefile.am
index f2e1523..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,24 +0,0 @@
-#
-# Makefile.am for bootloader/installation/agents
-#
-
-scrconf_DATA = $(patsubst $(srcdir)/%,%,$(wildcard $(srcdir)/*.scr))
-
-agent_SCRIPTS = ag_bootloader
-
-fillup_DATA = sysconfig.bootloader
-
-SYSCONFIGSOURCES = $(wildcard sysconfig.bootloader.*)
-
-EXTRA_DIST = $(scrconf_DATA) $(SYSCONFIGSOURCES) $(agent_SCRIPTS)
-
-if I386_OR_AMD64
-sysconfig.bootloader : sysconfig.bootloader.i386
-	cp sysconfig.bootloader.i386 sysconfig.bootloader
-else
-sysconfig.bootloader : sysconfig.bootloader.default
-	cp sysconfig.bootloader.default sysconfig.bootloader
-endif
-
-
-
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 1833773..d7b8303 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,5 +1,7 @@
 # Makefile.am for YCP module .../doc/autodocs
 
-AUTODOCS_YCP = $(top_srcdir)/src/*/*.ycp
-AUTODOCS_PM  = $(top_srcdir)/src/*/*.pm
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled
+
+#AUTODOCS_YCP = $(top_srcdir)/src/*/*.ycp
+#AUTODOCS_PM  = $(top_srcdir)/src/*/*.pm
+#include $(top_srcdir)/autodocs-ycp.ami
diff --git a/src/include/bootloader/elilo/options.ycp b/src/include/bootloader/elilo/options.ycp
index 1584c1e..6d977f9 100644
--- a/src/include/bootloader/elilo/options.ycp
+++ b/src/include/bootloader/elilo/options.ycp
@@ -21,7 +21,7 @@ import "Label";
 import "BootCommon";
 
 include "bootloader/elilo/helps.ycp";
-
+include "bootloader/routines/common_options.ycp";
 
 /**
  * Common widgets of global settings for ELILO
diff --git a/src/include/bootloader/grub/misc.ycp b/src/include/bootloader/grub/misc.ycp
index 7a50307..5fd7869 100644
--- a/src/include/bootloader/grub/misc.ycp
+++ b/src/include/bootloader/grub/misc.ycp
@@ -21,7 +21,9 @@
     import "Storage";
     import "StorageDevices";
     import "Mode";
+    import "Stage";
     import "BootCommon";
+    import "BootStorage";
     import "PackageSystem";
     import "Map";
 
diff --git a/src/include/bootloader/grub2/misc.ycp b/src/include/bootloader/grub2/misc.ycp
index 11bff4a..f0ef9ad 100644
--- a/src/include/bootloader/grub2/misc.ycp
+++ b/src/include/bootloader/grub2/misc.ycp
@@ -22,8 +22,10 @@
     import "StorageDevices";
     import "Mode";
     import "BootCommon";
+    import "BootStorage";
     import "PackageSystem";
     import "Map";
+    import "Stage";
 
 
 // --------------------------------------------------------------
diff --git a/src/include/bootloader/routines/common_options.ycp b/src/include/bootloader/routines/common_options.ycp
index f8f9d16..c83c68d 100644
--- a/src/include/bootloader/routines/common_options.ycp
+++ b/src/include/bootloader/routines/common_options.ycp
@@ -19,6 +19,9 @@
 
 textdomain "bootloader";
 
+import "BootCommon";
+import "Label";
+
 include "bootloader/routines/common_helps.ycp";
 
 /**
diff --git a/src/include/bootloader/routines/global_widgets.ycp b/src/include/bootloader/routines/global_widgets.ycp
index 078ae3c..25d5cc0 100644
--- a/src/include/bootloader/routines/global_widgets.ycp
+++ b/src/include/bootloader/routines/global_widgets.ycp
@@ -28,14 +28,17 @@ import "Mode";
 import "Storage";
 import "StorageDevices";
 import "Bootloader";
+import "BootStorage";
 import "Progress";
 import "PackageSystem";
 import "Package";
 import "Message";
-
+import "Popup";
+import "Stage";
 
 
 include "bootloader/routines/helps.ycp";
+include "bootloader/routines/popups.ycp";
 
 /**
  * Init function of widget
diff --git a/src/include/bootloader/routines/helps.ycp b/src/include/bootloader/routines/helps.ycp
index cf856da..e93908c 100644
--- a/src/include/bootloader/routines/helps.ycp
+++ b/src/include/bootloader/routines/helps.ycp
@@ -19,6 +19,7 @@
 {
     textdomain "bootloader";
 
+    import "BootCommon";
     import "StorageDevices";
     import "Arch";
 
diff --git a/src/include/bootloader/routines/lib_iface.ycp b/src/include/bootloader/routines/lib_iface.ycp
index 31b46a4..f54304e 100644
--- a/src/include/bootloader/routines/lib_iface.ycp
+++ b/src/include/bootloader/routines/lib_iface.ycp
@@ -25,9 +25,13 @@
 textdomain "bootloader";
 
 import "System::Bootloader_API";
-import "Storage";
+import "BootArch";
+import "BootStorage";
 import "Mode";
 
+include "bootloader/routines/popups.ycp";
+
+
 /**
  * Loader the library has been initialized to use
  */
diff --git a/src/include/bootloader/routines/popups.ycp b/src/include/bootloader/routines/popups.ycp
index c3e8bd3..ad8627f 100644
--- a/src/include/bootloader/routines/popups.ycp
+++ b/src/include/bootloader/routines/popups.ycp
@@ -28,6 +28,7 @@
     import "Popup";
     import "Report";
     import "String";
+    import "BootCommon";
 
 /**
  * Inform about not available functionality when no loader selected
diff --git a/src/include/bootloader/routines/section_widgets.ycp b/src/include/bootloader/routines/section_widgets.ycp
index 58c6dfa..77199af 100644
--- a/src/include/bootloader/routines/section_widgets.ycp
+++ b/src/include/bootloader/routines/section_widgets.ycp
@@ -32,10 +32,12 @@ import "Bootloader";
 import "BootStorage";
 import "Popup";
 import "FileUtils";
+import "Report";
 
 
 include "bootloader/routines/helps.ycp";
 include "bootloader/routines/section_helps.ycp";
+include "bootloader/routines/popups.ycp";
 
 boolean validate_image(string widget, map event){
   string value = (string) UI::QueryWidget(`id(widget),`Value);
diff --git a/src/modules/BootCommon.ycp b/src/modules/BootCommon.ycp
index f756b6c..5659a4f 100644
--- a/src/modules/BootCommon.ycp
+++ b/src/modules/BootCommon.ycp
@@ -36,6 +36,7 @@ import "Popup";
 import "Package";
 import "PackagesProposal";
 import "BootStorage";
+import "BootArch";
 
 import "Linuxrc";
 
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-bootloader.spec.in b/yast2-bootloader.spec.in
index abbe7fa..fb8280d 100644
--- a/yast2-bootloader.spec.in
+++ b/yast2-bootloader.spec.in
@@ -36,6 +36,8 @@ Obsoletes:	y2c_boot y2t_boot
 Provides:	y2t_inst-bootloader yast2-agent-liloconf-devel yast2-agent-prom-devel
 Obsoletes:	y2t_inst-bootloader yast2-agent-liloconf-devel yast2-agent-prom-devel
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Bootloader Configuration
 
 %description
@@ -66,9 +68,9 @@ This package contains the YaST2 component for bootloader configuration.
 %dir @yncludedir@/bootloader
 @yncludedir@/bootloader/*
 @moduledir@/*
-@clientdir@/bootloader*.ycp
-@clientdir@/print-product.ycp
-@clientdir@/inst_*.ycp
+@clientdir@/bootloader*.rb
+@clientdir@/print-product.rb
+@clientdir@/inst_*.rb
 @ybindir@/*
 @agentdir@/ag_*
 @scrconfdir@/*.scr
