diff --git a/agents/Makefile.am b/agents/Makefile.am
index 012844e..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,7 +0,0 @@
-# Makefile.am for boot-server/agents
-
-agent_SCRIPTS =
-
-scrconf_DATA =
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index ed7a5a6..3be59d2 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,10 +1,14 @@
 # Makefile.am for boot-server/doc/autodocs
 
-htmldir = $(docdir)/autodocs
 
-index.html: $(srcdir)/../../src/*.ycp
-	$(ybindir)/ycpdoc -d. $(srcdir)/../../src/*.ycp
+# TODO FIXME: temporarily disabled
 
-html_DATA = index.html $(wildcard *.html)
 
-CLEANFILES = $(html_DATA)
+#htmldir = $(docdir)/autodocs
+
+#index.html: $(srcdir)/../../src/*.ycp
+#	$(ybindir)/ycpdoc -d. $(srcdir)/../../src/*.ycp
+
+#html_DATA = index.html $(wildcard *.html)
+
+#CLEANFILES = $(html_DATA)
diff --git a/yast2-boot-server.spec.in b/yast2-boot-server.spec.in
index 4c884ee..95ad3d0 100644
--- a/yast2-boot-server.spec.in
+++ b/yast2-boot-server.spec.in
@@ -24,7 +24,7 @@ YaST2 module for network booting and Wake-On-Lan.
 
 %files
 %defattr(-,root,root)
-@clientdir@/wol.ycp
+@clientdir@/wol.rb
 @moduledir@/WOL.*
 %doc @docdir@
 @desktopdir@/wol.desktop
