diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index dca717a..0f67d40 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,4 +1,4 @@
 # Makefile.am for YCP module .../doc/autodocs
 
-AUTODOCS_PM = $(srcdir)/../../src/YaPI/*.pm
+AUTODOCS_PM = $(srcdir)/../../src/modules/YaPI/*.pm
 include $(top_srcdir)/autodocs-ycp.ami
diff --git a/yast2-ca-management.spec.in b/yast2-ca-management.spec.in
index 05afd3f..dce42da 100644
--- a/yast2-ca-management.spec.in
+++ b/yast2-ca-management.spec.in
@@ -27,6 +27,8 @@ Requires:       perl-gettext
 Requires:       yast2 >= 2.21.22
 Requires:       yast2-perl-bindings
 BuildArch:      noarch
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:        YaST2 - CAs, Certificates and Requests Management
 
 %description
@@ -45,14 +47,14 @@ Managing CAs, Certificates and Requests in an understanding way.
 %dir @moduledir@/YaPI
 %dir @moduledir@/YaST
 @yncludedir@/ca-management/*
-@clientdir@/ca-mgm.ycp
-@clientdir@/ca_mgm.ycp
-@clientdir@/ca_mgm_proposal.ycp
-@clientdir@/ca_select_proposal.ycp
-@clientdir@/common-cert.ycp
-@clientdir@/common_cert.ycp
-@clientdir@/ca_mgm_auto.ycp
-@moduledir@/CaMgm.y*
+@clientdir@/ca-mgm.rb
+@clientdir@/ca_mgm.rb
+@clientdir@/ca_mgm_proposal.rb
+@clientdir@/ca_select_proposal.rb
+@clientdir@/common-cert.rb
+@clientdir@/common_cert.rb
+@clientdir@/ca_mgm_auto.rb
+@moduledir@/CaMgm.rb
 @moduledir@/YaPI/CaManagement.pm
 @moduledir@/YaST/caUtils.pm
 @desktopdir@/ca_mgm.desktop
