diff --git a/src/include/certify/ui_tests.ycp b/src/include/certify/ui_tests.ycp
index 71b4c62..5c174aa 100644
--- a/src/include/certify/ui_tests.ycp
+++ b/src/include/certify/ui_tests.ycp
@@ -26,7 +26,7 @@
      * @return term
      * Create a label, a button and a outputfield
      */
-    global define term BuildTestContents (string caption, string helptext, string frameheader, term customwidget, string status) ``{
+    global define void BuildTestContents (string caption, string helptext, string frameheader, term customwidget, string status) ``{
 
 	string statusmsg = _("Current Status: %1");
 
