diff --git a/agents/Makefile.am b/agents/Makefile.am
index e8c13b5..df8bfcb 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,10 +1 @@
-# Makefile.am for cluster/agents
-
-agent_SCRIPTS = ag_openais
-
-scrconf_DATA = openais.scr \
-	csync2_ha.scr \
-	cfg_pacemaker.scr \
-	cfg_openais.scr
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
+# no longer needed
diff --git a/doc/Makefile.am b/doc/Makefile.am
index 627c681..39cfe13 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -1,6 +1,7 @@
 # Makefile.am for cluster/doc
 
-SUBDIRS = autodocs
+# TODO autodocs work now only for ycp and perl
+#SUBDIRS = autodocs
 
 htmldir = $(docdir)
 
diff --git a/yast2-cluster.spec.in b/yast2-cluster.spec.in
index 6063b4e..8c80699 100644
--- a/yast2-cluster.spec.in
+++ b/yast2-cluster.spec.in
@@ -6,6 +6,8 @@ BuildRequires:	perl-XML-Writer update-desktop-files yast2 yast2-devtools yast2-t
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	Configuration of cluster
 
 %description
@@ -23,12 +25,10 @@ Summary:	Configuration of cluster
 %defattr(-,root,root)
 %dir @yncludedir@/cluster
 @yncludedir@/cluster/*
-@clientdir@/cluster.ycp
-@clientdir@/cluster_*.ycp
+@clientdir@/cluster.rb
+@clientdir@/cluster_*.rb
 @moduledir@/Cluster.*
-@moduledir@/Cluster2.*
-@moduledir@/Cluster3.py
 @desktopdir@/cluster.desktop
-@scrconfdir@/openais.scr
+@scrconfdir@/*.scr
 @agentdir@/ag_openais
 %doc @docdir@
