diff --git a/console/Makefile.am b/console/Makefile.am
index f494239..ac2f5d6 100644
--- a/console/Makefile.am
+++ b/console/Makefile.am
@@ -1,3 +1,3 @@
 #
 # Makefile.am for country/console
-SUBDIRS = agents data src
+SUBDIRS = src
diff --git a/keyboard/Makefile.am b/keyboard/Makefile.am
index 4e2769f..f0f8114 100644
--- a/keyboard/Makefile.am
+++ b/keyboard/Makefile.am
@@ -1,4 +1,4 @@
 #
 # Makefile.am for country/keyboard
-SUBDIRS = agents data doc src testsuite
+SUBDIRS = doc src testsuite
 
diff --git a/keyboard/doc/autodocs/Makefile.am b/keyboard/doc/autodocs/Makefile.am
index e7ddf79..2a6f678 100644
--- a/keyboard/doc/autodocs/Makefile.am
+++ b/keyboard/doc/autodocs/Makefile.am
@@ -1,4 +1,3 @@
 # Makefile.am for YCP module .../doc/autodocs
 
-AUTODOCS_YCP=$(srcdir)/../../src/*.ycp
 include $(top_srcdir)/autodocs-ycp.ami
diff --git a/keyboard/testsuite/Makefile.am b/keyboard/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/keyboard/testsuite/Makefile.am
+++ b/keyboard/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/language/Makefile.am b/language/Makefile.am
index d90959c..9903361 100644
--- a/language/Makefile.am
+++ b/language/Makefile.am
@@ -1,4 +1,4 @@
 #
 # Makefile.am for country/language
-SUBDIRS = agents data doc src testsuite
+SUBDIRS = doc src testsuite
 
diff --git a/language/doc/autodocs/Makefile.am b/language/doc/autodocs/Makefile.am
index e7ddf79..2a6f678 100644
--- a/language/doc/autodocs/Makefile.am
+++ b/language/doc/autodocs/Makefile.am
@@ -1,4 +1,3 @@
 # Makefile.am for YCP module .../doc/autodocs
 
-AUTODOCS_YCP=$(srcdir)/../../src/*.ycp
 include $(top_srcdir)/autodocs-ycp.ami
diff --git a/language/testsuite/Makefile.am b/language/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/language/testsuite/Makefile.am
+++ b/language/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/timezone/Makefile.am b/timezone/Makefile.am
index 53f3572..ce84dc6 100644
--- a/timezone/Makefile.am
+++ b/timezone/Makefile.am
@@ -1,4 +1,4 @@
 #
 # Makefile.am for country/timezone
-SUBDIRS = agents data doc src testsuite
+SUBDIRS = doc src testsuite
 
diff --git a/timezone/doc/autodocs/Makefile.am b/timezone/doc/autodocs/Makefile.am
index e7ddf79..2a6f678 100644
--- a/timezone/doc/autodocs/Makefile.am
+++ b/timezone/doc/autodocs/Makefile.am
@@ -1,4 +1,3 @@
 # Makefile.am for YCP module .../doc/autodocs
 
-AUTODOCS_YCP=$(srcdir)/../../src/*.ycp
 include $(top_srcdir)/autodocs-ycp.ami
diff --git a/timezone/testsuite/Makefile.am b/timezone/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/timezone/testsuite/Makefile.am
+++ b/timezone/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-country.spec.in b/yast2-country.spec.in
index 2361cda..669eb09 100644
--- a/yast2-country.spec.in
+++ b/yast2-country.spec.in
@@ -50,6 +50,8 @@ Obsoletes:	y2t_inst-language
 Provides:	y2t_inst-environment
 Obsoletes:	y2t_inst-environment
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Country Settings (Language, Keyboard, and Timezone)
 
 %description
@@ -74,13 +76,13 @@ install -m 0644 %SOURCE2 $RPM_BUILD_ROOT/usr/share/polkit-1/actions/
 %defattr(-,root,root)
 %doc @docdir@
 %doc COPYING
-@moduledir@/Console.y*
-@moduledir@/Keyboard.y*
-@moduledir@/Timezone.y*
+@moduledir@/Console.rb
+@moduledir@/Keyboard.rb
+@moduledir@/Timezone.rb
 %dir @moduledir@/YaPI
 @moduledir@/YaPI/TIME.pm
 @moduledir@/YaPI/LANGUAGE.pm
-@clientdir@/*.ycp
+@clientdir@/*.rb
 @ydatadir@/*.ycp
 @yncludedir@/keyboard/
 @yncludedir@/timezone/
@@ -94,6 +96,8 @@ install -m 0644 %SOURCE2 $RPM_BUILD_ROOT/usr/share/polkit-1/actions/
 %attr(644,root,root) %config /usr/share/polkit-1/actions/org.opensuse.yast.modules.yapi.*.policy
 
 %package data
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Data files for Country settings
 Group:          System/YaST
 
@@ -105,4 +109,4 @@ functions (Language module)
 %defattr(-,root,root)
 %dir @ydatadir@/languages
 @ydatadir@/languages/*.ycp
-@moduledir@/Language.y*
+@moduledir@/Language.rb
