diff --git a/src/services/namespace_service/testsuite/modules/Makefile.am b/src/services/namespace_service/testsuite/modules/Makefile.am
index 3f35fba..a38d034 100644
--- a/src/services/namespace_service/testsuite/modules/Makefile.am
+++ b/src/services/namespace_service/testsuite/modules/Makefile.am
@@ -1,13 +1,4 @@
 
-yast_modules = TEST.ycp
+yast_modules = TEST.rb
 
 EXTRA_DIST = $(yast_modules)
-
-yast_modules_ybc = $(yast_modules:.ycp=.ybc)
-
-$(yast_modules_ybc): $(yast_modules)
-	ycpc -c $<
-
-all: $(yast_modules_ybc)
-
-CLEANFILES = $(wildcard *.ybc)
diff --git a/src/services/namespace_service/testsuite/org.freedesktop.DBus.Introspectable.Introspect.test b/src/services/namespace_service/testsuite/org.freedesktop.DBus.Introspectable.Introspect.test
index 471aa70..1469899 100644
--- a/src/services/namespace_service/testsuite/org.freedesktop.DBus.Introspectable.Introspect.test
+++ b/src/services/namespace_service/testsuite/org.freedesktop.DBus.Introspectable.Introspect.test
@@ -42,6 +42,7 @@
   <method name='String'><arg name='ret' type='s' direction='out'/></method>
   <method name='Symbol'><arg name='ret' type='s' direction='out'/></method>
   <method name='Void'></method>
+  <method name='last_exception'><arg name='ret' type='s' direction='out'/></method>
  </interface>
  <interface name='org.opensuse.YaST.YCPValues'>
   <method name='Any'><arg name='ret' type='(bsv)' direction='out'/></method>
@@ -80,6 +81,7 @@
   <method name='String'><arg name='ret' type='(bsv)' direction='out'/></method>
   <method name='Symbol'><arg name='ret' type='(bsv)' direction='out'/></method>
   <method name='Void'></method>
+  <method name='last_exception'><arg name='ret' type='(bsv)' direction='out'/></method>
  </interface>
 </node>
 "
diff --git a/yast2-dbus-server.spec.in b/yast2-dbus-server.spec.in
index 6b952e8..b76b753 100644
--- a/yast2-dbus-server.spec.in
+++ b/yast2-dbus-server.spec.in
@@ -25,6 +25,9 @@ BuildRequires:  dbus-1-python python-devel
 # its tests
 BuildRequires:  dbus-1-python python-devel
 
+# to run the tests
+BuildRequires:  yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - DBus Server
 
 %description
