diff --git a/agents/Makefile.am b/agents/Makefile.am
index c22e4db..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,10 +0,0 @@
-#
-# Makefile.am for devtools/devtools/skeletons/config/agents
-#
-
-scrconf_DATA =		cfg_dhcpd.scr \
-			etc_dhcpd_conf.scr
-
-agent_SCRIPTS =		ag_dhcpd_conf
-
-EXTRA_DIST = $(scrconf_DATA) $(agent_SCRIPTS)
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 2a6f678..6fade89 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,3 @@
 # Makefile.am for YCP module .../doc/autodocs
-
-include $(top_srcdir)/autodocs-ycp.ami
+# FIXME autodocs temporarily disabled
+# include $(top_srcdir)/autodocs-ycp.ami
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-dhcp-server.spec.in b/yast2-dhcp-server.spec.in
index 90af475..2aedfae 100644
--- a/yast2-dhcp-server.spec.in
+++ b/yast2-dhcp-server.spec.in
@@ -15,6 +15,8 @@ Requires: yast2-dns-server >= 2.13.16
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - DHCP Server Configuration
 
 %description
@@ -33,8 +35,8 @@ configuration.
 %defattr(-,root,root)
 %dir @yncludedir@/dhcp-server
 @yncludedir@/dhcp-server/*
-@clientdir@/dhcp-server.ycp
-@clientdir@/dhcp-server_*.ycp
+@clientdir@/dhcp-server.rb
+@clientdir@/dhcp-server_*.rb
 @moduledir@/*
 @desktopdir@/dhcp-server.desktop
 @scrconfdir@/cfg_dhcpd.scr
