diff --git a/control/Makefile.am b/control/Makefile.am
index 909653c..16fcc05 100644
--- a/control/Makefile.am
+++ b/control/Makefile.am
@@ -5,8 +5,7 @@ controldir = $(yast2dir)/control
 control_DATA = \
 	dirinstall.xml
 
-desktop_DATA = $(wildcard *.desktop)
-EXTRA_DIST = $(desktop_DATA) $(control_DATA)
+EXTRA_DIST = $(control_DATA)
 
 # validate the control file in 'make check' run
 check-local:
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index f8be896..2c14eba 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,5 +1,5 @@
 # Makefile.am for YCP module .../doc/autodocs
 
-AUTODOCS_YCP = $(wildcard $(srcdir)/../../src/*/*.ycp)
-AUTODOCS_PM  = $(wildcard $(srcdir)/../../src/*/*.pm)
-include $(top_srcdir)/autodocs-ycp.ami
+# AUTODOCS_YCP = $(wildcard $(srcdir)/../../src/*/*.ycp)
+# AUTODOCS_PM  = $(wildcard $(srcdir)/../../src/*/*.pm)
+# include $(top_srcdir)/autodocs-ycp.ami
diff --git a/src/modules/DirInstall.ycp b/src/modules/DirInstall.ycp
index c2bff86..79cc100 100644
--- a/src/modules/DirInstall.ycp
+++ b/src/modules/DirInstall.ycp
@@ -93,9 +93,9 @@ global string Propose()
     tmp = add (tmp, sformat (_("Target Directory: %1"), HTML::Colorize(Installation::dirinstall_target, color)));
 
     // Proposal for backup during update
-    tmp = add (tmp, sformat (_("Run YaST and SuSEconfig at First Boot: %1"), HTML::Colorize(DirInstall::runme_at_boot
+    tmp = add (tmp, sformat (_("Run YaST and SuSEconfig at First Boot: %1"), HTML::Colorize(DirInstall::runme_at_boot ?
 	// part of label in proposal
-	? _("Yes"):
+	_("Yes"):
 	// part of label in proposal
 	_("No"), color) ));
     // Proposal for dirinstall installation
diff --git a/yast2-dirinstall.spec.in b/yast2-dirinstall.spec.in
index 23e001e..c6e1a4b 100644
--- a/yast2-dirinstall.spec.in
+++ b/yast2-dirinstall.spec.in
@@ -23,6 +23,8 @@ Provides:	/usr/share/YaST2/clients/dirinstall.ycp
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Installation into Directory
 
 %description
@@ -41,7 +43,7 @@ directory.
 %defattr(-,root,root)
 %dir @yncludedir@/dirinstall
 @yncludedir@/dirinstall/*
-@clientdir@/*.ycp
+@clientdir@/*.rb
 @moduledir@/*
 @desktopdir@/*.desktop
 %dir /usr/share/YaST2/control
