diff --git a/agents/Makefile.am b/agents/Makefile.am
index dc803c8..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,15 +0,0 @@
-#
-# Makefile.am for devtools/devtools/skeletons/config/agents
-#
-
-agent_SCRIPTS =		ag_dns_zone	\
-			ag_named_forwarders	\
-			ag_convert_named_conf
-
-scrconf_DATA =		dns_zone.scr	\
-			dns_named.scr	\
-			cfg_named.scr	\
-			named_forwarders.scr	\
-			convert_named_conf.scr
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/src/include/dns-server/dialog-main.ycp b/src/include/dns-server/dialog-main.ycp
index 02ebf85..2589485 100644
--- a/src/include/dns-server/dialog-main.ycp
+++ b/src/include/dns-server/dialog-main.ycp
@@ -1593,7 +1598,9 @@ void RedrawZonesListWidget () {
 		// TRANSLATORS: Table item - Server type
 		type_trans = _("Forward");
 		break;
-	    y2warning("Unknown zone type %1", type_trans);
+            default:
+                y2warning("Unknown zone type %1", type_trans);
+                break;
 	}
 
 	return `item (`id (index), enc_to_dec[zone_name]:zone_name, type_trans);
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-dns-server.spec.in b/yast2-dns-server.spec.in
index cc654b2..baec9e1 100644
--- a/yast2-dns-server.spec.in
+++ b/yast2-dns-server.spec.in
@@ -33,6 +33,8 @@ Requires:	yast2 >= 2.17.8
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - DNS Server Configuration
 
 %description
@@ -50,8 +52,8 @@ This package contains the YaST2 component for DNS server configuration.
 %defattr(-,root,root)
 %dir @yncludedir@/dns-server
 @yncludedir@/dns-server/*
-@clientdir@/dns-server.ycp
-@clientdir@/dns-server_*.ycp
+@clientdir@/dns-server.rb
+@clientdir@/dns-server_*.rb
 @moduledir@/*
 @desktopdir@/dns-server.desktop
 @scrconfdir@/dns_named.scr
