diff --git a/agents/Makefile.am b/agents/Makefile.am
index a6ddabb..a9d7966 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -3,9 +3,6 @@
 agent_SCRIPTS = \
 	ag_drbd
 
-scrconf_DATA = \
-	drbd.scr
-
 ag_drbd: drbd.rb.yy
 	rm -f drbd.rb
 	racc -e "/usr/bin/ruby" -v -g drbd.rb.yy -o ag_drbd
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 3dca519..c7779b2 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,5 @@
 # Makefile.am for drbd/doc/autodocs
 
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled
+
+#include $(top_srcdir)/autodocs-ycp.ami
diff --git a/src/include/drbd/common.ycp b/src/include/drbd/common.ycp
index f5c46d3..5e7eb23 100644
--- a/src/include/drbd/common.ycp
+++ b/src/include/drbd/common.ycp
@@ -8,6 +8,8 @@
 	import "Popup";
 	import "CWM";
 
+        include "drbd/helps.ycp";
+
 	list<string> DIALOG = [
 		"startup_conf",
 		"global_conf",
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-drbd.spec.in b/yast2-drbd.spec.in
index 70d1f7b..55abaef 100644
--- a/yast2-drbd.spec.in
+++ b/yast2-drbd.spec.in
@@ -6,6 +6,8 @@ Group:          System/YaST
 BuildRequires:  perl-XML-Writer ruby rubygem-racc update-desktop-files yast2 yast2-devtools yast2-testsuite
 Requires:       yast2 
 BuildArch:      noarch
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:        YaST2 - DRBD Configuration
 
 %description
@@ -30,7 +32,7 @@ Authors:
 %files
 %defattr(-,root,root)
 @yncludedir@/drbd/
-@clientdir@/drbd.ycp
+@clientdir@/drbd.rb
 @moduledir@/Drbd.*
 @desktopdir@/drbd.desktop
 @scrconfdir@/*.scr
