diff --git a/agent-fprint/Makefile.am b/agent-fprint/Makefile.am
index 7cc37fd..b6fdb32 100644
--- a/agent-fprint/Makefile.am
+++ b/agent-fprint/Makefile.am
@@ -1 +1 @@
-SUBDIRS = conf doc src testsuite
+SUBDIRS = doc src testsuite
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-fingerprint-reader.spec.in b/yast2-fingerprint-reader.spec.in
index bcceba7..e64b479 100644
--- a/yast2-fingerprint-reader.spec.in
+++ b/yast2-fingerprint-reader.spec.in
@@ -10,6 +10,8 @@ BuildRequires:	doxygen gcc-c++ yast2-core-devel perl-XML-Writer update-desktop-f
 
 Conflicts:	yast2-hardware-detection < 2.15.7
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Fingerprint Reader Configuration
 
 %description
@@ -28,8 +30,8 @@ thinkfinger library.
 %defattr(-,root,root)
 %dir @yncludedir@/fingerprint-reader
 @yncludedir@/fingerprint-reader/*
-@clientdir@/fingerprint-reader.ycp
-@clientdir@/users_plugin_fingerprint_reader.ycp
+@clientdir@/fingerprint-reader.rb
+@clientdir@/users_plugin_fingerprint_reader.rb
 @moduledir@/FingerprintReader.*
 @moduledir@/UsersPluginFingerprintReader.pm
 @desktopdir@/fingerprint-reader.desktop
