diff --git a/agent-fprint/Makefile.am b/agent-fprint/Makefile.am
index 7cc37fd..b6fdb32 100644
--- a/agent-fprint/Makefile.am
+++ b/agent-fprint/Makefile.am
@@ -1 +1 @@
-SUBDIRS = conf doc src testsuite
+SUBDIRS = doc src testsuite
diff --git a/agent-fprint/doc/autodocs/Makefile.am b/agent-fprint/doc/autodocs/Makefile.am
index e4bab2c..bec31c5 100644
--- a/agent-fprint/doc/autodocs/Makefile.am
+++ b/agent-fprint/doc/autodocs/Makefile.am
@@ -2,5 +2,7 @@
 # Makefile.am for .../agent-fprint/doc/autodocs
 #
 
-AUTODOCS_SUBDIR=agent-fprint
-include $(top_srcdir)/autodocs-cc.ami
+# TODO FIXME: temporarily disabled
+
+#AUTODOCS_SUBDIR=agent-fprint
+#include $(top_srcdir)/autodocs-cc.ami
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 6233695..96989d4 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,5 @@
 # Makefile.am for fingerprint-reader/doc/autodocs
 
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled
+
+#include $(top_srcdir)/autodocs-ycp.ami
diff --git a/src/clients/users_plugin_fingerprint_reader.ycp b/src/clients/users_plugin_fingerprint_reader.ycp
index 1763905..dee084e 100644
--- a/src/clients/users_plugin_fingerprint_reader.ycp
+++ b/src/clients/users_plugin_fingerprint_reader.ycp
@@ -43,6 +43,7 @@
     import "Users";
     import "UsersPluginFingerprintReader";
     import "Wizard";
+    import "Users";
 
     any ret			= nil;
     string func			= "";
diff --git a/yast2-fingerprint-reader.spec.in b/yast2-fingerprint-reader.spec.in
index bcceba7..0788492 100644
--- a/yast2-fingerprint-reader.spec.in
+++ b/yast2-fingerprint-reader.spec.in
@@ -28,8 +28,8 @@ thinkfinger library.
 %defattr(-,root,root)
 %dir @yncludedir@/fingerprint-reader
 @yncludedir@/fingerprint-reader/*
-@clientdir@/fingerprint-reader.ycp
-@clientdir@/users_plugin_fingerprint_reader.ycp
+@clientdir@/fingerprint-reader.rb
+@clientdir@/users_plugin_fingerprint_reader.rb
 @moduledir@/FingerprintReader.*
 @moduledir@/UsersPluginFingerprintReader.pm
 @desktopdir@/fingerprint-reader.desktop
