diff --git a/agent-fprint/Makefile.am b/agent-fprint/Makefile.am
index 7cc37fd..b6fdb32 100644
--- a/agent-fprint/Makefile.am
+++ b/agent-fprint/Makefile.am
@@ -1 +1 @@
-SUBDIRS = conf doc src testsuite
+SUBDIRS = doc src testsuite
diff --git a/agent-fprint/doc/autodocs/Makefile.am b/agent-fprint/doc/autodocs/Makefile.am
index e4bab2c..bec31c5 100644
--- a/agent-fprint/doc/autodocs/Makefile.am
+++ b/agent-fprint/doc/autodocs/Makefile.am
@@ -2,5 +2,7 @@
 # Makefile.am for .../agent-fprint/doc/autodocs
 #
 
-AUTODOCS_SUBDIR=agent-fprint
-include $(top_srcdir)/autodocs-cc.ami
+# TODO FIXME: temporarily disabled
+
+#AUTODOCS_SUBDIR=agent-fprint
+#include $(top_srcdir)/autodocs-cc.ami
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 6233695..96989d4 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,5 @@
 # Makefile.am for fingerprint-reader/doc/autodocs
 
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled
+
+#include $(top_srcdir)/autodocs-ycp.ami
diff --git a/src/include/fingerprint-reader/dialogs.ycp b/src/include/fingerprint-reader/dialogs.ycp
index b1775c3..752d197 100644
--- a/src/include/fingerprint-reader/dialogs.ycp
+++ b/src/include/fingerprint-reader/dialogs.ycp
@@ -82,10 +82,10 @@ any FingerprintReaderDialog() {
     string help_text = _("<p>
 <b>Fingerprint Authentication</b><br>
 The fingerprint reader configuration updates your PAM settings to enable authentication with fingerprints.</p>
-")
+") +
 
     // help text, cont.
-    + _("<p>To register fingerprints for a user, start YaST User Management and see the Plug-Ins tab of the selected user. To automatically start YaST User Management, activate <b>Start User Management After Finish</b>.</p>");
+    _("<p>To register fingerprints for a user, start YaST User Management and see the Plug-Ins tab of the selected user. To automatically start YaST User Management, activate <b>Start User Management After Finish</b>.</p>");
 
     term con = `HBox (`HSpacing (3), `VBox (
 	// frame label
diff --git a/yast2-fingerprint-reader.spec.in b/yast2-fingerprint-reader.spec.in
index bcceba7..e64b479 100644
--- a/yast2-fingerprint-reader.spec.in
+++ b/yast2-fingerprint-reader.spec.in
@@ -10,6 +10,8 @@ BuildRequires:	doxygen gcc-c++ yast2-core-devel perl-XML-Writer update-desktop-f
 
 Conflicts:	yast2-hardware-detection < 2.15.7
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Fingerprint Reader Configuration
 
 %description
@@ -28,8 +30,8 @@ thinkfinger library.
 %defattr(-,root,root)
 %dir @yncludedir@/fingerprint-reader
 @yncludedir@/fingerprint-reader/*
-@clientdir@/fingerprint-reader.ycp
-@clientdir@/users_plugin_fingerprint_reader.ycp
+@clientdir@/fingerprint-reader.rb
+@clientdir@/users_plugin_fingerprint_reader.rb
 @moduledir@/FingerprintReader.*
 @moduledir@/UsersPluginFingerprintReader.pm
 @desktopdir@/fingerprint-reader.desktop
