diff --git a/config/Makefile.am b/config/Makefile.am
index 77a568c..bbcd6d6 100644
--- a/config/Makefile.am
+++ b/config/Makefile.am
@@ -1,7 +1,3 @@
 # Makefile.am for source/firewall/src/config
 # $Id: $
-
-rncdir =  $(schemadir)/autoyast/rnc
-rnc_DATA = firewall.rnc
-
-EXTRA_DIST = $(rnc_DATA)
+# moved
diff --git a/src/clients/firewall.ycp b/src/clients/firewall.ycp
index b6ec9d2..0ad6b3e 100644
--- a/src/clients/firewall.ycp
+++ b/src/clients/firewall.ycp
@@ -35,6 +35,9 @@ you may find current contact information at www.novell.com
     y2milestone("----------------------------------------");
     y2milestone("Firewall module started");
 
+    // moved to toplevel because it is no harm to load it and y2r doesn't handle well include inside deep code
+    include "firewall/wizards.ycp";
+
     textdomain "firewall";
 
     import "SuSEFirewall";
@@ -53,7 +56,6 @@ you may find current contact information at www.novell.com
 	SuSEFirewallCMDLine::Run();
     // GUI or TextUI
     } else {
-	include "firewall/wizards.ycp";
 	// installation has other sequence
 	if (Mode::installation()) {
 	    ret = FirewallInstallationSequence();
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-firewall.spec.in b/yast2-firewall.spec.in
index d1c1669..672f7e4 100644
--- a/yast2-firewall.spec.in
+++ b/yast2-firewall.spec.in
@@ -20,6 +20,8 @@ Obsoletes:	yast2-trans-firewall
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Firewall Configuration
 
 %description
