diff --git a/config/Makefile.am b/config/Makefile.am
index 77a568c..bbcd6d6 100644
--- a/config/Makefile.am
+++ b/config/Makefile.am
@@ -1,7 +1,3 @@
 # Makefile.am for source/firewall/src/config
 # $Id: $
-
-rncdir =  $(schemadir)/autoyast/rnc
-rnc_DATA = firewall.rnc
-
-EXTRA_DIST = $(rnc_DATA)
+# moved
diff --git a/doc/Makefile.am b/doc/Makefile.am
index 406af07..4df4944 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -4,7 +4,8 @@
 # $Id$
 #
 
-SUBDIRS = autodocs
+# Caanot generate autodocs for ruby now
+#SUBDIRS = autodocs
 
 htmldir = @docdir@
 html_DATA =
diff --git a/src/clients/firewall.ycp b/src/clients/firewall.ycp
index b6ec9d2..0ad6b3e 100644
--- a/src/clients/firewall.ycp
+++ b/src/clients/firewall.ycp
@@ -35,6 +35,9 @@ you may find current contact information at www.novell.com
     y2milestone("----------------------------------------");
     y2milestone("Firewall module started");
 
+    // moved to toplevel because it is no harm to load it and y2r doesn't handle well include inside deep code
+    include "firewall/wizards.ycp";
+
     textdomain "firewall";
 
     import "SuSEFirewall";
@@ -53,7 +56,6 @@ you may find current contact information at www.novell.com
 	SuSEFirewallCMDLine::Run();
     // GUI or TextUI
     } else {
-	include "firewall/wizards.ycp";
 	// installation has other sequence
 	if (Mode::installation()) {
 	    ret = FirewallInstallationSequence();
diff --git a/src/include/firewall/complex.ycp b/src/include/firewall/complex.ycp
index ccbb1a4..039e8a9 100644
--- a/src/include/firewall/complex.ycp
+++ b/src/include/firewall/complex.ycp
@@ -38,6 +38,7 @@ you may find current contact information at www.novell.com
     import "Confirm";
     import "Report";
     import "Message";
+    import "Label";
 
     include "firewall/helps.ycp";
 
diff --git a/yast2-firewall.spec.in b/yast2-firewall.spec.in
index d1c1669..c9097f0 100644
--- a/yast2-firewall.spec.in
+++ b/yast2-firewall.spec.in
@@ -20,6 +20,8 @@ Obsoletes:	yast2-trans-firewall
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Firewall Configuration
 
 %description
diff --git a/src/include/firewall/summary.ycp b/src/include/firewall/summary.ycp
index 92f05ae..bd651b8 100644
--- a/src/include/firewall/summary.ycp
+++ b/src/include/firewall/summary.ycp
@@ -95,9 +95,9 @@ you may find current contact information at www.novell.com
 		    _("Any unassigned interface will be assigned to this zone.");
 		//);
 	    } else {
-		ret_val = " '" + spec_interface + "' "
+		ret_val = " '" + spec_interface + "' " +
 		    // TRANSLATORS: informative text
-		    + _("Currently supported only in external zone.");
+		    _("Currently supported only in external zone.");
 	    }
 	} else {
 	    // TRANSLATORS: informative text
