diff --git a/agents/Makefile.am b/agents/Makefile.am
index 112eb8f..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,7 +0,0 @@
-# Makefile.am for firstboot/agents
-
-agent_SCRIPTS =
-
-scrconf_DATA = cfg_firstboot.scr
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/desktop/Makefile.am b/desktop/Makefile.am
index 1f8f3a5..832dcf6 100644
--- a/desktop/Makefile.am
+++ b/desktop/Makefile.am
@@ -1,15 +1,9 @@
 # Makefile.am for firstboot/src/data
 
 
-fillupdir = /var/adm/fillup-templates/
-fillup_DATA = sysconfig.firstboot
-
 modconfigdir = $(prefix)/share/autoinstall/modules
 modconfig_DATA = firstboot.desktop
 
-schemafilesdir =  $(schemadir)/autoyast/rnc
-schemafiles_DATA = $(wildcard *.rnc)
-
-EXTRA_DIST = $(fillup_DATA) $(modconfig_DATA) $(schemafiles_DATA)
+EXTRA_DIST = $(modconfig_DATA)
 
 include $(top_srcdir)/Makefile.am.common
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 2a6f678..9de7be0 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,5 @@
 # Makefile.am for YCP module .../doc/autodocs
 
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled
+
+#include $(top_srcdir)/autodocs-ycp.ami
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-firstboot.spec.in b/yast2-firstboot.spec.in
index b0a1cd5..442cbe8 100644
--- a/yast2-firstboot.spec.in
+++ b/yast2-firstboot.spec.in
@@ -20,6 +20,8 @@ Requires:	yast2-installation >= 2.19.0
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Initial System Configuration
 PreReq:         %fillup_prereq
 
@@ -53,10 +55,10 @@ mkdir -p $RPM_BUILD_ROOT/usr/share/firstboot/scripts
 %dir @ystartupdir@/startup/Firstboot-Stage
 @ystartupdir@/startup/Firstboot-Stage/*
 @ystartupdir@/startup/YaST2.Firstboot
-@clientdir@/firstboot_*.ycp
-@clientdir@/firstboot.ycp
+@clientdir@/firstboot_*.rb
+@clientdir@/firstboot.rb
 %dir @yncludedir@/firstboot
-@yncludedir@/firstboot/*.ycp
+@yncludedir@/firstboot/*.rb
 @moduledir@/Firstboot.*
 @scrconfdir@/*.scr
 /var/adm/fillup-templates/sysconfig.firstboot
