diff --git a/agents/Makefile.am b/agents/Makefile.am
index d75de90..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,9 +0,0 @@
-# Makefile.am for ftpd/agents
-
-agent_SCRIPTS =
-
-scrconf_DATA = 		\
-	pure-ftpd.scr   \
-	vsftpd.scr
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/config/Makefile.am b/config/Makefile.am
index a2b7e6f..e69de29 100644
--- a/config/Makefile.am
+++ b/config/Makefile.am
@@ -1,7 +0,0 @@
-# Makefile.am for source/ftp-server/src/config
-# $Id: $
-
-rncdir =  $(schemadir)/autoyast/rnc
-rnc_DATA = ftp-server.rnc
-
-EXTRA_DIST = $(rnc_DATA)
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-ftp-server.spec.in b/yast2-ftp-server.spec.in
index 896a8c7..68cfa8b 100644
--- a/yast2-ftp-server.spec.in
+++ b/yast2-ftp-server.spec.in
@@ -10,6 +10,8 @@ BuildRequires:	perl-XML-Writer update-desktop-files yast2 yast2-devtools yast2-t
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - FTP configuration
 
 %description
@@ -28,8 +30,8 @@ configure two daemons: pure-ftpd and vsftpd.
 %defattr(-,root,root)
 %dir @yncludedir@/ftp-server
 @yncludedir@/ftp-server/*
-@clientdir@/ftp-server.ycp
-@clientdir@/ftp-server_*.ycp
+@clientdir@/ftp-server.rb
+@clientdir@/ftp-server_*.rb
 @moduledir@/FtpServer.*
 @desktopdir@/ftp-server.desktop
 @schemadir@/autoyast/rnc/ftp-server.rnc
