diff --git a/conf/Makefile.am b/conf/Makefile.am
index 18ebdc2..e69de29 100644
--- a/conf/Makefile.am
+++ b/conf/Makefile.am
@@ -1,7 +0,0 @@
-#
-# Makefile.am for core/agent-probe/conf
-#
-
-scrconf_DATA = probe.scr
-
-EXTRA_DIST = $(scrconf_DATA)
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index a710e62..20ad188 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -17,5 +17,5 @@ runag_hwprobe_SOURCES = runag_hwprobe.cc
 runag_hwprobe_LDADD = ${AGENT_LIBADD}
 runag_hwprobe_LDFLAGS = 			\
 	-Xlinker --whole-archive		\
-	../src/libpy2ag_hwprobe.la		\
+	../agent/libpy2ag_hwprobe.la		\
 	-Xlinker --no-whole-archive
diff --git a/testsuite/runag_hwprobe.cc b/testsuite/runag_hwprobe.cc
index 08f8d0d..59d2218 100644
--- a/testsuite/runag_hwprobe.cc
+++ b/testsuite/runag_hwprobe.cc
@@ -7,7 +7,7 @@
 
 #include <scr/run_agent.h>
 
-#include "../src/HwProbe.h"
+#include "../agent/HwProbe.h"
 
 
 int
diff --git a/yast2-hardware-detection.spec.in b/yast2-hardware-detection.spec.in
index 3f57467..13fb51e 100644
--- a/yast2-hardware-detection.spec.in
+++ b/yast2-hardware-detection.spec.in
@@ -22,6 +22,8 @@ BuildRequires:	openssl-devel
 BuildRequires:	hwinfo-devel
 # we check for hwinfo
 BuildRequires:	pkg-config
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Hardware Detection Interface
 # hwinfo-13.38: hw_fingerprint
 Requires:       hwinfo >= 13.38
