diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 6152b88..425ed35 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,10 +1,11 @@
 # Makefile.am for heartbeat/doc/autodocs
 
-htmldir = $(docdir)/autodocs
+# FIXME temporarily disabled
+# htmldir = $(docdir)/autodocs
 
-index.html: $(srcdir)/../../src/*.ycp
-	$(ybindir)/ycpdoc -d. $(srcdir)/../../src/*.ycp
+# index.html: $(srcdir)/../../src/*.rb
+# 	$(ybindir)/ycpdoc -d. $(srcdir)/../../src/*.rb
 
-html_DATA = index.html $(wildcard *.html)
+# html_DATA = index.html $(wildcard *.html)
 
-CLEANFILES = $(html_DATA)
+# CLEANFILES = $(html_DATA)
diff --git a/src/include/heartbeat/auth_conf.ycp b/src/include/heartbeat/auth_conf.ycp
index 921999a..611a0c6 100644
--- a/src/include/heartbeat/auth_conf.ycp
+++ b/src/include/heartbeat/auth_conf.ycp
@@ -54,6 +54,7 @@ textdomain "heartbeat";
 import "Label";
 import "Wizard";
 import "Heartbeat";
+import "Report";
 
 include "heartbeat/helps.ycp";
 include "heartbeat/common.ycp";
diff --git a/src/include/heartbeat/common.ycp b/src/include/heartbeat/common.ycp
index 11f21c5..54e0c1f 100644
--- a/src/include/heartbeat/common.ycp
+++ b/src/include/heartbeat/common.ycp
@@ -57,7 +57,7 @@ import "Heartbeat";
 import "Popup";
 import "CWM";
 
-//include "heartbeat/helps.ycp";
+include "heartbeat/helps.ycp";
 
 list<string> DIALOG = [
     "node_conf",
diff --git a/src/include/heartbeat/ipfail_conf.ycp b/src/include/heartbeat/ipfail_conf.ycp
index 4d29b98..c049388 100644
--- a/src/include/heartbeat/ipfail_conf.ycp
+++ b/src/include/heartbeat/ipfail_conf.ycp
@@ -55,6 +55,7 @@ import "Wizard";
 import "Heartbeat";
 import "Report";
 import "SCR";
+import "IP";
 
 include "heartbeat/helps.ycp";
 include "heartbeat/common.ycp";
diff --git a/src/include/heartbeat/log_conf.ycp b/src/include/heartbeat/log_conf.ycp
index 1302d3a..e15de3c 100644
--- a/src/include/heartbeat/log_conf.ycp
+++ b/src/include/heartbeat/log_conf.ycp
@@ -93,7 +93,9 @@ term timeouts_conf_getDialog()
 
 any ConfigureTimeoutsDialog () {
 
-    timeouts_conf_Read();
+   // FIXME not defined in this file; 
+   // it's present in the timeouts_conf.ycp file but that is not included
+   //timeouts_conf_Read();
     
     my_SetContents("timeouts_conf", timeouts_conf_getDialog());
     
@@ -119,7 +121,9 @@ any ConfigureTimeoutsDialog () {
 
 	if (ret == `next || ret == `back || contains(DIALOG, (string)ret)) {
 	
-	    if (!timeouts_conf_Write()) continue;
+    // FIXME not defined in this file;
+    // it's present in the timeouts_conf.ycp, however it expects 2 parameters
+	  // if (!timeouts_conf_Write()) continue;
 	    
 	    if (ret != `next && ret != `back)
 		ret = symbolof(toterm(ret));
diff --git a/src/include/heartbeat/startup_conf.ycp b/src/include/heartbeat/startup_conf.ycp
index ae99787..7e9ee32 100644
--- a/src/include/heartbeat/startup_conf.ycp
+++ b/src/include/heartbeat/startup_conf.ycp
@@ -55,6 +55,7 @@ import "Label";
 import "Wizard";
 import "Service";
 import "Heartbeat";
+import "Report";
 
 include "heartbeat/helps.ycp";
 include "heartbeat/common.ycp";
diff --git a/src/include/heartbeat/stonith_conf.ycp b/src/include/heartbeat/stonith_conf.ycp
index ba90cfe..01014ae 100644
--- a/src/include/heartbeat/stonith_conf.ycp
+++ b/src/include/heartbeat/stonith_conf.ycp
@@ -54,6 +54,7 @@ textdomain "heartbeat";
 import "Label";
 import "Wizard";
 import "Heartbeat";
+import "Report";
 
 include "heartbeat/helps.ycp";
 include "heartbeat/common.ycp";
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-heartbeat.spec.in b/yast2-heartbeat.spec.in
index 6193558..e3700c6 100644
--- a/yast2-heartbeat.spec.in
+++ b/yast2-heartbeat.spec.in
@@ -8,6 +8,8 @@ Requires:	yast2 >= 2.21.22
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - High Availability Configuration
 
 %description
@@ -25,8 +27,7 @@ YaST-Module to configure heartbeat
 %defattr(-,root,root)
 %dir @yncludedir@/heartbeat
 @yncludedir@/heartbeat/*
-@clientdir@/heartbeat.ycp
-# @clientdir@/heartbeat_*.ycp
+@clientdir@/heartbeat.rb
 @moduledir@/Heartbeat.*
 @desktopdir@/heartbeat.desktop
 @scrconfdir@/*.scr
