diff --git a/agents/Makefile.am b/agents/Makefile.am
index 3ee8332..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,9 +0,0 @@
-#
-# Makefile.am for devtools/devtools/skeletons/config/agents
-#
-
-agent_SCRIPTS = ag_http_server
-
-scrconf_DATA = cfg_http_server.scr
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-http-server.spec.in b/yast2-http-server.spec.in
index 2821125..c6e77e7 100644
--- a/yast2-http-server.spec.in
+++ b/yast2-http-server.spec.in
@@ -11,6 +11,8 @@ Requires:       yast2 >= 2.21.22
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - HTTP Server Configuration
 
 %description
@@ -30,8 +32,8 @@ configuration.
 %dir @yncludedir@/http-server
 @schemadir@/autoyast/rnc/http-server.rnc
 @yncludedir@/http-server/*
-@clientdir@/http-server.ycp
-@clientdir@/http-server_*.ycp
+@clientdir@/http-server.rb
+@clientdir@/http-server_*.rb
 @moduledir@/*
 @desktopdir@/http-server.desktop
 @scrconfdir@/*
