diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 2a6f678..b9be0f7 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,4 @@
 # Makefile.am for YCP module .../doc/autodocs
-
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO
+# FIXME temporarily disabled
+# include $(top_srcdir)/autodocs-ycp.ami
diff --git a/src/include/inetd/default_conf_xinetd.ycp b/src/include/inetd/default_conf_xinetd.ycp
index 8e1465e..e9aa7f9 100644
--- a/src/include/inetd/default_conf_xinetd.ycp
+++ b/src/include/inetd/default_conf_xinetd.ycp
@@ -38,8 +38,7 @@ string GetBinPath () {
 
 string x_bin_path = GetBinPath();
 
-
-global services_t default_conf =
+global list<map <string, any> > default_conf =
 [
   $[
     "comment" : " admind: admin daemon for thin clients\n uses ident and host name lookups for authentication\n security: weak authentication, just enough for thin clients\n client: adminc.pl\n default config file: /etc/opt/SLES/POS/admind.conf\n",
diff --git a/src/include/inetd/dialogs.ycp b/src/include/inetd/dialogs.ycp
index c232268..2c207f4 100644
--- a/src/include/inetd/dialogs.ycp
+++ b/src/include/inetd/dialogs.ycp
@@ -361,7 +361,8 @@ define symbol InetdDialog () ``{
 	    `HBox(
 		/* Translators: Table Header: The "Ch" label is short of "Changed". Please, make the
 				translation as short as possible. */
-		`Table(`id(`table),// `opt(`notify),
+		`Table(`id(`table),
+    // `opt(`notify),
 		//`opt(`keepSorting),
 		`header(`Center(_("Ch")), `Center(_("Status")),
 		    _("Service"), _("Type "), `Center(_("Protocol")), _("Wait"),
diff --git a/src/modules/Inetd.ycp b/src/modules/Inetd.ycp
index 8d63468..1171923 100644
--- a/src/modules/Inetd.ycp
+++ b/src/modules/Inetd.ycp
@@ -439,8 +439,7 @@ global define services_t MergeAyProfile (services_t target, services_t changes)
 	// if the matching turns out to be too clever:
 	// skip matching
 	string change_iid = change_s["iid"]:"";
-	if (search (change_iid, "new") != 0
-	    /* || find (change_iid, "inst") == 0*/ ) // do nothing if matches is 0 and we add the service
+	if (search (change_iid, "new") != 0 /* || find (change_iid, "inst") == 0*/ ) // do nothing if matches is 0 and we add the service
 	{
 	    // apply the change to the target list:
 	    target = maplist(service_t target_s, target, ``{
diff --git a/yast2-inetd.spec.in b/yast2-inetd.spec.in
index 74e1dd3..9274beb 100644
--- a/yast2-inetd.spec.in
+++ b/yast2-inetd.spec.in
@@ -18,6 +18,8 @@ Obsoletes:	yast2-trans-inet y2t_inet
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Network Services Configuration
 
 %description
@@ -35,10 +35,9 @@ The YaST2 component for configuring the inetd and xinetd daemons.
 %defattr(-,root,root)
 %dir @yncludedir@/inetd
 @yncludedir@/inetd/*
-@clientdir@/inetd*.ycp
-@clientdir@/xinetd*.ycp
-@moduledir@/Inetd.ycp
-@moduledir@/Inetd.ybc
+@clientdir@/inetd*.rb
+@clientdir@/xinetd*.rb
+@moduledir@/Inetd.rb
 @desktopdir@/inetd.desktop
 @schemadir@/autoyast/rnc/inetd.rnc
 %doc %dir @docdir@
