diff --git a/src/clients/inst_automatic_configuration.ycp b/src/clients/inst_automatic_configuration.ycp
index 0689035..66b2263 100644
--- a/src/clients/inst_automatic_configuration.ycp
+++ b/src/clients/inst_automatic_configuration.ycp
@@ -369,7 +369,9 @@
 	    case "proposals":
 		CallProposals (one_autoconf_call["items"]:[]);
 		break;
-	    y2error ("Unknown script type '%1'", type);
+            default:
+                y2error ("Unknown script type '%1'", type);
+                break;
 	}
     });
 
