diff --git a/SUBDIRS b/SUBDIRS
index 12fab8a..40ebadc 100644
--- a/SUBDIRS
+++ b/SUBDIRS
@@ -1 +1 @@
-src agents doc startup control testsuite
+src agents doc startup control testsuite autoyast_desktop
diff --git a/agents/Makefile.am b/agents/Makefile.am
index 939f34b..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,7 +0,0 @@
-#
-# Makefile.am for installation/agents
-#
-scrconf_DATA = $(wildcard *.scr)
-fillup_DATA = sysconfig.security-checksig
-EXTRA_DIST = $(scrconf_DATA) $(fillup_DATA)
-
diff --git a/autoyast_desktop/Makefile.am b/autoyast_desktop/Makefile.am
index 6cdcfce..4dfc77c 100644
--- a/autoyast_desktop/Makefile.am
+++ b/autoyast_desktop/Makefile.am
@@ -1,12 +1,8 @@
 # Makefile.am for installation/src/config
 
-desktop_DATA = release_notes.desktop
 modconfigdir = $(prefix)/share/autoinstall/modules
 modconfig_DATA = deploy_image.desktop
 
-schemafilesdir =  $(schemadir)/autoyast/rnc
-schemafiles_DATA = $(wildcard *.rnc)
-
-EXTRA_DIST = $(desktop_DATA) $(modconfig_DATA) $(schemafiles_DATA)
+EXTRA_DIST = $(modconfig_DATA)
 
 include $(top_srcdir)/Makefile.am.common
diff --git a/control/tests/Makefile.am b/control/tests/Makefile.am
index d7a786a..c928825 100644
--- a/control/tests/Makefile.am
+++ b/control/tests/Makefile.am
@@ -1,7 +1,8 @@
 # Makefile.am for installation/general
 
 client_DATA =					\
-	inst_test_workflow.ycp
+	inst_test_workflow.rb                   \
+        complete_workflow.rb
 
 EXTRA_DIST = $(client_DATA)
 
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index d153b93..0125218 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,15 +1,19 @@
 # Makefile.am for y2m_inst/doc/autodocs
 
-htmldir = $(docdir)/autodocs
+##htmldir = $(docdir)/autodocs
 
-html_DATA = $(wildcard *.html)
-CLEANFILES = $(html_DATA)
+#html_DATA = $(wildcard *.html)
+#CLEANFILES = $(html_DATA)
 
-all-local: index.html
+#all-local: index.html
 
 # Note that we look one _and_ two subdirectories deep.
 # There might be some duplicate filenames
 # but let's solve that later. For now we want to find all Modules::.
-index.html: $(srcdir)/../../src/*/*.ycp
-	${YCPDOC} -d. $^
+
+# TODO FIXME: temporarily disabled, enable back when autodocs
+# support is implemented
+
+#index.html: $(srcdir)/../../src/*/*.ycp
+#	${YCPDOC} -d. $^
 
diff --git a/doc/proposal/Makefile.am b/doc/proposal/Makefile.am
index a8733bf..58fc742 100644
--- a/doc/proposal/Makefile.am
+++ b/doc/proposal/Makefile.am
@@ -15,7 +15,7 @@ html/yast2docs.css: html/index.html
 	cp @STYLESHEET_CSS@ html
 	cp -a `pkg-config --variable=yast2dir yast2-devtools`/data/docbook/images html
 
-EXTRA_DIST =  $(xml_files) dummy_proposal.ycp
+EXTRA_DIST =  $(xml_files) dummy_proposal.rb
 
 CLEANFILES =  $(html_DATA)
 
diff --git a/doc/proposal/proposal-API-section.xml b/doc/proposal/proposal-API-section.xml
index 3ef84e6..80773e2 100644
--- a/doc/proposal/proposal-API-section.xml
+++ b/doc/proposal/proposal-API-section.xml
@@ -155,7 +155,7 @@
     <title>Dummy Proposal</title>
     <programlisting>
 
-      <xi:include href="dummy_proposal.ycp" parse="text"
+      <xi:include href="dummy_proposal.rb" parse="text"
 	xmlns:xi="http://www.w3.org/2001/XInclude"/>
 
     </programlisting>
diff --git a/src/clients/driver_update2_finish.ycp b/src/clients/driver_update2_finish.ycp
index 909e81c..19f4a13 100644
--- a/src/clients/driver_update2_finish.ycp
+++ b/src/clients/driver_update2_finish.ycp
@@ -64,7 +64,10 @@ if (func == "Info")
 else if (func == "Write")
 {
     Vendor::DriverUpdate2 ();
-    if (WFM::Read (.local.size, "/usr/share/YaST2/clients/product_finish.ycp") > 0)
+    // product_finish client is provided by the driver update,
+    // support both YCP and Ruby clients
+    if (WFM::Read (.local.size, "/usr/share/YaST2/clients/product_finish.ycp") > 0
+      || WFM::Read (.local.size, "/usr/share/YaST2/clients/product_finish.rb") > 0)
 	WFM::CallFunction ("product_finish", [Mode::update ()]);
 }
 else
diff --git a/src/clients/inst_automatic_configuration.ycp b/src/clients/inst_automatic_configuration.ycp
index 0689035..66b2263 100644
--- a/src/clients/inst_automatic_configuration.ycp
+++ b/src/clients/inst_automatic_configuration.ycp
@@ -369,7 +369,9 @@
 	    case "proposals":
 		CallProposals (one_autoconf_call["items"]:[]);
 		break;
-	    y2error ("Unknown script type '%1'", type);
+            default:
+                y2error ("Unknown script type '%1'", type);
+                break;
 	}
     });
 
diff --git a/src/clients/inst_congratulate.ycp b/src/clients/inst_congratulate.ycp
index 91cf573..e5cca69 100644
--- a/src/clients/inst_congratulate.ycp
+++ b/src/clients/inst_congratulate.ycp
@@ -280,7 +280,3 @@ selected, a profile of the current system is stored in <tt>/root/autoinst.xml</t
 
     return ret;
 }
-
-// indentation is f*cked up but this comes close at least
-// vim: set noexpandtab:
-// vim: set tabstop=8:
diff --git a/src/clients/inst_finish.ycp b/src/clients/inst_finish.ycp
index 2bfb0fa..5f99928 100644
--- a/src/clients/inst_finish.ycp
+++ b/src/clients/inst_finish.ycp
@@ -434,9 +434,9 @@ if (required_to_open_sl_dialog) {
 if (size (Misc::boot_msg) > 0 && !Mode::autoinst ())
 {
     // bugzilla #245742, #160301
-    if ((Linuxrc::usessh () && !Linuxrc::vnc())
+    if ((Linuxrc::usessh () && !Linuxrc::vnc()) ||
 	// also live installation - bzilla #297691
-	|| Mode::live_installation ())
+	Mode::live_installation ())
     {
 	// Display the message and wait for user to accept it
 	Report::DisplayMessages (true, 0);
diff --git a/src/clients/inst_proposal.ycp b/src/clients/inst_proposal.ycp
index dcba708..da2d99d 100644
--- a/src/clients/inst_proposal.ycp
+++ b/src/clients/inst_proposal.ycp
@@ -1068,9 +1068,9 @@ Put the hardware settings into effect by pressing <b>Next</b>.
 	else if (proposal_mode == "uml")
 	{
 	    // Proposal in uml module
-	    help_text_string = _("<P><B>UML Installation Proposal</B></P>")
+	    help_text_string = _("<P><B>UML Installation Proposal</B></P>") +
 	    // help text
-	    + _("<P>UML (User Mode Linux) installation allows you to start independent
+	    _("<P>UML (User Mode Linux) installation allows you to start independent
 Linux virtual machines in the host system.</P>");
 	}
 	else if (proposal_properties["help"]:"" != "")
diff --git a/src/clients/stroj-casu.ycp b/src/clients/stroj-casu.ycp
index e4904e7..33ea01f 100644
--- a/src/clients/stroj-casu.ycp
+++ b/src/clients/stroj-casu.ycp
@@ -47,10 +47,10 @@
 so much younger than today,<br>...</p>");
     Verzeichnis<string> headings = [
 	Tr("Time Machine"),
-	Tr("Temporal Range"),	// 
-	Tr("Spatial Range"),	// 
-	Tr("Temporal Accuracy"),	// 
-	Tr("Spatial Accuracy"),	// 
+	Tr("Temporal Range"),
+	Tr("Spatial Range"),
+	Tr("Temporal Accuracy"),
+	Tr("Spatial Accuracy"),
 	];
     Verzeichnis<Verzeichnis> buttons = [
 	[ `repair, Tr("Self &Repair") ],
diff --git a/src/include/installation/inst_inc_first.ycp b/src/include/installation/inst_inc_first.ycp
index 143de16..30dbddd 100644
--- a/src/include/installation/inst_inc_first.ycp
+++ b/src/include/installation/inst_inc_first.ycp
@@ -167,7 +167,7 @@
 		"disks_activate", // disks activated before running upgrade
 		"mode", // always doing update, is already preselected
 		"update_partition", // no mounting
-		"prepdisk", // disks already mounted, it is dummy in update anyway
+		"prepdisk" // disks already mounted, it is dummy in update anyway
 	    ], {
 		ProductControl::DisableModule (m);
 	    });
diff --git a/src/include/installation/misc.ycp b/src/include/installation/misc.ycp
index 9dd0c2e..aa9b992 100644
--- a/src/include/installation/misc.ycp
+++ b/src/include/installation/misc.ycp
@@ -191,13 +191,13 @@ previous dialogs.</p>");
 
 	    body =
 	    // Text for confirmation popup before the update really starts 1/3
-		_("<p>Information required to perform an update is now complete.</p>")
+		_("<p>Information required to perform an update is now complete.</p>") +
 	    // Text for confirmation popup before the update really starts 2/3
-		+ _("
+		_("
 <p>If you continue now, data on your hard disk will be overwritten
-according to the settings in the previous dialogs.</p>")
+according to the settings in the previous dialogs.</p>") +
 	    // Text for confirmation popup before the update really starts 3/3
-		+ _("<p>Go back and check the settings if you are unsure.</p>")
+		_("<p>Go back and check the settings if you are unsure.</p>")
 		;
 
 	    // Label for the button that confirms startint the installation
diff --git a/yast2-installation.spec.in b/yast2-installation.spec.in
index 2ff4767..78d35a5 100644
--- a/yast2-installation.spec.in
+++ b/yast2-installation.spec.in
@@ -3,6 +3,8 @@
 @HEADER@
 Group:          System/YaST
 License:        GPL-2.0
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:        YaST2 - Installation Parts
 
 Source1:	YaST2-Second-Stage.service
@@ -167,9 +169,8 @@ install -m 644 %{SOURCE2} %{buildroot}%{_unitdir}
 %{_unitdir}/YaST2-Second-Stage.service
 %{_unitdir}/YaST2-Firstboot.service
 
-@clientdir@/*.ycp
-@moduledir@/*.ycp
-@moduledir@/*.ybc
+@clientdir@/*.rb
+@moduledir@/*.rb
 @desktopdir@/*.desktop
 /usr/share/autoinstall/modules/*.desktop
 /usr/share/YaST2/schema/autoyast/rnc/deploy_image.rnc
