diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 2a6f678..19b5419 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,3 @@
 # Makefile.am for YCP module .../doc/autodocs
 
-include $(top_srcdir)/autodocs-ycp.ami
+# include $(top_srcdir)/autodocs-ycp.ami
diff --git a/httpd/Makefile.am b/httpd/Makefile.am
index 93ce8ad..1d261db 100644
--- a/httpd/Makefile.am
+++ b/httpd/Makefile.am
@@ -1,4 +1,7 @@
 apacheconfdir = /etc/apache2/conf.d
 apacheconf_DATA = inst_server.conf.in
 
+install-data-local:
+	install -d -m 755 $(DESTDIR)/etc/YaST2/instserver
+
 EXTRA_DIST = $(apacheconf_DATA) 
diff --git a/src/include/instserver/dialogs.ycp b/src/include/instserver/dialogs.ycp
index 4471693..fcf1c08 100644
--- a/src/include/instserver/dialogs.ycp
+++ b/src/include/instserver/dialogs.ycp
@@ -99,11 +99,11 @@
 	    basecontent = $[];
 	}
 
-	if ((tolower(content["BASEPRODUCT"]:"") == tolower(basecontent["BASEPRODUCT"]:"")
-	    && content["BASEVERSION"]:"" == basecontent["BASEVERSION"]:"")
+	if ((tolower(content["BASEPRODUCT"]:"") == tolower(basecontent["BASEPRODUCT"]:"") &&
+	    content["BASEVERSION"]:"" == basecontent["BASEVERSION"]:"") ||
 	    // or it's a service pack for maintained product (e.g. NLD)
 	    // compare BASEPRODUCT and PRODUCT in that case
-	    || (!haskey(basecontent, "BASEPRODUCT") && !haskey(basecontent, "BASEVERSION") &&
+	    (!haskey(basecontent, "BASEPRODUCT") && !haskey(basecontent, "BASEVERSION") &&
 		tolower(content["BASEPRODUCT"]:"") == tolower(basecontent["PRODUCT"]:"")
 		&& content["BASEVERSION"]:"" == basecontent["VERSION"]:"" )
 	)
diff --git a/yast2-instserver.spec.in b/yast2-instserver.spec.in
index e452ea7..9e39b33 100644
--- a/yast2-instserver.spec.in
+++ b/yast2-instserver.spec.in
@@ -12,6 +12,8 @@ Requires:	yast2-slp
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Installation Server Configuration and Management
 
 %description
@@ -31,7 +33,7 @@ NFS sources are supported.
 %defattr(-,root,root)
 %dir @yncludedir@/instserver
 @yncludedir@/instserver/*
-@clientdir@/instserver.ycp
+@clientdir@/instserver.rb
 @moduledir@/Instserver.*
 @desktopdir@/instserver.desktop
 %doc @docdir@
