diff --git a/SUBDIRS b/SUBDIRS
index 12fab8a..40ebadc 100644
--- a/SUBDIRS
+++ b/SUBDIRS
@@ -1 +1 @@
-src agents doc startup control testsuite
+src agents doc startup control testsuite autoyast_desktop
diff --git a/agents/Makefile.am b/agents/Makefile.am
index 939f34b..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,7 +0,0 @@
-#
-# Makefile.am for installation/agents
-#
-scrconf_DATA = $(wildcard *.scr)
-fillup_DATA = sysconfig.security-checksig
-EXTRA_DIST = $(scrconf_DATA) $(fillup_DATA)
-
diff --git a/autoyast_desktop/Makefile.am b/autoyast_desktop/Makefile.am
index 6cdcfce..4dfc77c 100644
--- a/autoyast_desktop/Makefile.am
+++ b/autoyast_desktop/Makefile.am
@@ -1,12 +1,8 @@
 # Makefile.am for installation/src/config
 
-desktop_DATA = release_notes.desktop
 modconfigdir = $(prefix)/share/autoinstall/modules
 modconfig_DATA = deploy_image.desktop
 
-schemafilesdir =  $(schemadir)/autoyast/rnc
-schemafiles_DATA = $(wildcard *.rnc)
-
-EXTRA_DIST = $(desktop_DATA) $(modconfig_DATA) $(schemafiles_DATA)
+EXTRA_DIST = $(modconfig_DATA)
 
 include $(top_srcdir)/Makefile.am.common
diff --git a/control/tests/Makefile.am b/control/tests/Makefile.am
index d7a786a..c928825 100644
--- a/control/tests/Makefile.am
+++ b/control/tests/Makefile.am
@@ -1,7 +1,8 @@
 # Makefile.am for installation/general
 
 client_DATA =					\
-	inst_test_workflow.ycp
+	inst_test_workflow.rb                   \
+        complete_workflow.rb
 
 EXTRA_DIST = $(client_DATA)
 
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index d153b93..0125218 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,15 +1,19 @@
 # Makefile.am for y2m_inst/doc/autodocs
 
-htmldir = $(docdir)/autodocs
+##htmldir = $(docdir)/autodocs
 
-html_DATA = $(wildcard *.html)
-CLEANFILES = $(html_DATA)
+#html_DATA = $(wildcard *.html)
+#CLEANFILES = $(html_DATA)
 
-all-local: index.html
+#all-local: index.html
 
 # Note that we look one _and_ two subdirectories deep.
 # There might be some duplicate filenames
 # but let's solve that later. For now we want to find all Modules::.
-index.html: $(srcdir)/../../src/*/*.ycp
-	${YCPDOC} -d. $^
+
+# TODO FIXME: temporarily disabled, enable back when autodocs
+# support is implemented
+
+#index.html: $(srcdir)/../../src/*/*.ycp
+#	${YCPDOC} -d. $^
 
diff --git a/doc/proposal/Makefile.am b/doc/proposal/Makefile.am
index a8733bf..58fc742 100644
--- a/doc/proposal/Makefile.am
+++ b/doc/proposal/Makefile.am
@@ -15,7 +15,7 @@ html/yast2docs.css: html/index.html
 	cp @STYLESHEET_CSS@ html
 	cp -a `pkg-config --variable=yast2dir yast2-devtools`/data/docbook/images html
 
-EXTRA_DIST =  $(xml_files) dummy_proposal.ycp
+EXTRA_DIST =  $(xml_files) dummy_proposal.rb
 
 CLEANFILES =  $(html_DATA)
 
diff --git a/doc/proposal/proposal-API-section.xml b/doc/proposal/proposal-API-section.xml
index 3ef84e6..80773e2 100644
--- a/doc/proposal/proposal-API-section.xml
+++ b/doc/proposal/proposal-API-section.xml
@@ -155,7 +155,7 @@
     <title>Dummy Proposal</title>
     <programlisting>
 
-      <xi:include href="dummy_proposal.ycp" parse="text"
+      <xi:include href="dummy_proposal.rb" parse="text"
 	xmlns:xi="http://www.w3.org/2001/XInclude"/>
 
     </programlisting>
diff --git a/src/clients/inst_automatic_configuration.ycp b/src/clients/inst_automatic_configuration.ycp
index 0689035..66b2263 100644
--- a/src/clients/inst_automatic_configuration.ycp
+++ b/src/clients/inst_automatic_configuration.ycp
@@ -369,7 +369,9 @@
 	    case "proposals":
 		CallProposals (one_autoconf_call["items"]:[]);
 		break;
-	    y2error ("Unknown script type '%1'", type);
+            default:
+                y2error ("Unknown script type '%1'", type);
+                break;
 	}
     });
 
diff --git a/yast2-installation.spec.in b/yast2-installation.spec.in
index 2ff4767..78d35a5 100644
--- a/yast2-installation.spec.in
+++ b/yast2-installation.spec.in
@@ -3,6 +3,8 @@
 @HEADER@
 Group:          System/YaST
 License:        GPL-2.0
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:        YaST2 - Installation Parts
 
 Source1:	YaST2-Second-Stage.service
@@ -167,9 +169,8 @@ install -m 644 %{SOURCE2} %{buildroot}%{_unitdir}
 %{_unitdir}/YaST2-Second-Stage.service
 %{_unitdir}/YaST2-Firstboot.service
 
-@clientdir@/*.ycp
-@moduledir@/*.ycp
-@moduledir@/*.ybc
+@clientdir@/*.rb
+@moduledir@/*.rb
 @desktopdir@/*.desktop
 /usr/share/autoinstall/modules/*.desktop
 /usr/share/YaST2/schema/autoyast/rnc/deploy_image.rnc
