diff --git a/httpd/Makefile.am b/httpd/Makefile.am
index 93ce8ad..1d261db 100644
--- a/httpd/Makefile.am
+++ b/httpd/Makefile.am
@@ -1,4 +1,7 @@
 apacheconfdir = /etc/apache2/conf.d
 apacheconf_DATA = inst_server.conf.in
 
+install-data-local:
+	install -d -m 755 $(DESTDIR)/etc/YaST2/instserver
+
 EXTRA_DIST = $(apacheconf_DATA) 
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-instserver.spec.in b/yast2-instserver.spec.in
index e452ea7..9e39b33 100644
--- a/yast2-instserver.spec.in
+++ b/yast2-instserver.spec.in
@@ -12,6 +12,8 @@ Requires:	yast2-slp
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Installation Server Configuration and Management
 
 %description
@@ -31,7 +33,7 @@ NFS sources are supported.
 %defattr(-,root,root)
 %dir @yncludedir@/instserver
 @yncludedir@/instserver/*
-@clientdir@/instserver.ycp
+@clientdir@/instserver.rb
 @moduledir@/Instserver.*
 @desktopdir@/instserver.desktop
 %doc @docdir@
