diff --git a/src/include/iscsi-client/complex.ycp b/src/include/iscsi-client/complex.ycp
index 378c26a..c90b295 100644
--- a/src/include/iscsi-client/complex.ycp
+++ b/src/include/iscsi-client/complex.ycp
@@ -50,10 +50,6 @@ boolean Modified() {
     return IscsiClient::Modified();
 }
 
-boolean ReallyAbort() {
-    return !IscsiClient::Modified() || Popup::ReallyAbort(true);
-}
-
 boolean PollAbort() {
     return UI::PollInput() == `abort;
 }
diff --git a/src/include/iscsi-client/dialogs.ycp b/src/include/iscsi-client/dialogs.ycp
index aa83de1..9b3639d 100644
--- a/src/include/iscsi-client/dialogs.ycp
+++ b/src/include/iscsi-client/dialogs.ycp
@@ -41,6 +41,7 @@ import "CWMServiceStart";
 import "CWMFirewallInterfaces";
 import "CWMTab";
 import "CWM";
+import "Popup";
 import "Stage";
 import "Mode";
 
@@ -322,6 +323,9 @@ map tabs_descr = $[
       ];
 
 
+boolean ReallyAbort() {
+    return !IscsiClient::Modified() || Popup::ReallyAbort(true);
+}
 
 // main tabbed dialog
 any GlobalDialog() {
diff --git a/src/include/iscsi-client/widgets.ycp b/src/include/iscsi-client/widgets.ycp
index 8c49b17..52fb8e9 100644
--- a/src/include/iscsi-client/widgets.ycp
+++ b/src/include/iscsi-client/widgets.ycp
@@ -24,6 +24,10 @@
 {
  textdomain "iscsi-client";
  import "IP";
+ import "IscsiClientLib";
+ import "Label";
+ import "Popup";
+ import "Stage";
 
  boolean stat = false;
  list<string> curr_rec = [];
@@ -153,6 +157,7 @@ symbol handleConnectedTable (string key, map event){
 		// goto DiscAuthDialog("client")) ()
 		y2milestone("Goto dicovered authentication dialog");
 		return `add;
+                break;
   case(`del):
 		// delete (logout from) connected target
 		record = setRecord();
