diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-iscsi-client.spec.in b/yast2-iscsi-client.spec.in
index d945497..04e8803 100644
--- a/yast2-iscsi-client.spec.in
+++ b/yast2-iscsi-client.spec.in
@@ -17,6 +17,8 @@ Requires:       yast2 >= 2.21.22
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - iSCSI Client Configuration
 
 %description
@@ -35,9 +37,9 @@ client.
 %defattr(-,root,root)
 %dir @yncludedir@/iscsi-client
 @yncludedir@/iscsi-client/*
-@clientdir@/iscsi-client.ycp
-@clientdir@/iscsi-client_*.ycp
-@clientdir@/inst_iscsi-client.ycp
+@clientdir@/iscsi-client.rb
+@clientdir@/iscsi-client_*.rb
+@clientdir@/inst_iscsi-client.rb
 @moduledir@/IscsiClient.*
 @moduledir@/IscsiClientLib.*
 @desktopdir@/iscsi-client.desktop
