diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index bb7ebdc..e34d7ca 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,5 @@
 # Makefile.am for iscsi-client/doc/autodocs
 
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled
+
+#include $(top_srcdir)/autodocs-ycp.ami
diff --git a/src/include/iscsi-client/complex.ycp b/src/include/iscsi-client/complex.ycp
index 378c26a..c90b295 100644
--- a/src/include/iscsi-client/complex.ycp
+++ b/src/include/iscsi-client/complex.ycp
@@ -50,10 +50,6 @@ boolean Modified() {
     return IscsiClient::Modified();
 }
 
-boolean ReallyAbort() {
-    return !IscsiClient::Modified() || Popup::ReallyAbort(true);
-}
-
 boolean PollAbort() {
     return UI::PollInput() == `abort;
 }
diff --git a/src/include/iscsi-client/dialogs.ycp b/src/include/iscsi-client/dialogs.ycp
index aa83de1..9b3639d 100644
--- a/src/include/iscsi-client/dialogs.ycp
+++ b/src/include/iscsi-client/dialogs.ycp
@@ -41,6 +41,7 @@ import "CWMServiceStart";
 import "CWMFirewallInterfaces";
 import "CWMTab";
 import "CWM";
+import "Popup";
 import "Stage";
 import "Mode";
 
@@ -322,6 +323,9 @@ map tabs_descr = $[
       ];
 
 
+boolean ReallyAbort() {
+    return !IscsiClient::Modified() || Popup::ReallyAbort(true);
+}
 
 // main tabbed dialog
 any GlobalDialog() {
diff --git a/src/include/iscsi-client/widgets.ycp b/src/include/iscsi-client/widgets.ycp
index 8c49b17..52fb8e9 100644
--- a/src/include/iscsi-client/widgets.ycp
+++ b/src/include/iscsi-client/widgets.ycp
@@ -24,6 +24,10 @@
 {
  textdomain "iscsi-client";
  import "IP";
+ import "IscsiClientLib";
+ import "Label";
+ import "Popup";
+ import "Stage";
 
  boolean stat = false;
  list<string> curr_rec = [];
@@ -153,6 +157,7 @@ symbol handleConnectedTable (string key, map event){
 		// goto DiscAuthDialog("client")) ()
 		y2milestone("Goto dicovered authentication dialog");
 		return `add;
+                break;
   case(`del):
 		// delete (logout from) connected target
 		record = setRecord();
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-iscsi-client.spec.in b/yast2-iscsi-client.spec.in
index d945497..04e8803 100644
--- a/yast2-iscsi-client.spec.in
+++ b/yast2-iscsi-client.spec.in
@@ -17,6 +17,8 @@ Requires:       yast2 >= 2.21.22
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - iSCSI Client Configuration
 
 %description
@@ -35,9 +37,9 @@ client.
 %defattr(-,root,root)
 %dir @yncludedir@/iscsi-client
 @yncludedir@/iscsi-client/*
-@clientdir@/iscsi-client.ycp
-@clientdir@/iscsi-client_*.ycp
-@clientdir@/inst_iscsi-client.ycp
+@clientdir@/iscsi-client.rb
+@clientdir@/iscsi-client_*.rb
+@clientdir@/inst_iscsi-client.rb
 @moduledir@/IscsiClient.*
 @moduledir@/IscsiClientLib.*
 @desktopdir@/iscsi-client.desktop
