diff --git a/agents/Makefile.am b/agents/Makefile.am
index 358303c..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,7 +0,0 @@
-# Makefile.am for iscsi-server/agents
-
-agent_SCRIPTS =
-
-scrconf_DATA = ietd.scr
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 6461f9c..6d99b70 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,5 @@
 # Makefile.am for iscsi-server/doc/autodocs
 
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled
+
+#include $(top_srcdir)/autodocs-ycp.ami
diff --git a/src/include/iscsi-server/complex.ycp b/src/include/iscsi-server/complex.ycp
index 6920367..30ae128 100644
--- a/src/include/iscsi-server/complex.ycp
+++ b/src/include/iscsi-server/complex.ycp
@@ -49,10 +49,6 @@ boolean Modified() {
     return IscsiServer::Modified();
 }
 
-boolean ReallyAbort() {
-    return !IscsiServer::Modified() || Popup::ReallyAbort(true);
-}
-
 boolean PollAbort() {
     return UI::PollInput() == `abort;
 }
diff --git a/src/include/iscsi-server/dialogs.ycp b/src/include/iscsi-server/dialogs.ycp
index 7d14f2f..0e9c26a 100644
--- a/src/include/iscsi-server/dialogs.ycp
+++ b/src/include/iscsi-server/dialogs.ycp
@@ -319,7 +319,9 @@ map<string, map <string, any> > widgets = $[
 	];
 
 
-
+boolean ReallyAbort() {
+    return !IscsiServer::Modified() || Popup::ReallyAbort(true);
+}
 
 /**
  * Summary dialog
diff --git a/src/include/iscsi-server/widgets.ycp b/src/include/iscsi-server/widgets.ycp
index 59ae6d6..5e10ff5 100644
--- a/src/include/iscsi-server/widgets.ycp
+++ b/src/include/iscsi-server/widgets.ycp
@@ -24,9 +24,11 @@
 {
 
 textdomain "iscsi-server";
+import "IP";
 import "IscsiServerFunctions";
 import "Label";
-import "IP";
+import "Popup";
+
 //	**************** global funcions and variables *****
 string curr_target = "";
 string modify_record = "";
diff --git a/src/include/iscsi-server/wizards.ycp b/src/include/iscsi-server/wizards.ycp
index 1141380..b2aee43 100644
--- a/src/include/iscsi-server/wizards.ycp
+++ b/src/include/iscsi-server/wizards.ycp
@@ -34,6 +34,7 @@
 
 textdomain "iscsi-server";
 
+import "Label";
 import "Sequencer";
 import "Wizard";
 
diff --git a/yast2-iscsi-server.spec.in b/yast2-iscsi-server.spec.in
index 94a45c6..7f968a1 100644
--- a/yast2-iscsi-server.spec.in
+++ b/yast2-iscsi-server.spec.in
@@ -30,8 +30,8 @@ Configuration of iSCSI target
 %defattr(-,root,root)
 %dir @yncludedir@/iscsi-server
 @yncludedir@/iscsi-server/*
-@clientdir@/iscsi-server.ycp
-@clientdir@/iscsi-server_*.ycp
+@clientdir@/iscsi-server.rb
+@clientdir@/iscsi-server_*.rb
 @moduledir@/IscsiServer*
 @desktopdir@/iscsi-server.desktop
 @scrconfdir@/ietd.scr
