diff --git a/agents/Makefile.am b/agents/Makefile.am
index c0ce51b..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,7 +0,0 @@
-# Makefile.am for isns/agents
-
-agent_SCRIPTS =
-
-scrconf_DATA = isnsd.scr
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index a5421e3..af859b1 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,5 @@
 # Makefile.am for isns/doc/autodocs
 
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled
+
+#include $(top_srcdir)/autodocs-ycp.ami
diff --git a/src/clients/isns.ycp b/src/clients/isns.ycp
index 0fd9c06..dd422f2 100644
--- a/src/clients/isns.ycp
+++ b/src/clients/isns.ycp
@@ -36,14 +36,14 @@ map cmdline_description = $[
     "initialize"        : IsnsServer::Read,
     "finish"            : IsnsServer::Write,
     "actions"           : $[
-	// FIXME TODO: fill the functionality description here
-    ],
+
+    ],	// FIXME TODO: fill the functionality description here
     "options"		: $[
-	// FIXME TODO: fill the option descriptions here
-    ],
+
+    ],	// FIXME TODO: fill the option descriptions here
     "mappings"		: $[
-	// FIXME TODO: fill the mappings of actions and options here
-    ]
+
+    ]	// FIXME TODO: fill the mappings of actions and options here
 ];
 
 /* is this proposal or not? */
diff --git a/src/include/isns/dialogs.ycp b/src/include/isns/dialogs.ycp
index 309ade0..244253d 100644
--- a/src/include/isns/dialogs.ycp
+++ b/src/include/isns/dialogs.ycp
@@ -22,6 +22,7 @@ import "TablePopup";
 
 include "isns/helps.ycp";
 include "isns/widgets.ycp";
+include "isns/complex.ycp";
 
 // store current here
 string current_tab = "service";
diff --git a/src/include/isns/widgets.ycp b/src/include/isns/widgets.ycp
index 3cbab5e..2e5d255 100644
--- a/src/include/isns/widgets.ycp
+++ b/src/include/isns/widgets.ycp
@@ -5,6 +5,7 @@ import "Label";
 import "IsnsServer";
 import "String";
 import "Report";
+import "Popup";
 
 string address = "";  //global
 
diff --git a/yast2-isns.spec.in b/yast2-isns.spec.in
index 6803626..046b103 100644
--- a/yast2-isns.spec.in
+++ b/yast2-isns.spec.in
@@ -7,6 +7,8 @@ BuildRequires:	perl-XML-Writer update-desktop-files yast2 yast2-devtools yast2-t
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	Configuration of isns
 
 %description
@@ -24,10 +26,9 @@ Summary:	Configuration of isns
 %defattr(-,root,root)
 %dir @yncludedir@/isns
 @yncludedir@/isns/*
-@clientdir@/isns.ycp
-@clientdir@/isns_*.ycp
-@moduledir@/*.ycp
-@moduledir@/*.ybc
+@clientdir@/isns.rb
+@clientdir@/isns_*.rb
+@moduledir@/*.rb
 @desktopdir@/isns.desktop
 @scrconfdir@/*.scr
 %doc @docdir@
