diff --git a/agents/Makefile.am b/agents/Makefile.am
index c0ce51b..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,7 +0,0 @@
-# Makefile.am for isns/agents
-
-agent_SCRIPTS =
-
-scrconf_DATA = isnsd.scr
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-isns.spec.in b/yast2-isns.spec.in
index 6803626..046b103 100644
--- a/yast2-isns.spec.in
+++ b/yast2-isns.spec.in
@@ -7,6 +7,8 @@ BuildRequires:	perl-XML-Writer update-desktop-files yast2 yast2-devtools yast2-t
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	Configuration of isns
 
 %description
@@ -24,10 +26,9 @@ Summary:	Configuration of isns
 %defattr(-,root,root)
 %dir @yncludedir@/isns
 @yncludedir@/isns/*
-@clientdir@/isns.ycp
-@clientdir@/isns_*.ycp
-@moduledir@/*.ycp
-@moduledir@/*.ybc
+@clientdir@/isns.rb
+@clientdir@/isns_*.rb
+@moduledir@/*.rb
 @desktopdir@/isns.desktop
 @scrconfdir@/*.scr
 %doc @docdir@
