diff --git a/agents/Makefile.am b/agents/Makefile.am
index cee9c90..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,7 +0,0 @@
-# Makefile.am for kdump/agents
-
-agent_SCRIPTS =
-
-scrconf_DATA = cfg_kdump.scr
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/config/Makefile.am b/config/Makefile.am
index 1fdb9de..e69de29 100644
--- a/config/Makefile.am
+++ b/config/Makefile.am
@@ -1,7 +0,0 @@
-# Makefile.am for source/kdump/config
-# $Id: $
-
-rncdir =  $(schemadir)/autoyast/rnc
-rnc_DATA = kdump.rnc
-
-EXTRA_DIST = $(rnc_DATA)
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 9454e58..35c4a63 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,5 @@
 # Makefile.am for kdump/doc/autodocs
 
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled
+
+# include $(top_srcdir)/autodocs-ycp.ami
diff --git a/src/include/kdump/uifunctions.ycp b/src/include/kdump/uifunctions.ycp
index 61fa93d..8e02d7c 100644
--- a/src/include/kdump/uifunctions.ycp
+++ b/src/include/kdump/uifunctions.ycp
@@ -17,6 +17,8 @@ import "Service";
 import "Package";
 import "PackageSystem";
 import "Mode";
+import "CommandLine";
+import "Message";
 
     // EXAMPLE FUNCTIONS
 /*
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-kdump.spec.in b/yast2-kdump.spec.in
index 01bcdc8..3bab3e5 100644
--- a/yast2-kdump.spec.in
+++ b/yast2-kdump.spec.in
@@ -12,6 +12,8 @@ Recommends:     makedumpfile
 BuildRequires:	yast2 >= 2.21.22
 BuildRequires: yast2-packager >= 2.17.24
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	Configuration of kdump
 
 %description
@@ -29,8 +31,8 @@ Configuration of kdump
 %defattr(-,root,root)
 %dir @yncludedir@/kdump
 @yncludedir@/kdump/*
-@clientdir@/kdump.ycp
-@clientdir@/kdump_*.ycp
+@clientdir@/kdump.rb
+@clientdir@/kdump_*.rb
 @moduledir@/Kdump.*
 @desktopdir@/kdump.desktop
 @schemadir@/autoyast/rnc/kdump.rnc
