diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-kerberos-server.spec.in b/yast2-kerberos-server.spec.in
index 1c43fd4..4cce044 100644
--- a/yast2-kerberos-server.spec.in
+++ b/yast2-kerberos-server.spec.in
@@ -10,6 +10,8 @@ BuildRequires:	perl-XML-Writer update-desktop-files yast2 yast2-devtools yast2-t
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Kerberos Server Configuration
 
 %description
@@ -28,7 +30,7 @@ Control Center.
 %defattr(-,root,root)
 %dir @yncludedir@/kerberos-server
 @yncludedir@/kerberos-server/*
-@clientdir@/kerberos-server.ycp
+@clientdir@/kerberos-server.rb
 @moduledir@/KerberosServer.*
 @desktopdir@/kerberos-server.desktop
 @scrconfdir@/*.scr
