diff --git a/SUBDIRS b/SUBDIRS
new file mode 100644
index 0000000..542fb19 100644
--- /dev/null
+++ b/SUBDIRS
@@ -0,0 +1,1 @@
+src lib agent tools doc
diff --git a/agent/Makefile.am b/agent/Makefile.am
index a86d0d8..c1c0ddf 100644
--- a/agent/Makefile.am
+++ b/agent/Makefile.am
@@ -1,7 +1,7 @@
 #
 # Makefile.am for SlapdConfig agent
 #
-AM_CPPFLAGS= -I$(top_srcdir)/src/lib  -I/usr/include/YaST2
+AM_CPPFLAGS= -I$(top_srcdir)/lib  -I/usr/include/YaST2
 AM_CXXFLAGS = -DY2LOG=\"agent-slapdconfig\" -Wall
 plugin_LTLIBRARIES = libpy2ag_slapdconfig.la
 noinst_LTLIBRARIES = liby2ag_slapdconfig.la
diff --git a/agents/Makefile.am b/agents/Makefile.am
deleted file mode 100644
index 05267a4..0000000
--- a/agents/Makefile.am
+++ /dev/null
@@ -1,8 +0,0 @@
-# Makefile.am for ldap-server/agents
-
-
-scrconf_DATA = \
-    ldapserver.scr  \
-    cfg_openldap.scr
-
-EXTRA_DIST = $(scrconf_DATA)
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index dca717a..0f67d40 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,4 +1,4 @@
 # Makefile.am for YCP module .../doc/autodocs
 
-AUTODOCS_PM = $(srcdir)/../../src/YaPI/*.pm
+AUTODOCS_PM = $(srcdir)/../../src/modules/YaPI/*.pm
 include $(top_srcdir)/autodocs-ycp.ami
diff --git a/src/Makefile.am b/src/Makefile.am
deleted file mode 100644
index 26f55d7..0000000
--- a/src/Makefile.am
+++ /dev/null
@@ -1,37 +0,0 @@
-# Makefile.am for ldap-server/src
-
-SUBDIRS = YaPI config lib agent tools
-
-yncludedir = @yncludedir@/ldap-server
-
-client_DATA =			\
-	ldap-server.ycp		\
-	openldap-mirrormode.ycp \
-	ldap-server_auto.ycp	\
-	ldap-server_proposal.ycp
-
-ynclude_DATA =			\
-	helps.ycp		\
-	wizards.ycp		\
-	widgets.ycp		\
-	dialogs.ycp		\
-	complex.ycp 		\
-	tree_structure.ycp	\
-	mirrormode-wizard.ycp
-
-desktop_DATA =			\
-	ldap-server.desktop	\
-	openldap-mirrormode.desktop
-
-module_DATA =			\
-	LdapServer.pm		\
-	LdapDatabase.ycp
-
-ldap-server:
-	ln -sf . ldap-server
-
-ycpchook = ldap-server 
-
-EXTRA_DIST = $(client_DATA) $(ynclude_DATA) $(module_DATA) $(desktop_DATA)
-
-include $(top_srcdir)/Makefile.am.common
diff --git a/yast2-ldap-server.spec.in b/yast2-ldap-server.spec.in
index 3b7d684..572317d 100644
--- a/yast2-ldap-server.spec.in
+++ b/yast2-ldap-server.spec.in
@@ -12,6 +12,8 @@ Requires:       yast2-users >= 2.22.3
 
 # Wizard::SetDesktopTitleAndIcon
 Requires:       yast2 >= 2.21.22
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - OpenLDAP Server Configuration
 
 %description
@@ -37,9 +39,9 @@ rm -f $RPM_BUILD_ROOT/@plugindir@/libpy2ag_slapdconfig.so
 %dir @yncludedir@/ldap-server
 %dir @moduledir@/YaPI
 @yncludedir@/ldap-server/*
-@clientdir@/ldap-server.ycp
-@clientdir@/openldap-mirrormode.ycp
-@clientdir@/ldap-server_*.ycp
+@clientdir@/ldap-server.rb
+@clientdir@/openldap-mirrormode.rb
+@clientdir@/ldap-server_*.rb
 @moduledir@/LdapServer.*
 @moduledir@/LdapDatabase.*
 @moduledir@/YaPI/LdapServer.pm
