diff --git a/src/include/ldap/ui.ycp b/src/include/ldap/ui.ycp
index e46187e..bf1d98d 100644
--- a/src/include/ldap/ui.ycp
+++ b/src/include/ldap/ui.ycp
@@ -749,6 +749,107 @@ Enable certificate checks now?")))
 	return result;
     }
 
+    define void set_client_term (boolean sssd_with_krb, string krb5_realm, string krb5_kdcip, list<string> ldap_schemas, string sssd_ldap_schema, boolean sssd_enumerate, boolean sssd_cache_credentials, list pam_password_items, list<term> member_attributes, boolean has_tabs) {
+
+
+	term cont = `Top (`HBox(`HSpacing (4), `VBox(
+	    `VSpacing (1),
+
+		// checkbox label
+		`Left (`CheckBox (`id (`sssd_with_krb), `opt (`notify), _("&Use Kerberos"), sssd_with_krb)),
+		`VSpacing(0.4),
+		`HBox (
+                    `HSpacing (2),
+		    // textentry label
+		    `TextEntry (`id (`krb5_realm), _("Default Real&m"), krb5_realm),
+		    // textentry label
+		    `TextEntry (`id (`krb5_kdcip), _("&KDC Server Address"), krb5_kdcip)
+		),
+		`VSpacing (),
+		// combobox label
+		`ComboBox (`id (`sssd_ldap_schema), `opt (`notify, `hstretch), _("LDAP Schema"),
+			maplist (string s, ldap_schemas, ``(`item (`id (s), s, s == sssd_ldap_schema)))
+		),
+		`VSpacing (0.4),
+		`HBox (
+		    `HSpacing (0.4),
+		    // checkbox label
+		    `Left (`CheckBox (`id (`sssd_enumerate), _("Enable user and group enumeration"),
+			sssd_enumerate))
+		),
+		`VSpacing (0.4),
+		`HBox (
+		    `HSpacing (0.4),
+		    // check box label
+		    `Left (`CheckBox (`id (`sssd_cache_credentials), _("SSSD O&ffline Authentication"),
+			sssd_cache_credentials))
+		),
+
+	    `VSpacing (),
+	    `ComboBox (`id (`pam_password), `opt(`notify,`hstretch,`editable),
+		// combobox label
+		_("Passwor&d Change Protocol"), pam_password_items),
+	    `ComboBox (`id (`group_style), `opt (`notify, `hstretch),
+		// combobox label
+		_("Group Member &Attribute"), member_attributes)
+	), `HSpacing (4)));
+
+	UI::ReplaceWidget (`tabContents, cont);
+	if (has_tabs)
+	    UI::ChangeWidget (`id (`tabs), `CurrentItem, `client);
+	if (Ldap::sssd)
+	{
+	    UI::ChangeWidget (`id (`krb5_realm), `Enabled, sssd_with_krb);
+	    UI::ChangeWidget (`id (`krb5_kdcip), `Enabled, sssd_with_krb);
+	}
+    }
+
+    define void set_admin_term (string base_config_dn, string bind_dn, boolean append_base, boolean create_ldap, boolean has_tabs) {
+
+	term cont = `HBox (`HSpacing (4), `VBox(
+	    `VSpacing (0.4),
+	    `HBox (
+		    `InputField (`id (`base_config_dn), `opt (`hstretch),
+			// textentry label
+			_("Configuration &Base DN"), base_config_dn),
+		    `VBox (
+			`Label (""),
+			// button label
+			`PushButton (`id(`br), `opt(`key_F6), _("Bro&wse"))
+		    )
+	    ),
+	    `VSpacing(0.4),
+	    `HBox (
+		`InputField (`id (`bind_dn),  `opt (`hstretch),
+		    // textentry label
+		    _("Administrator &DN"), bind_dn),
+		`VBox (
+		    `Label (""),
+		    // checkbox label
+		    `CheckBox (`id(`append),_("A&ppend Base DN"), append_base)
+		)
+	    ),
+	    `VSpacing (0.4),
+	    `Left (`CheckBox (`id(`create_ldap),
+		    // checkbox label
+		    _("Crea&te Default Configuration Objects"), create_ldap)),
+	    `VSpacing(0.5),
+	    `Right (`PushButton (`id(`configure),
+		    // pushbutton label
+		    _("Configure User Management &Settings..."))),
+            `VStretch ()
+	), `HSpacing (4));
+
+	UI::ReplaceWidget (`tabContents, cont);
+	if (has_tabs)
+	    UI::ChangeWidget (`id (`tabs), `CurrentItem, `admin);
+
+	// in autoyast-config mode, don't attach to server...
+	if (Mode::config ())
+	    UI::ChangeWidget (`id(`configure), `Enabled, false);
+
+    }
+
 
 
 /**
@@ -885,107 +986,6 @@ have changed your configuration.</p>
 
 
 
-    define void set_client_term () {
-
-
-	term cont = `Top (`HBox(`HSpacing (4), `VBox(
-	    `VSpacing (1),
-
-		// checkbox label
-		`Left (`CheckBox (`id (`sssd_with_krb), `opt (`notify), _("&Use Kerberos"), sssd_with_krb)),
-		`VSpacing(0.4),
-		`HBox (
-                    `HSpacing (2),
-		    // textentry label
-		    `TextEntry (`id (`krb5_realm), _("Default Real&m"), krb5_realm),
-		    // textentry label
-		    `TextEntry (`id (`krb5_kdcip), _("&KDC Server Address"), krb5_kdcip)
-		),
-		`VSpacing (),
-		// combobox label
-		`ComboBox (`id (`sssd_ldap_schema), `opt (`notify, `hstretch), _("LDAP Schema"),
-			maplist (string s, ldap_schemas, ``(`item (`id (s), s, s == sssd_ldap_schema)))
-		),
-		`VSpacing (0.4),
-		`HBox (
-		    `HSpacing (0.4),
-		    // checkbox label
-		    `Left (`CheckBox (`id (`sssd_enumerate), _("Enable user and group enumeration"),
-			sssd_enumerate))
-		),
-		`VSpacing (0.4),
-		`HBox (
-		    `HSpacing (0.4),
-		    // check box label
-		    `Left (`CheckBox (`id (`sssd_cache_credentials), _("SSSD O&ffline Authentication"),
-			sssd_cache_credentials))
-		),
-
-	    `VSpacing (),
-	    `ComboBox (`id (`pam_password), `opt(`notify,`hstretch,`editable),
-		// combobox label
-		_("Passwor&d Change Protocol"), pam_password_items),
-	    `ComboBox (`id (`group_style), `opt (`notify, `hstretch),
-		// combobox label
-		_("Group Member &Attribute"), member_attributes)
-	), `HSpacing (4)));
-
-	UI::ReplaceWidget (`tabContents, cont);
-	if (has_tabs)
-	    UI::ChangeWidget (`id (`tabs), `CurrentItem, `client);
-	if (Ldap::sssd)
-	{
-	    UI::ChangeWidget (`id (`krb5_realm), `Enabled, sssd_with_krb);
-	    UI::ChangeWidget (`id (`krb5_kdcip), `Enabled, sssd_with_krb);
-	}
-    }
-
-    define void set_admin_term () {
-
-	term cont = `HBox (`HSpacing (4), `VBox(
-	    `VSpacing (0.4),
-	    `HBox (
-		    `InputField (`id (`base_config_dn), `opt (`hstretch),
-			// textentry label
-			_("Configuration &Base DN"), base_config_dn),
-		    `VBox (
-			`Label (""),
-			// button label
-			`PushButton (`id(`br), `opt(`key_F6), _("Bro&wse"))
-		    )
-	    ),
-	    `VSpacing(0.4),
-	    `HBox (
-		`InputField (`id (`bind_dn),  `opt (`hstretch),
-		    // textentry label
-		    _("Administrator &DN"), bind_dn),
-		`VBox (
-		    `Label (""),
-		    // checkbox label
-		    `CheckBox (`id(`append),_("A&ppend Base DN"), append_base)
-		)
-	    ),
-	    `VSpacing (0.4),
-	    `Left (`CheckBox (`id(`create_ldap),
-		    // checkbox label
-		    _("Crea&te Default Configuration Objects"), create_ldap)),
-	    `VSpacing(0.5),
-	    `Right (`PushButton (`id(`configure),
-		    // pushbutton label
-		    _("Configure User Management &Settings..."))),
-            `VStretch ()
-	), `HSpacing (4));
-
-	UI::ReplaceWidget (`tabContents, cont);
-	if (has_tabs)
-	    UI::ChangeWidget (`id (`tabs), `CurrentItem, `admin);
-
-	// in autoyast-config mode, don't attach to server...
-	if (Mode::config ())
-	    UI::ChangeWidget (`id(`configure), `Enabled, false);
-
-    }
-
 
     // dialog label
     Wizard::SetContentsButtons (_("Advanced Configuration"), contents,
@@ -996,7 +996,7 @@ have changed your configuration.</p>
     any result	= `notnext;
     symbol current	= `client;
 
-    set_client_term ();
+    set_client_term (sssd_with_krb, krb5_realm, krb5_kdcip, ldap_schemas, sssd_ldap_schema, sssd_enumerate, sssd_cache_credentials, pam_password_items, member_attributes, has_tabs);
 
     while (true)
     {
@@ -1044,9 +1044,10 @@ have changed your configuration.</p>
 	    current = (symbol) result;
 	    Wizard::SetHelpText (help_text[current]:"");
 	    if (result == `client)
-		set_client_term ();
+                set_client_term (sssd_with_krb, krb5_realm, krb5_kdcip, ldap_schemas, sssd_ldap_schema, sssd_enumerate, sssd_cache_credentials, pam_password_items, member_attributes, has_tabs);
 	    else
-		set_admin_term ();
+                set_admin_term (base_config_dn, bind_dn, append_base, create_ldap, has_tabs);
+
 	}
 
 	// 3. other events
@@ -1201,6 +1202,19 @@ all changes will be lost. Really reread?
     return `next;
 }
 
+    // helper function converting list value to string
+    define string to_table (string attr, list<string> val) ``{
+
+	if (Ldap::SingleValued (attr) || attr == "cn")
+	    return val[0]:"";
+	else if (contains (
+	    ["susesecondarygroup", "susedefaulttemplate"], tolower(attr)))
+	    return mergestring (val, " ");
+	else
+	    return mergestring (val, ",");
+    }
+
+
 /**
  * Dialog for configuration one object template
  */
@@ -1227,18 +1241,6 @@ removing current ones.</p>
     list table_items = [];
     map<string,any> template = (map<string,any>) eval (templ);
 
-    // helper function converting list value to string
-    define string to_table (string attr, list<string> val) ``{
-
-	if (Ldap::SingleValued (attr) || attr == "cn")
-	    return val[0]:"";
-	else if (contains (
-	    ["susesecondarygroup", "susedefaulttemplate"], tolower(attr)))
-	    return mergestring (val, " ");
-	else
-	    return mergestring (val, ",");
-    }
-
     foreach (string attr, any value, template, ``{
 	any val = value;
 	// do not show internal attributes
@@ -1428,46 +1430,10 @@ Enter a value."), attr));
     return template;
 }
 
-/**
- * Dialog for configuration of one "configuration module"
- */
-define symbol ModuleConfigurationDialog () ``{
-
-    // helptext 1/4
-    string help_text = _("<p>Manage the configuration stored in the LDAP directory.</p>") +
-
-    // helptext 2/4
-    _("<p>Each configuration set is called a \"configuration module.\" If there
-is no configuration module in the provided location (base configuration),
-create one with <b>New</b>. Delete the current module
-using <b>Delete</b>.</p>
-") +
-
-    // helptext 3/4
-    _("<p>Edit the values of attributes in the table with <b>Edit</b>.
-Some values have special meanings, for example, changing the <b>cn</b> value renames the
-current module.</p>
-") +
-
-    // helptext 4/4
-    _("<p>To configure the default template of the current module,
-click <b>Configure Template</b>.
-</p>
-");
-
-    string current_dn		= Ldap::current_module_dn;
-    map  modules_attrs_items	= $[]; // map of list (table items), index is cn
-    map<string,map<string,any> > modules	=
-	(map<string,map<string,any> >) Ldap::GetConfigModules ();
-    map<string,map<string,any> > templates	=
-	(map<string,map<string,any> >) Ldap::GetTemplates ();
-    list<string> names		= [];
-    list<string> templates_dns	= maplist (string dn, map t, templates, ``(dn));
-
     /**
      * Helper for creating table items in ModuleConfiguration Dialog
      */
-    define list create_attrs_items (string cn) ``{
+    list create_attrs_items (string cn, map<string,map<string,any> > modules) {
 
 	list attrs_items = [];
 	string dn	= get_dn (cn);
@@ -1491,22 +1457,11 @@ click <b>Configure Template</b>.
 	return attrs_items;
     }
 
-    foreach (string dn, map mod, modules, ``{
-	string cn = get_string (mod, "cn");
-	if (cn == "")
-	    return;
-	names = add (names, cn);
-	// attributes for table
-	modules_attrs_items [cn] = create_attrs_items (cn);
-	if (current_dn == "")
-	    current_dn = dn;
-    });
-    string current_cn = modules [current_dn, "cn",0]:get_cn (current_dn);
 
     /**
      * Helper for updating widgets in ModuleConfiguration Dialog
      */
-    define void replace_module_names () ``{
+    void replace_module_names (list<string> names, string current_cn) {
 
 	list modules_items = []; // list of module names
 	foreach (string cn, names, ``{
@@ -1528,7 +1483,7 @@ click <b>Configure Template</b>.
     /**
      * Helper for updating widgets in ModuleConfiguration Dialog
      */
-    define void replace_templates_items () ``{
+    void replace_templates_items (map<string,map<string,any> > modules, string current_dn) {
 
 	list items = maplist (string dn,
 	    modules[current_dn,"suseDefaultTemplate"]:[],
@@ -1541,6 +1496,54 @@ click <b>Configure Template</b>.
 	UI::ChangeWidget (`id(`templ_pb), `Enabled, items != []);
     }
 
+/**
+ * Dialog for configuration of one "configuration module"
+ */
+define symbol ModuleConfigurationDialog () ``{
+
+    // helptext 1/4
+    string help_text = _("<p>Manage the configuration stored in the LDAP directory.</p>") +
+
+    // helptext 2/4
+    _("<p>Each configuration set is called a \"configuration module.\" If there
+is no configuration module in the provided location (base configuration),
+create one with <b>New</b>. Delete the current module
+using <b>Delete</b>.</p>
+") +
+
+    // helptext 3/4
+    _("<p>Edit the values of attributes in the table with <b>Edit</b>.
+Some values have special meanings, for example, changing the <b>cn</b> value renames the
+current module.</p>
+") +
+
+    // helptext 4/4
+    _("<p>To configure the default template of the current module,
+click <b>Configure Template</b>.
+</p>
+");
+
+    string current_dn		= Ldap::current_module_dn;
+    map  modules_attrs_items	= $[]; // map of list (table items), index is cn
+    map<string,map<string,any> > modules	=
+	(map<string,map<string,any> >) Ldap::GetConfigModules ();
+    map<string,map<string,any> > templates	=
+	(map<string,map<string,any> >) Ldap::GetTemplates ();
+    list<string> names		= [];
+    list<string> templates_dns	= maplist (string dn, map t, templates, ``(dn));
+
+    foreach (string dn, map mod, modules, ``{
+	string cn = get_string (mod, "cn");
+	if (cn == "")
+	    return;
+	names = add (names, cn);
+	// attributes for table
+	modules_attrs_items [cn] = create_attrs_items (cn, modules);
+	if (current_dn == "")
+	    current_dn = dn;
+    });
+    string current_cn = modules [current_dn, "cn",0]:get_cn (current_dn);
+
     term contents = `HBox(`HSpacing (1.5), `VBox(
         `VSpacing(0.5),
 	`HBox (
@@ -1579,8 +1582,8 @@ click <b>Configure Template</b>.
     {
 	UI::SetFocus (`id(`table));
     }
-    replace_templates_items ();
-    replace_module_names ();
+    replace_templates_items (modules, current_dn);
+    replace_module_names (names, current_cn);
 
     // result could be symbol or string
     any result = nil;
@@ -1616,7 +1619,7 @@ Enter a value."), attr));
 		current_dn	= tolower (current_dn);
 	    UI::ChangeWidget (`id(`table), `Items,
 		modules_attrs_items[current_cn]:[]);
-	    replace_templates_items ();
+	    replace_templates_items (modules, current_dn);
 	}
 	// delete the module
 	if (result == `delete)
@@ -1634,8 +1637,8 @@ Enter a value."), attr));
 	    current_dn = get_dn (current_cn);
 	    if (!haskey (modules, current_dn))
 		current_dn	= tolower (current_dn);
-	    replace_module_names ();
-	    replace_templates_items ();
+	    replace_module_names (names, current_cn);
+	    replace_templates_items (modules, current_dn);
 	    UI::ChangeWidget (`id(`table), `Items,
 		modules_attrs_items[current_cn]:[]);
 	}
@@ -1670,9 +1673,9 @@ type, therefore you cannot add a new one.
 		current_dn	= tolower (current_dn);
 	    modules [current_dn] = Ldap::CreateModule (cn, new ["class"]:"");
 	    names = add (names, cn);
-	    modules_attrs_items [cn] = create_attrs_items (cn);
-	    replace_module_names ();
-	    replace_templates_items ();
+	    modules_attrs_items [cn] = create_attrs_items (cn, modules);
+	    replace_module_names (names, current_cn);
+	    replace_templates_items (modules, current_dn);
 	    UI::ChangeWidget (`id(`table), `Items,
 		modules_attrs_items[current_cn]:[]);
 	}
@@ -1706,7 +1709,7 @@ type, therefore you cannot add a new one.
 	    if (value == modules [current_dn, attr]:[]) //nothing was changed
 		continue;
 	    modules [current_dn, attr] = value;
-	    modules_attrs_items [current_cn] = create_attrs_items (current_cn);
+	    modules_attrs_items [current_cn] = create_attrs_items (current_cn, modules);
 	    UI::ChangeWidget (`id(`table), `Items,
 		modules_attrs_items[current_cn]:[]);
 	    UI::ChangeWidget (`id(`table), `CurrentItem, attr);
@@ -1728,10 +1731,10 @@ type, therefore you cannot add a new one.
 		names = add (names, cn);
 		current_cn = cn;
 		current_dn = get_dn (cn);
-		replace_module_names ();
+		replace_module_names (names, current_cn);
 	    }
 	    if (tolower (attr) == "susedefaulttemplate")
-		replace_templates_items ();
+		replace_templates_items (modules, current_dn);
 	}
 	// configure template
 	if (result == `templ_pb)
@@ -1779,10 +1782,10 @@ type, therefore you cannot add a new one.
 			return dn;
 		});
 		modules_attrs_items [current_cn] =
-		    create_attrs_items (current_cn);
+                  create_attrs_items (current_cn, modules);
 		UI::ChangeWidget (`id(`table), `Items,
 		    modules_attrs_items[current_cn]:[]);
-		replace_templates_items ();
+		replace_templates_items (modules, current_dn);
 	    }
 	    UI::SetFocus (`id(`table));
 	}
diff --git a/src/modules/LdapPopup.ycp b/src/modules/LdapPopup.ycp
index 76100cd..e4322d9 100644
--- a/src/modules/LdapPopup.ycp
+++ b/src/modules/LdapPopup.ycp
@@ -39,6 +39,26 @@ import "Label";
 import "Popup";
 import "Wizard";
 
+    define list<term> update_items (list<term> its, string current_dn, list<string> &subdns, map &open_items, boolean textmode) {
+
+       return maplist (term it, its, {
+           string dn	= it[0]:"";
+           if (dn == current_dn)
+           {
+               return `item (dn, true, maplist (string k,
+                                         subdns, ``(`item (k, false, []))));
+           }
+           integer last = size (it) - 1;
+           if (size (it[last]:[]) == 0)
+               return it;
+           // `OpenItems doesn't work in ncurses...
+           boolean open = haskey (open_items, dn) && !textmode;
+           return `item (dn, open,
+                         update_items (it[last]:[],
+                                       current_dn, subdns, open_items, textmode));
+         });
+     }
+
     /**
      * Popup for browsing LDAP tree and selecting the DN
      * WARNING we expect that LDAP connection is already correctly initialized !
@@ -129,24 +149,6 @@ import "Wizard";
 	list<string> subdns	= [];
 	map open_items		= $[];
 
-	define list<term> update_items (list<term> its) {
-
-	    return maplist (term it, its, {
-		string dn	= it[0]:"";
-		if (dn == current_dn)
-		{
-		    return `item (dn, true, maplist (string k,
-			subdns, ``(`item (k, false, []))));
-		}
-		integer last = size (it) - 1;
-		if (size (it[last]:[]) == 0)
-		    return it;
-		// `OpenItems doesn't work in ncurses...
-		boolean open = haskey (open_items, dn) && !textmode;
-		return `item (dn, open, update_items (it[last]:[]));
-	    });
-	}
-
 	string retval = root_dn;
 	while (true)
 	{
@@ -166,7 +168,7 @@ import "Wizard";
 		    if (size (subdns) > 0)
 		    {
 			open_items = (map) UI::QueryWidget (`tree, `OpenItems);
-			items	= update_items (items);
+			items	= update_items (items, current_dn, subdns, open_items, textmode);
 			UI::ReplaceWidget (`id (`reptree), textmode ?
 			    `Tree (`id(`tree), root_dn, items) :
 			    `Tree (`id(`tree), `opt(`notify), root_dn, items));
@@ -222,6 +224,38 @@ import "Wizard";
 	return BrowseTree (root_dn);
     }
 
+    /**
+     * Helper for creating items for EditAttribute Popup
+     * The result term is returned by reference
+     * so that the return value is left for an integer
+     * @return new hsize
+     */
+    define integer generate_value_list (term &ret, boolean single, string attr, list<string> &value, integer value_size, integer hsize) {
+        ret = `VBox();
+	if (single)
+	{
+	    ret = add (ret, `InputField (`id (0), `opt (`hstretch),
+		// textentry label
+		sformat(_("&Value of \"%1\" Attribute"), attr), value[0]:""));
+	}
+	else
+	{
+	    ret = add (ret, `InputField (`id (0), `opt (`hstretch),
+		// textentry label
+		sformat(_("&Values of \"%1\" Attribute"), attr), value[0]:""));
+	    integer i = 1;
+	    while (i < value_size)
+	    {
+		ret = add (ret, `InputField (`id (i), `opt (`hstretch),
+		    "", value[i]:""));
+		if (size (value[i]:"" + 10) > hsize)
+		    hsize = size (value[i]:"") + 10;
+		i = i + 1;
+	    }
+	}
+	return hsize;
+    }
+
 
 /**
  * Generic popup for editing attribute's value
@@ -277,36 +311,8 @@ it is possible to choose it from the LDAP tree using <b>Browse</b>.
     // horizontal size of popup for
     integer hsize = size (value[0]:"") + 10;
 
-    /**
-     * Helper for creating items for EditAttribute Popup
-     */
-    define term generate_value_list () ``{
-	term ret = `VBox();
-	if (single)
-	{
-	    ret = add (ret, `InputField (`id (0), `opt (`hstretch),
-		// textentry label
-		sformat(_("&Value of \"%1\" Attribute"), attr), value[0]:""));
-	}
-	else
-	{
-	    ret = add (ret, `InputField (`id (0), `opt (`hstretch),
-		// textentry label
-		sformat(_("&Values of \"%1\" Attribute"), attr), value[0]:""));
-	    integer i = 1;
-	    while (i < value_size)
-	    {
-		ret = add (ret, `InputField (`id (i), `opt (`hstretch),
-		    "", value[i]:""));
-		if (size (value[i]:"" + 10) > hsize)
-		    hsize = size (value[i]:"") + 10;
-		i = i + 1;
-	    }
-	}
-	return ret;
-    }
-
-    term values = generate_value_list ();
+    term values = nil;
+    hsize = generate_value_list (values, single, attr, value, value_size, hsize);
     // button label
     term add_button = `PushButton(`id(`new), `opt (`key_F3), _("&Add Value"));
 
@@ -383,7 +389,9 @@ it is possible to choose it from the LDAP tree using <b>Browse</b>.
 		value = add (value, (result == `new)? "" : (string) result);
 		value_size = value_size + 1;
 	    }
-	    UI::ReplaceWidget (`id(`rp), generate_value_list ());
+            term values = nil;
+            hsize = generate_value_list (values, single, attr, value, value_size, hsize);
+            UI::ReplaceWidget (`id(`rp), values);
 	    UI::SetFocus (`id(value_size-1));
 	}
 	if (result == `ok)
