diff --git a/agents/Makefile.am b/agents/Makefile.am
deleted file mode 100644
index 26da395..0000000
--- a/agents/Makefile.am
+++ /dev/null
@@ -1,13 +0,0 @@
-#
-# Makefile.am for y2c_ldap_client/agents
-#
-# $Id$
-#
-
-scrconf_DATA = ldap_conf.scr    \
-	       cfg_ldap.scr     \
-               nscd_conf.scr
-
-fillup_DATA = sysconfig.ldap
-
-EXTRA_DIST = $(scrconf_DATA) $(fillup_DATA)
diff --git a/doc/Makefile.am b/doc/Makefile.am
index dc4947b..94296e3 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -2,7 +2,8 @@
 # Makefile.am for y2c_nisplus/doc
 #
 
-SUBDIRS = autodocs
+# ycpdoc does not work on ruby
+#SUBDIRS = autodocs
 
 htmldir = @docdir@
 html_DATA =
diff --git a/src/Makefile.am b/src/Makefile.am
deleted file mode 100644
index 39c0d4c..0000000
--- a/src/Makefile.am
+++ /dev/null
@@ -1,41 +0,0 @@
-#
-# Makefile.am for ldap-client/src
-#
-# $Id$
-#
-
-yncludedir = @yncludedir@/ldap
-
-rncdir =  $(schemadir)/autoyast/rnc
-
-YCPCFLAGS = -I .
-
-client_DATA =		\
-	ldap.ycp	\
-	ldap_auto.ycp	\
-	ldap-client.ycp	\
-	ldap_config.ycp	\
-	ldap_browser.ycp
-
-ynclude_DATA =		\
-	ui.ycp		\
-	wizards.ycp	\
-	routines.ycp
-
-module_DATA =		\
-	Ldap.ycp	\
-	LdapPopup.ycp
-
-rnc_DATA =              \
-	ldap_client.rnc
-
-desktop_DATA = ldap.desktop ldap_browser.desktop
-
-EXTRA_DIST = $(client_DATA) $(ynclude_DATA) $(module_DATA) $(desktop_DATA) $(rnc_DATA)
-
-ldap:
-	ln -sf . ldap
-
-ycpchook = ldap
-
-include $(top_srcdir)/Makefile.am.common
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-ldap-client.spec.in b/yast2-ldap-client.spec.in
index 32deeaf..dc784d1 100644
--- a/yast2-ldap-client.spec.in
+++ b/yast2-ldap-client.spec.in
@@ -27,6 +27,8 @@ Obsoletes:	yast2-trans-ldap_client
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - LDAP Client Configuration
 
 %description
@@ -50,9 +52,9 @@ OpenLDAP server will be used for user authentication.
 @desktopdir@/ldap_browser.desktop
 %dir @yncludedir@/ldap
 @yncludedir@/ldap/*
-@moduledir@/Ldap.y*
-@moduledir@/LdapPopup.y*
-@clientdir@/ldap*.ycp
+@moduledir@/Ldap.rb
+@moduledir@/LdapPopup.rb
+@clientdir@/ldap*.rb
 @scrconfdir@/*.scr
 @schemadir@/autoyast/rnc/ldap_client.rnc
 %doc @docdir@
