diff --git a/agents/Makefile.am b/agents/Makefile.am
index 6b40188..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,7 +0,0 @@
-# Makefile.am for live-installer/agents
-
-agent_SCRIPTS =
-
-scrconf_DATA =
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 1b55e6e..0c9cbf9 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,5 @@
 # Makefile.am for live-installer/doc/autodocs
 
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled
+
+#include $(top_srcdir)/autodocs-ycp.ami
diff --git a/src/clients/inst_live_simple_proposal.ycp b/src/clients/inst_live_simple_proposal.ycp
index 7cecb5d..cea856a 100644
--- a/src/clients/inst_live_simple_proposal.ycp
+++ b/src/clients/inst_live_simple_proposal.ycp
@@ -157,14 +157,14 @@ contents = `VBox (
 
 // help text 1/3
 string help = _("<p>
-Use <b>Accept</b> to perform a new installation with the values displayed.</p>")
+Use <b>Accept</b> to perform a new installation with the values displayed.</p>") +
 // help text 2/3
-+ _("<p>
+_("<p>
 To change the values, click the respective headline
 or select <b>Change Installation Settings</b>.</p>
-")
+") +
 // help text 3/3
-+ _("<p>
+_("<p>
 Your hard disk has not been modified, you can still safely abort.
 </p>");
 
diff --git a/yast2-live-installer.spec.in b/yast2-live-installer.spec.in
index bff3e12..ab64197 100644
--- a/yast2-live-installer.spec.in
+++ b/yast2-live-installer.spec.in
@@ -16,6 +16,8 @@ BuildRequires:	perl-XML-Writer update-desktop-files yast2 yast2-devtools yast2-t
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Installation from Live Media
 
 %description
@@ -32,7 +34,7 @@ hard disk of the computer.
 
 %files
 %defattr(-,root,root)
-@clientdir@/*.ycp
+@clientdir@/*.rb
 @moduledir@/LiveInstaller.*
 @desktopdir@/live-installer.desktop
 %doc @docdir@
