diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-lxc.spec.in b/yast2-lxc.spec.in
index 11e214c..4d2cee5 100644
--- a/yast2-lxc.spec.in
+++ b/yast2-lxc.spec.in
@@ -6,6 +6,8 @@ BuildRequires:	perl-XML-Writer update-desktop-files yast2 yast2-devtools yast2-t
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	Management tool for Linux Containers (LXC)
 
 %description
@@ -23,7 +25,7 @@ Graphical management tool for Linux Containers (LXC)
 %defattr(-,root,root)
 %dir @yncludedir@/lxc
 @yncludedir@/lxc/*
-@clientdir@/lxc.ycp
+@clientdir@/lxc.rb
 @moduledir@/Lxc.*
 @desktopdir@/lxc.desktop
 %doc @docdir@
