diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 2175008..05b5d02 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,5 @@
 # Makefile.am for lxc/doc/autodocs
 
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled
+
+#include $(top_srcdir)/autodocs-ycp.ami
diff --git a/src/include/lxc/dialogs.ycp b/src/include/lxc/dialogs.ycp
index a58f723..4206737 100644
--- a/src/include/lxc/dialogs.ycp
+++ b/src/include/lxc/dialogs.ycp
@@ -40,6 +40,7 @@ import "Report";
 import "Security";
 import "String";
 import "Wizard";
+import "Lxc";
 
 include "lxc/helps.ycp";
 
diff --git a/yast2-lxc.spec.in b/yast2-lxc.spec.in
index 11e214c..844928a 100644
--- a/yast2-lxc.spec.in
+++ b/yast2-lxc.spec.in
@@ -23,7 +23,7 @@ Graphical management tool for Linux Containers (LXC)
 %defattr(-,root,root)
 %dir @yncludedir@/lxc
 @yncludedir@/lxc/*
-@clientdir@/lxc.ycp
+@clientdir@/lxc.rb
 @moduledir@/Lxc.*
 @desktopdir@/lxc.desktop
 %doc @docdir@
