diff --git a/SUBDIRS b/SUBDIRS
index 922f4d3..ea3a6ee 100644
--- a/SUBDIRS
+++ b/SUBDIRS
@@ -1 +1 @@
-agents doc permissions src users-plugin
+agents doc permissions src users-plugin schema
diff --git a/agents/Makefile.am b/agents/Makefile.am
index 5a6dbe5..fd6424b 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -4,38 +4,6 @@
 # $Id$
 #
 
-agent_SCRIPTS = ag_fetchmailrc \
-		ag_smtp_auth \
-		ag_mailtable \
-		setup_dkim_verifying.pl \
-		ag_mailconfig \
-		MasterCFParser.pm \
-		ag_mail_ldaptable \
-                ag_cyrus_conf \
-		CyrusConfParser.pm
-
-scrconf_DATA = \
-	cfg_amavis.scr		\
-	cfg_mail.scr		\
-	cfg_postfix.scr		\
-	cfg_sendmail.scr	\
-	mail_aliases.scr	\
-	mail_fetchmail.scr	\
-	mail_p_auth.scr		\
-	mail_p_canonical.scr	\
-	mail_p_virtual.scr	\
-	mail_s_auth.scr		\
-	mail_s_generics.scr	\
-	mail_s_virtuser.scr     \
-	\
-	mail_maincf.scr \
-	mail_saslpasswd.scr \
-	etc_imapd_conf.scr \
-        mail_ldaptable.scr \
-	mail_cyrusconf.scr \
-	cfg_fetchmail.scr
-
-fillup_DATA =  sysconfig.mail
 
 # ag_fetchmailrc is tested in testsuite dir
 TESTS = test-mailtable
@@ -48,4 +16,4 @@ testdata = \
 	test-mailtable-4.in test-mailtable-4.out \
 	test-mailtable-18487.in test-mailtable-18487.out
 
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA) $(TESTS) $(testdata) $(fillup_DATA)
+EXTRA_DIST = $(TESTS) $(testdata)
diff --git a/schema/Makefile.am b/schema/Makefile.am
index 5093469..749104b 100644
--- a/schema/Makefile.am
+++ b/schema/Makefile.am
@@ -4,51 +4,10 @@
 # $Id$
 #
 
-SUBDIRS = YaPI
-
-myyncludedir = $(yncludedir)/mail
 ldapschemadir  = /etc/openldap/schema
 
-client_DATA =			\
-	mail.ycp		\
-	mail_auto.ycp		\
-        mail-standard.ycp       \
-        mail-advanced.ycp       \
-        mail-server_auto.ycp    \
-	mail-server_ldap-setup.ycp \
-        mail-server_proposal.ycp \
-	mail-server_update.ycp
-
-myynclude_DATA =		\
-	helps.ycp		\
-	wj.ycp			\
-	widgets.ycp		\
-	ui.ycp			\
-	\
-	mail-server_helps.ycp   \
-        mail-server_wizards.ycp \
-        mail-server_dialogs.ycp \
-        mail-server_complex.ycp
-
-desktop_DATA =			\
-	mail.desktop
-
-module_DATA =			\
-	Mail.ycp		\
-	MailServer.ycp		\
-	MailServerLDAP.pm
-
 ldapschema_DATA = suse-mailserver.schema
 
-rncdir =  $(schemadir)/autoyast/rnc
-rnc_DATA = mail.rnc
-
-mail:
-	ln -sf . mail
-
-ycpchook = mail
-
-EXTRA_DIST = $(client_DATA) $(myynclude_DATA) $(module_DATA) $(desktop_DATA) $(rnc_DATA) \
-             $(ldapschema_DATA)
+EXTRA_DIST = $(ldapschema_DATA)
 
 include $(top_srcdir)/Makefile.am.common
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index 02a627f..06bc411 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -4,10 +4,10 @@
 
 AUTOMAKE_OPTIONS = dejagnu
 
-ycp_testfiles = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+ycp_testfiles = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 epag_misc = lib/yast2_epag.exp run/runtest-epag.sh
-epag_testfiles = $(wildcard $(RPMNAME).test/ag_*.exp tests.epag/*.[rw].run tests.epag/*.[rw].data tests.epag/*.rw.ycp)
+epag_testfiles = $(wildcard $(RPMNAME).test/ag_*.exp tests.epag/*.[rw].run tests.epag/*.[rw].data tests.epag/*.rw.rb)
 EXTRA_DIST = $(ycp_testfiles) $(epag_testfiles) $(epag_misc)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
diff --git a/yast2-mail.spec.in b/yast2-mail.spec.in
index b85d006..a9fa5e1 100644
--- a/yast2-mail.spec.in
+++ b/yast2-mail.spec.in
@@ -28,6 +28,8 @@ Provides:       yast2-mail-server = %{version}
 Obsoletes:      yast2-mail-server <= %{version}
 Conflicts:	aaa_base < 10.3
 BuildArch:	noarch
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Mail Configuration
 
 %description
@@ -35,6 +37,8 @@ The YaST2 component for mail configuration. It handles Postfix, Cyrus,
 Amavis and Fetchmail.
 
 %package plugins
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Users/Group Plugins for the mail delivery configuration
 Group:		System/YaST
 Requires:       perl-NetxAP acl
@@ -44,7 +48,6 @@ Plugins for the YaST2 users module for enterprise mail server
 configuration.
 
 @PREP@
-cp agents/MasterCFParser.pm src
 
 @BUILD@
 
@@ -62,11 +65,9 @@ cp agents/MasterCFParser.pm src
 %dir @clientdir@
 @clientdir@/mail*
 %dir @moduledir@
-@moduledir@/Mail.ybc
-@moduledir@/MailServer.ybc
-@moduledir@/MailServer.ycp
+@moduledir@/MailServer.rb
 @moduledir@/MailServerLDAP.pm
-@moduledir@/Mail.ycp
+@moduledir@/Mail.rb
 %dir @moduledir@/YaPI
 @moduledir@/YaPI/Mail*
 %dir @desktopdir@
