diff --git a/SUBDIRS b/SUBDIRS
index 922f4d3..ea3a6ee 100644
--- a/SUBDIRS
+++ b/SUBDIRS
@@ -1 +1 @@
-agents doc permissions src users-plugin
+agents doc permissions src users-plugin schema
diff --git a/agents/Makefile.am b/agents/Makefile.am
index 5a6dbe5..fd6424b 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -4,38 +4,6 @@
 # $Id$
 #
 
-agent_SCRIPTS = ag_fetchmailrc \
-		ag_smtp_auth \
-		ag_mailtable \
-		setup_dkim_verifying.pl \
-		ag_mailconfig \
-		MasterCFParser.pm \
-		ag_mail_ldaptable \
-                ag_cyrus_conf \
-		CyrusConfParser.pm
-
-scrconf_DATA = \
-	cfg_amavis.scr		\
-	cfg_mail.scr		\
-	cfg_postfix.scr		\
-	cfg_sendmail.scr	\
-	mail_aliases.scr	\
-	mail_fetchmail.scr	\
-	mail_p_auth.scr		\
-	mail_p_canonical.scr	\
-	mail_p_virtual.scr	\
-	mail_s_auth.scr		\
-	mail_s_generics.scr	\
-	mail_s_virtuser.scr     \
-	\
-	mail_maincf.scr \
-	mail_saslpasswd.scr \
-	etc_imapd_conf.scr \
-        mail_ldaptable.scr \
-	mail_cyrusconf.scr \
-	cfg_fetchmail.scr
-
-fillup_DATA =  sysconfig.mail
 
 # ag_fetchmailrc is tested in testsuite dir
 TESTS = test-mailtable
@@ -48,4 +16,4 @@ testdata = \
 	test-mailtable-4.in test-mailtable-4.out \
 	test-mailtable-18487.in test-mailtable-18487.out
 
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA) $(TESTS) $(testdata) $(fillup_DATA)
+EXTRA_DIST = $(TESTS) $(testdata)
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index df3ffca..9de7be0 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,2 +1,5 @@
 # Makefile.am for YCP module .../doc/autodocs
-include $(top_srcdir)/autodocs-ycp.ami
+
+# TODO FIXME: temporarily disabled
+
+#include $(top_srcdir)/autodocs-ycp.ami
diff --git a/schema/Makefile.am b/schema/Makefile.am
index 5093469..749104b 100644
--- a/schema/Makefile.am
+++ b/schema/Makefile.am
@@ -4,51 +4,10 @@
 # $Id$
 #
 
-SUBDIRS = YaPI
-
-myyncludedir = $(yncludedir)/mail
 ldapschemadir  = /etc/openldap/schema
 
-client_DATA =			\
-	mail.ycp		\
-	mail_auto.ycp		\
-        mail-standard.ycp       \
-        mail-advanced.ycp       \
-        mail-server_auto.ycp    \
-	mail-server_ldap-setup.ycp \
-        mail-server_proposal.ycp \
-	mail-server_update.ycp
-
-myynclude_DATA =		\
-	helps.ycp		\
-	wj.ycp			\
-	widgets.ycp		\
-	ui.ycp			\
-	\
-	mail-server_helps.ycp   \
-        mail-server_wizards.ycp \
-        mail-server_dialogs.ycp \
-        mail-server_complex.ycp
-
-desktop_DATA =			\
-	mail.desktop
-
-module_DATA =			\
-	Mail.ycp		\
-	MailServer.ycp		\
-	MailServerLDAP.pm
-
 ldapschema_DATA = suse-mailserver.schema
 
-rncdir =  $(schemadir)/autoyast/rnc
-rnc_DATA = mail.rnc
-
-mail:
-	ln -sf . mail
-
-ycpchook = mail
-
-EXTRA_DIST = $(client_DATA) $(myynclude_DATA) $(module_DATA) $(desktop_DATA) $(rnc_DATA) \
-             $(ldapschema_DATA)
+EXTRA_DIST = $(ldapschema_DATA)
 
 include $(top_srcdir)/Makefile.am.common
diff --git a/src/clients/mail-advanced.ycp b/src/clients/mail-advanced.ycp
index a0d6353..5d818d7 100644
--- a/src/clients/mail-advanced.ycp
+++ b/src/clients/mail-advanced.ycp
@@ -57,11 +57,11 @@ $[
        ],
     ],
     "options"		: $[
-	// FIXME TODO: fill the option descriptions here
-    ],
+
+    ],	// FIXME TODO: fill the option descriptions here
     "mapping"		: $[
-	// FIXME TODO: fill the mappings of actions and options here
-    ]
+
+    ]	// FIXME TODO: fill the mappings of actions and options here
 ];
 
 /* analyze command line */
diff --git a/src/clients/mail-server.ycp b/src/clients/mail-server.ycp
index 988ba5d..b054251 100644
--- a/src/clients/mail-server.ycp
+++ b/src/clients/mail-server.ycp
@@ -38,14 +38,14 @@ map cmdline_description = $[
     "initialize"        : MailServer::Read,
     "finish"            : MailServer::Write,
     "actions"           : $[
-	// FIXME TODO: fill the functionality description here
+
     ],
     "options"		: $[
-	// FIXME TODO: fill the option descriptions here
-    ],
+
+    ],	// FIXME TODO: fill the option descriptions here
     "mapping"		: $[
-	// FIXME TODO: fill the mappings of actions and options here
-    ]
+
+    ]	// FIXME TODO: fill the mappings of actions and options here
 ];
 
 /*
diff --git a/src/include/mail/mail-server_dialogs.ycp b/src/include/mail/mail-server_dialogs.ycp
index 1aa3e3b..05d8c27 100644
--- a/src/include/mail/mail-server_dialogs.ycp
+++ b/src/include/mail/mail-server_dialogs.ycp
@@ -13,13 +13,17 @@ textdomain "mail";
 
 import "Label";
 import "Ldap";
-import "Wizard";
+import "MailServer";
+import "Popup";
+import "Progress";
 import "Report";
 import "Service";
 import "Users";
+import "Wizard";
 import "YaPI::MailServer";
 
 include "mail/helps.ycp";
+include "mail/mail-server_helps.ycp";
 
 boolean ReallyAbort() {
     return !MailServer::Modified() || Popup::ReallyAbort(true);
diff --git a/src/include/mail/widgets.ycp b/src/include/mail/widgets.ycp
index 346977f..ca3eaf3 100644
--- a/src/include/mail/widgets.ycp
+++ b/src/include/mail/widgets.ycp
@@ -650,7 +650,7 @@ define list Choices_delivery_mode () ``{
 	    `item (`id (`dm_procmail), _("Through procmail")),
 	    // combo box choice:
 	    // deliver mail to cyrus-imapd using LMTP
-	    `item (`id (`dm_cyrus), _("To Cyrus IMAP Server")),
+	    `item (`id (`dm_cyrus), _("To Cyrus IMAP Server"))
 	    // TODO: should check whether cyrus-imapd is installed.
 	    // And show the choice only if it is.
 	    // But config.postfix falls back to local if it's not, so OK
@@ -738,14 +738,14 @@ a POP or an IMAP server using <b>fetchmail</b>.</p>"),
 	"help": _("
 <p><b>Enabling virus scanning (AMaViS)</b> checks incoming and outgoing mail
 with AMaViS.</p>
-")
+") +
 	// help text
-+ _("<p>Enabling AMaViS will also enable the following modules: </p>
+_("<p>Enabling AMaViS will also enable the following modules: </p>
 <p><b>Spamassassin</b> SPAM scanner</p>
 <p><b>DKIM</b> checks Domain Key signed incomming mails</p>
-<p><b>Clamav</b> open source virus scanner engine</p>")
+<p><b>Clamav</b> open source virus scanner engine</p>") +
 	// help text
-+ _("
+_("
 <p>If AMaViS is not installed and you want to use it, it will be installed
 automatically.</p>
 "),
@@ -759,9 +759,9 @@ automatically.</p>
 	// help text
 	"help": _("
 <p><b>Enabling DKIM signig for outgoing mails.</b></p>
-")
+") +
 	// help text
-+ _("
+_("
 <p>Enabling DKIM for outgoing emails requires additional actions. A SSL key
 will be generated for the 'mydomain' value defined in Postfix. A new service
 'submission' will be configured in Postfix. After this is set up you can send
diff --git a/testsuite/tests/readwrite.ycp b/testsuite/tests/readwrite.ycp
index 81ff8e1..e814e03 100644
--- a/testsuite/tests/readwrite.ycp
+++ b/testsuite/tests/readwrite.ycp
@@ -165,7 +165,7 @@
     Progress::off ();
     TEST (``(Mail::Read (dont_abort)), [READ, WRITE, [
 					    EXECUTE_0, //rpm sendmail
-					    EXECUTE_0, //rpm amavisd-new
+					    EXECUTE_0  //rpm amavisd-new
 					    ]], nil);
     Mail::write_only = false;
     TEST (``(Mail::Write (dont_abort)), [READ, WRITE, [
@@ -173,7 +173,7 @@
 					     EXECUTE_0, //SuSEconfig
 					     EXECUTE_0, //amavis stop
 					     EXECUTE_0, //amavis start
-					     EXECUTE_0, //sendmail restart
+					     EXECUTE_0  //sendmail restart
 					     ]], nil);
 
 }
diff --git a/yast2-mail.spec.in b/yast2-mail.spec.in
index b85d006..a9fa5e1 100644
--- a/yast2-mail.spec.in
+++ b/yast2-mail.spec.in
@@ -28,6 +28,8 @@ Provides:       yast2-mail-server = %{version}
 Obsoletes:      yast2-mail-server <= %{version}
 Conflicts:	aaa_base < 10.3
 BuildArch:	noarch
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Mail Configuration
 
 %description
@@ -35,6 +37,8 @@ The YaST2 component for mail configuration. It handles Postfix, Cyrus,
 Amavis and Fetchmail.
 
 %package plugins
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Users/Group Plugins for the mail delivery configuration
 Group:		System/YaST
 Requires:       perl-NetxAP acl
@@ -44,7 +48,6 @@ Plugins for the YaST2 users module for enterprise mail server
 configuration.
 
 @PREP@
-cp agents/MasterCFParser.pm src
 
 @BUILD@
 
@@ -62,11 +65,9 @@ cp agents/MasterCFParser.pm src
 %dir @clientdir@
 @clientdir@/mail*
 %dir @moduledir@
-@moduledir@/Mail.ybc
-@moduledir@/MailServer.ybc
-@moduledir@/MailServer.ycp
+@moduledir@/MailServer.rb
 @moduledir@/MailServerLDAP.pm
-@moduledir@/Mail.ycp
+@moduledir@/Mail.rb
 %dir @moduledir@/YaPI
 @moduledir@/YaPI/Mail*
 %dir @desktopdir@
