diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 10187ee..2a6f678 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,4 +1,3 @@
 # Makefile.am for YCP module .../doc/autodocs
 
-AUTODOCS_YCP=$(srcdir)/../../src/*.ycp $(srcdir)/../../src/*/*.ycp
 include $(top_srcdir)/autodocs-ycp.ami
diff --git a/yast2-mouse.spec.in b/yast2-mouse.spec.in
index 441235f..161a314 100644
--- a/yast2-mouse.spec.in
+++ b/yast2-mouse.spec.in
@@ -19,6 +19,8 @@ Provides:       y2t_inst-mouse
 Obsoletes:      y2t_inst-mouse
 # Mouse-related installation scripts were moved here
 Conflicts:      yast2-installation < 2.18.2
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:        YaST2 - Mouse Configuration
 
 %description
@@ -41,7 +43,7 @@ rm -f $RPM_BUILD_ROOT@desktopdir@/mouse.desktop
 %dir @yncludedir@/mouse
 @yncludedir@/mouse/*
 @moduledir@/*
-@clientdir@/*.ycp
+@clientdir@/*.rb
 %ifnarch s390 s390x
 @desktopdir@/mouse.desktop
 %endif
