diff --git a/agents/.cvsignore b/agents/.cvsignore
deleted file mode 100644
index 282522d..0000000
--- a/agents/.cvsignore
+++ /dev/null
@@ -1,2 +0,0 @@
-Makefile
-Makefile.in
diff --git a/agents/Makefile.am b/agents/Makefile.am
deleted file mode 100644
index 5d589cd..0000000
--- a/agents/Makefile.am
+++ /dev/null
@@ -1,7 +0,0 @@
-# Makefile.am for multipath/agents
-
-agent_SCRIPTS = ag_multipath
-
-scrconf_DATA = etc_multipath.scr multipath.scr
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/doc/Makefile.am b/doc/Makefile.am
index 4ab8f24..eb43d2a 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -1,6 +1,7 @@
 # Makefile.am for multipath/doc
 
-SUBDIRS = autodocs
+# TODO missing support of ruby in auto-docs
+#SUBDIRS = autodocs
 
 htmldir = $(docdir)
 
diff --git a/src/include/multipath/complex.ycp b/src/include/multipath/complex.ycp
index de38fe1..7fbc5f6 100644
--- a/src/include/multipath/complex.ycp
+++ b/src/include/multipath/complex.ycp
@@ -30,10 +30,18 @@
  * by Multipath.ycp.
  */
 
+{
 textdomain "multipath";
 
 import "Service";
 import "Storage";
+import "Label";
+import "Wizard";
+import "Mode";
+import "Multipath";
+import "Stage";
+
+include "multipath/helps.ycp";
 
 integer service_status = 0;
 boolean has_dumbtab = false;
@@ -1018,7 +1026,7 @@ void Edit_Multipaths_Dialog(term table_item) {
 	} else if (ret == `ok) {
 	    if (check_mp_config(temp_cur_item) == true)
 	    {
-		config_modified = true;
+		Multipath::config_modified = true;
 		/* update the multipaths configuraton items */
 		if(table_item != nil) {
 		    multipaths_items = maplist( map item, multipaths_items, {
@@ -1054,7 +1062,7 @@ void Multipath_Dialog(symbol option)
 			boolean ret = false;
 			ret = (item["alias"]:"NA" != alias);
 			if(ret == false)
-			    config_modified = true;
+			    Multipath::config_modified = true;
 			return (ret);
 	});
 	UI::ChangeWidget(`id(`multipaths_table_id), `Items, Build_MultipathsTable());
@@ -1227,7 +1235,7 @@ void Edit_Defaults_Dialog(term table_item) {
 		string name_str = cur_item["name"]:"NA";
 		string value_str = cur_item["value"]:"";
 		defaults_items[name_str] = value_str;
-		config_modified = true;
+		Multipath::config_modified = true;
 		break;
 	    }
 	} else {
@@ -1507,7 +1515,7 @@ void delete_blacklist_item(term table_item) {
 			    match = false;
 			}
 			if(match == true)
-			    config_modified = true;
+			    Multipath::config_modified = true;
 			return (!match);
 		});
 }
@@ -1775,7 +1783,7 @@ void Edit_Blacklist_Dialog(term table_item) {
 		Popup::Message(sformat("Unexpected item type %1", type_str));
 	    }
 	    if(check_bl_config(new_item) == true) {
-		config_modified = true;
+		Multipath::config_modified = true;
 		blacklist_items = maplist(map item, blacklist_items, {
 			boolean match = true;
 			if(table_item[1]:"" == "device" && 
@@ -1876,7 +1884,7 @@ void Add_Blacklist_Dialog() {
 		id = get_newid(blacklist_items);
 		new_item["id"] = id;
 		if(check_bl_config(new_item) == true) {
-		    config_modified = true;
+		    Multipath::config_modified = true;
 		    blacklist_items = add(blacklist_items, new_item);
 		    break;
 		}
@@ -1957,7 +1965,7 @@ void delete_blacklist_exception_item(term table_item) {
 			    match = false;
 			}
 			if(match == true)
-			    config_modified = true;
+			    Multipath::config_modified = true;
 			return (!match);
 		});
 }
@@ -2225,7 +2233,7 @@ void Edit_Blacklist_Exception_Dialog(term table_item) {
 		Popup::Message(sformat("Unexpected item type %1", type_str));
 	    }
 	    if(check_ble_config(new_item) == true) {
-		config_modified = true;
+		Multipath::config_modified = true;
 		blacklist_exception_items = maplist(map item, blacklist_exception_items, {
 			boolean match = true;
 			if(table_item[1]:"" == "device" && 
@@ -2327,7 +2335,7 @@ void Add_Blacklist_Exception_Dialog() {
 		id = get_newid(blacklist_exception_items);
 		new_item["id"] = id;
 		if(check_ble_config(new_item) == true) {
-		    config_modified = true;
+		    Multipath::config_modified = true;
 		    blacklist_exception_items = add(blacklist_exception_items, new_item);
 		    break;
 		}
@@ -2588,7 +2596,7 @@ void Edit_Devices_Dialog(term table_item) {
 	    temp_cur_item["vendor"] = rm_quotes(temp_cur_item["vendor"]:"");
 	    temp_cur_item["product"] = rm_quotes(temp_cur_item["product"]:"");
 	    if(check_dv_config(temp_cur_item) == true) {
-		config_modified = true;
+		Multipath::config_modified = true;
 		cur_item = temp_cur_item;
 		if(table_item != nil) 
 		{
@@ -2633,7 +2641,7 @@ any Devices_Dialog() {
 			boolean match = (item["vendor"]:"NA" == cur_item[1]:"na" &&
                                         item["product"]:"NA" == cur_item[2]:"na");
                         if(match == true)
-                            config_modified = true;
+                            Multipath::config_modified = true;
                         return (!match);
                 });
 	    UI::ChangeWidget(`id(`devices_table_id), `Items, Build_DevicesTable());
@@ -2798,5 +2806,5 @@ void Stop_Service() {
     Update_Service_Status();
 }
 
-
+}
 
diff --git a/src/include/multipath/options.ycp b/src/include/multipath/options.ycp
index b9bce1d..e8a30cf 100644
--- a/src/include/multipath/options.ycp
+++ b/src/include/multipath/options.ycp
@@ -31,6 +31,10 @@
  * The default options can be used in defauts section, and other sections.
  */
 
+{
+
+import "Popup";
+
 textdomain "multipath";
 
 /* hold temporary customized string value of combobox */
@@ -1807,4 +1811,4 @@ list<symbol> blacklist_section_items = [
 	`product
     ];
 
-
+}
diff --git a/yast2-multipath.spec.in b/yast2-multipath.spec.in
index 07a0732..cf40490 100644
--- a/yast2-multipath.spec.in
+++ b/yast2-multipath.spec.in
@@ -29,11 +29,11 @@ You can configure your multipathed devices with this module.
 
 %files
 %defattr(-,root,root)
-%dir %{prefix}/share/YaST2/include/multipath
-%{prefix}/share/YaST2/include/multipath/*
-%{prefix}/share/YaST2/clients/multipath.ycp
-%{prefix}/share/YaST2/modules/Multipath.*
-%{prefix}/share/applications/YaST2/multipath.desktop
-%{prefix}/share/YaST2/scrconf/*.scr
-%{prefix}/lib/YaST2/servers_non_y2/*
-%doc %{prefix}/share/doc/packages/yast2-multipath
+%dir %{_prefix}/share/YaST2/include/multipath
+%{_prefix}/share/YaST2/include/multipath/*
+%{_prefix}/share/YaST2/clients/multipath.rb
+%{_prefix}/share/YaST2/modules/Multipath.*
+%{_prefix}/share/applications/YaST2/multipath.desktop
+%{_prefix}/share/YaST2/scrconf/*.scr
+%{_prefix}/lib/YaST2/servers_non_y2/*
+%doc %{_prefix}/share/doc/packages/yast2-multipath
