diff --git a/agents/Makefile.am b/agents/Makefile.am
index 945046b..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,9 +0,0 @@
-#
-# Makefile.am for y2c_network/agents
-#
-# $Id$
-#
-
-agent_SCRIPTS = ag_udev_persistent
-scrconf_DATA = $(wildcard *.scr)
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/data/Makefile.am b/data/Makefile.am
index 15ffd95..e69de29 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -1,4 +0,0 @@
-# Makefile.am for .../source/network/data
-
-ydata_DATA = modem-t35-keys.ycp phone-codes.ycp
-EXTRA_DIST = $(ydata_DATA)
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index a708e92..4b93a56 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,4 +1,6 @@
 # Makefile.am for YCP module .../doc/autodocs
 
-AUTODOCS_YCP = $(srcdir)/../../src/*/*.ycp
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled
+
+#AUTODOCS_YCP = $(srcdir)/../../src/*/*.ycp
+#include $(top_srcdir)/autodocs-ycp.ami
diff --git a/src/include/network/lan/complex.ycp b/src/include/network/lan/complex.ycp
index e11a8a4..0dc4672 100644
--- a/src/include/network/lan/complex.ycp
+++ b/src/include/network/lan/complex.ycp
@@ -355,7 +355,9 @@ If you edit the settings for this interface here,
 the interface will no longer be managed by NetworkManager.
 ")))
                 {
-                    break; // means cancel
+                    // y2r: cannot break from middle of switch
+                    // but in this function return will do
+                    return nil; // means cancel
                 }
 
                 // TODO move the defaults to GetDefaultsForHW
@@ -367,7 +369,9 @@ the interface will no longer be managed by NetworkManager.
             if (! AddInterface ()) 
             {
                 y2error( "handleOverview: AddInterface failed.");
-                break;
+                // y2r: cannot break from middle of switch
+                // but in this function return will do
+                return nil;
             }
 
             if( !DeviceReady( LanItems::getCurrentItem()[ "hwinfo", "dev_name"]:"") )
@@ -386,11 +390,20 @@ will be deleted as well.
 Really continue?
 "), LanItems::getCurrentItem()["ifcfg"]:"");
 	    if(LanItems::InterfaceHasAliases() && Popup::YesNoHeadline(Label::WarningMsg(), pop) != true)
-		break;
+            {
+                // y2r: cannot break from middle of switch
+                // but in this function return will do
+		return nil;
+            }
 
 	    // warn user when device to delete has STARTMODE=nfsroot (bnc#433867)
 	    if(NetworkInterfaces::GetValue(LanItems::getCurrentItem()["ifcfg"]:"", "STARTMODE")=="nfsroot"){
-		if(!Popup::YesNoHeadline(Label::WarningMsg(), _("Device you select has STARTMODE=nfsroot. Really delete?"))) break;
+		if(!Popup::YesNoHeadline(Label::WarningMsg(), _("Device you select has STARTMODE=nfsroot. Really delete?")))
+                {
+                    // y2r: cannot break from middle of switch
+                    // but in this function return will do
+                    return nil;
+                }
 	    }
 
 	    LanItems::DeleteItem();
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-network.spec.in b/yast2-network.spec.in
index de9eec7..65d4ac9 100644
--- a/yast2-network.spec.in
+++ b/yast2-network.spec.in
@@ -29,6 +29,8 @@ PreReq:         /bin/rm
 # carrier detection
 Conflicts:      yast2-core < 2.10.6
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:        YaST2 - Network Configuration
 
 %package devel-doc
@@ -71,10 +73,9 @@ fi
 @ybindir@/*
 @ydatadir@/*
 @yncludedir@/network
-@clientdir@/*.ycp
+@clientdir@/*.rb
 @moduledir@/YaPI/NETWORK.pm
-@moduledir@/*.ycp
-@moduledir@/*.ybc
+@moduledir@/*.rb
 @desktopdir@/*.desktop
 @scrconfdir@/*.scr
 @agentdir@/ag_udev_persistent
