diff --git a/doc/Makefile.am b/doc/Makefile.am
index b3ffc5a..1fb0e86 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -4,7 +4,8 @@
 # $Id$
 #
 
-SUBDIRS = autodocs
+# autodocs not yet support ruby
+#SUBDIRS = autodocs
 
 htmldir = $(docdir)
 
diff --git a/yast2-nfs-client.spec.in b/yast2-nfs-client.spec.in
index facbd30..4ecb378 100644
--- a/yast2-nfs-client.spec.in
+++ b/yast2-nfs-client.spec.in
@@ -26,6 +26,8 @@ Provides:       yast2-config-network:/usr/lib/YaST2/clients/lan_nfs_client.ycp
 
 BuildArch:      noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:        YaST2 - NFS Configuration
 Url:            http://en.opensuse.org/Portal:YaST
 
@@ -46,13 +48,12 @@ file system access. It allows access to files on remote machines.
 %dir @yncludedir@/nfs
 @yncludedir@/nfs/*
 %dir @clientdir@
-@clientdir@/nfs.ycp
-@clientdir@/nfs-client.ycp
-@clientdir@/nfs_auto.ycp
-@clientdir@/nfs-client4part.ycp
+@clientdir@/nfs.rb
+@clientdir@/nfs-client.rb
+@clientdir@/nfs_auto.rb
+@clientdir@/nfs-client4part.rb
 %dir @moduledir@
-@moduledir@/Nfs.ycp
-@moduledir@/Nfs.ybc
+@moduledir@/Nfs.rb
 %dir @desktopdir@
 @desktopdir@/nfs.desktop
 %doc @docdir@
