diff --git a/agents/Makefile.am b/agents/Makefile.am
index 71f2253..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,13 +0,0 @@
-#
-# Makefile.am for nfs-server/agents
-#
-# $Id$
-#
-
-agent_SCRIPTS = ag_exports
-
-scrconf_DATA = etc_exports.scr\
-		etc_idmapd_conf.scr\
-		cfg_nfs.scr
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index 02a627f..06bc411 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -4,10 +4,10 @@
 
 AUTOMAKE_OPTIONS = dejagnu
 
-ycp_testfiles = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+ycp_testfiles = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 epag_misc = lib/yast2_epag.exp run/runtest-epag.sh
-epag_testfiles = $(wildcard $(RPMNAME).test/ag_*.exp tests.epag/*.[rw].run tests.epag/*.[rw].data tests.epag/*.rw.ycp)
+epag_testfiles = $(wildcard $(RPMNAME).test/ag_*.exp tests.epag/*.[rw].run tests.epag/*.[rw].data tests.epag/*.rw.rb)
 EXTRA_DIST = $(ycp_testfiles) $(epag_testfiles) $(epag_misc)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
diff --git a/testsuite/tests/routines2.out b/testsuite/tests/routines2.out
index 3f8d61e..637ac16 100644
--- a/testsuite/tests/routines2.out
+++ b/testsuite/tests/routines2.out
@@ -6,6 +6,7 @@ Return	false
 Return	false
 Return	false
 Return	true
+Log	Invalid option.
 Return	false
 Return	true
 Return	true
diff --git a/yast2-nfs-server.spec.in b/yast2-nfs-server.spec.in
index d723fa4..c9169f1 100644
--- a/yast2-nfs-server.spec.in
+++ b/yast2-nfs-server.spec.in
@@ -14,6 +14,8 @@ Recommends:     nfs-kernel-server
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - NFS Server Configuration
 
 %description
@@ -40,11 +42,10 @@ Group:		System/YaST
 %defattr(-,root,root)
 %dir @yncludedir@/nfs_server
 @yncludedir@/nfs_server/*
-@clientdir@/nfs-server.ycp
-@clientdir@/nfs_server.ycp
-@clientdir@/nfs_server_auto.ycp
-@moduledir@/NfsServer.ycp
-@moduledir@/NfsServer.ybc
+@clientdir@/nfs-server.rb
+@clientdir@/nfs_server.rb
+@clientdir@/nfs_server_auto.rb
+@moduledir@/NfsServer.rb
 @desktopdir@/nfs_server.desktop
 @scrconfdir@/etc_exports.scr
 @agentdir@/ag_exports
