diff --git a/agents/Makefile.am b/agents/Makefile.am
index e503682..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,7 +0,0 @@
-#
-# Makefile.am for nis-server/agents
-#
-
-scrconf_DATA = $(wildcard $(srcdir)/*.scr)
-
-EXTRA_DIST = $(scrconf_DATA)
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-nis-server.spec.in b/yast2-nis-server.spec.in
index 67be058..1b150e2 100644
--- a/yast2-nis-server.spec.in
+++ b/yast2-nis-server.spec.in
@@ -18,6 +18,8 @@ Obsoletes:	yast2-trans-nis-server
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Network Information Services (NIS) Server Configuration
 
 %package devel-doc
@@ -44,14 +46,13 @@ This package contains documentation for yast2-nis-server
 %files
 %defattr(-,root,root)
 %dir @yncludedir@/nis_server
-@yncludedir@/nis_server/*.ycp
+@yncludedir@/nis_server/*.rb
 %dir @moduledir@
-@moduledir@/NisServer.ycp
-@moduledir@/NisServer.ybc
+@moduledir@/NisServer.rb
 %dir @clientdir@
-@clientdir@/nis_server.ycp
-@clientdir@/nis_server_auto.ycp
-@clientdir@/nis-server.ycp
+@clientdir@/nis_server.rb
+@clientdir@/nis_server_auto.rb
+@clientdir@/nis-server.rb
 %dir @desktopdir@
 @desktopdir@/nis_server.desktop
 %dir @scrconfdir@
