diff --git a/agents/Makefile.am b/agents/Makefile.am
index 3b65eee..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,9 +0,0 @@
-#
-# Makefile.am for devtools/devtools/skeletons/config/agents
-#
-
-scrconf_DATA =		\
-	etc_ntp.scr	\
-	cfg_ntp.scr
-
-EXTRA_DIST = $(scrconf_DATA)
diff --git a/data/Makefile.am b/data/Makefile.am
index 2178917..e69de29 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -1,9 +0,0 @@
-#
-# Makefile.am for ntp-client/data
-#
-
-ydata_DATA = \
-	ntp_servers.ycp
-
-
-EXTRA_DIST = $(ydata_DATA)
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 2a6f678..4cb582c 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,5 @@
 # Makefile.am for YCP module .../doc/autodocs
 
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled
+
+# include $(top_srcdir)/autodocs-ycp.ami
diff --git a/yast2-ntp-client.spec.in b/yast2-ntp-client.spec.in
index 4b69c82..f479a2a 100644
--- a/yast2-ntp-client.spec.in
+++ b/yast2-ntp-client.spec.in
@@ -17,6 +17,8 @@ Requires:       yast2-country-data
 
 BuildArch:      noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:        YaST2 - NTP Client Configuration
 
 %description
@@ -34,11 +36,11 @@ This package contains the YaST2 component for NTP client configuration.
 %defattr(-,root,root)
 %dir @yncludedir@/ntp-client
 @yncludedir@/ntp-client/*
-@clientdir@/ntp-client.ycp
-@clientdir@/ntp-client_*.ycp
+@clientdir@/ntp-client.rb
+@clientdir@/ntp-client_*.rb
 @scrconfdir@/cfg_ntp.scr
 @scrconfdir@/etc_ntp.scr
-@moduledir@/*.y*
+@moduledir@/*.rb
 @desktopdir@/ntp-client.desktop
 @ydatadir@/ntp_servers.ycp
 %doc @docdir@
