diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-online-update-configuration.spec.in b/yast2-online-update-configuration.spec.in
index 88dbcc1..a4fcc40 100644
--- a/yast2-online-update-configuration.spec.in
+++ b/yast2-online-update-configuration.spec.in
@@ -34,6 +34,8 @@ BuildRequires:  yast2-pkg-bindings >= 2.17.20
 BuildArch:      noarch
 
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:        Configuration of Online Update
 
 %description
@@ -55,9 +57,8 @@ Allows to configure automatic online update.
 %doc @docdir@
 %dir @yncludedir@/online-update-configuration
 @yncludedir@/online-update-configuration/*
-@clientdir@/*.ycp
-@moduledir@/*.ycp
-@moduledir@/*.ybc
+@clientdir@/*.rb
+@moduledir@/*.rb
 @desktopdir@/online_update_configuration.desktop
 @schemadir@/autoyast/rnc/*.rnc
 /usr/lib/YaST2/bin/online_update
