diff --git a/src/include/online-update-configuration/OUCDialogs.ycp b/src/include/online-update-configuration/OUCDialogs.ycp
index 36b4715..3897b5d 100644
--- a/src/include/online-update-configuration/OUCDialogs.ycp
+++ b/src/include/online-update-configuration/OUCDialogs.ycp
@@ -36,6 +36,7 @@ you may find current contact information at www.novell.com
 textdomain "online-update-configuration";
 
 import "OnlineUpdateConfiguration";
+import "Label";
 
 // module title
 string moduleTitle = _("Online Update Configuration");
diff --git a/yast2-online-update-configuration.spec.in b/yast2-online-update-configuration.spec.in
index 88dbcc1..4d6976b 100644
--- a/yast2-online-update-configuration.spec.in
+++ b/yast2-online-update-configuration.spec.in
@@ -34,6 +34,8 @@ BuildRequires:  yast2-pkg-bindings >= 2.17.20
 BuildArch:      noarch
 
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:        Configuration of Online Update
 
 %description
@@ -55,9 +57,8 @@ Allows to configure automatic online update.
 %doc @docdir@
 %dir @yncludedir@/online-update-configuration
 @yncludedir@/online-update-configuration/*
-@clientdir@/*.ycp
-@moduledir@/*.ycp
-@moduledir@/*.ybc
+@clientdir@/*.rb
+@moduledir@/*.rb
 @desktopdir@/online_update_configuration.desktop
 @schemadir@/autoyast/rnc/*.rnc
 /usr/lib/YaST2/bin/online_update
