diff --git a/src/clients/sw_single.ycp b/src/clients/sw_single.ycp
index 3caebc8..f6685c5 100644
--- a/src/clients/sw_single.ycp
+++ b/src/clients/sw_single.ycp
@@ -41,39 +41,13 @@
     import "String";
     import "URL";
 
-boolean force_summary = false;
-
-symbol StartSWSingle()
-{
-
-    Wizard::CreateDialog();
-    Wizard::SetDesktopTitleAndIcon("sw_single");
-
-    // a stage in the progress dialog
-    list<string> stages = [_("Initialize the Target System"), _("Load the Configured Repositories")];
-
-    // an extra step is needed in dir inst mode
-    if (Installation::destdir != "/")
-    {
-	y2internal("Extra step is needed");
-	// %1 is path to the target system (e.g. /tmp/dirinstall
-	sformat(_("Reset the target system to %1"), Installation::destdir);
-    }
-
-    // a stage in the progress dialog
-    Progress::New(_("Starting the Software Manager"), "", 2, stages, [], "");
-    Progress::NextStage();
-
-    import "Packages";
-
-    list<string> packagelist = [];		// list of packages to act on
+    boolean force_summary = false;
 
     // `install
     // `remove, `update: used from gnome-main-menu (#222757)
     symbol action = `install;
-
     boolean test_popup = false;
-    boolean skip_source = false;
+    list<string> packagelist = [];		// list of packages to act on
 
     // =============================================================
 
@@ -444,7 +418,30 @@ Only installed packages are displayed."));
    }
 
    // =============================================================
+symbol StartSWSingle()
+{
+
+    Wizard::CreateDialog();
+    Wizard::SetDesktopTitleAndIcon("sw_single");
 
+    // a stage in the progress dialog
+    list<string> stages = [_("Initialize the Target System"), _("Load the Configured Repositories")];
+
+    // an extra step is needed in dir inst mode
+    if (Installation::destdir != "/")
+    {
+	y2internal("Extra step is needed");
+	// %1 is path to the target system (e.g. /tmp/dirinstall
+	sformat(_("Reset the target system to %1"), Installation::destdir);
+    }
+
+    // a stage in the progress dialog
+    Progress::New(_("Starting the Software Manager"), "", 2, stages, [], "");
+    Progress::NextStage();
+
+    import "Packages";
+
+    boolean skip_source = false;
 
    // check whether running as root
    // and having the packager for ourselves
diff --git a/testsuite/manual-tests/test_convert.ycp b/testsuite/manual-tests/test_convert.ycp
index 94f1e3c..e2a9b6d 100644
--- a/testsuite/manual-tests/test_convert.ycp
+++ b/testsuite/manual-tests/test_convert.ycp
@@ -7,17 +7,8 @@
     while (p < 100)
     {
 	p = p + 5;
-    	PackageCallbacks::ProgressConvertDB (p, 1, 2, 3, 4);
+	PackageCallbacks::ProgressConvertDB(42, "/path/old");
 	sleep (100);
-
-	if (p == 20)
-	    PackageCallbacks::NotifyConvertDB ("Nindb", 50, "kdebase3");
-
-	if (p == 40)
-	    PackageCallbacks::NotifyConvertDB ("Eread", 50, "");
-
-	if (p == 60)
-	    PackageCallbacks::NotifyConvertDB ("Ewrite", 50, "");
     }
 
     PackageCallbacks::StopConvertDB (1, "kernel is on vacation");
diff --git a/testsuite/manual-tests/test_download.ycp b/testsuite/manual-tests/test_download.ycp
index ff37c59..3f614d0 100644
--- a/testsuite/manual-tests/test_download.ycp
+++ b/testsuite/manual-tests/test_download.ycp
@@ -8,7 +8,7 @@
     while (p < 100)
     {
 	p = p + 5;
-    	PackageCallbacks::ProgressDownload (p, 100000);
+	PackageCallbacks::ProgressDownload (p, 100000, 100000);
 	sleep (100);
     }
 
diff --git a/testsuite/manual-tests/test_rebuild.ycp b/testsuite/manual-tests/test_rebuild.ycp
index 51ccd4a..ab5bf71 100644
--- a/testsuite/manual-tests/test_rebuild.ycp
+++ b/testsuite/manual-tests/test_rebuild.ycp
@@ -7,11 +7,8 @@
     while (p < 100)
     {
 	p = p + 5;
-    	PackageCallbacks::ProgressRebuildDB (p);
+	PackageCallbacks::ProgressRebuildDB (p);
 	sleep (100);
-
-	if (p == 40)
-	    PackageCallbacks::NotifyRebuildDB ("In three weeks I'm going insane");
     }
 
     PackageCallbacks::StopRebuildDB (1, "kernel is on vacation");
diff --git a/testsuite/manual-tests/test_url.ycp b/testsuite/manual-tests/test_url.ycp
index d3bda08..c7a4e5d 100644
--- a/testsuite/manual-tests/test_url.ycp
+++ b/testsuite/manual-tests/test_url.ycp
@@ -1,5 +1,7 @@
 {
     import "InstURL";
+    import "URL";
+
     string url = InstURL::installInf2Url("");
     string log_url = URL::HidePassword(url);
     y2milestone("URL %1", url );
