diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 2c8851b..4c23223 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -2,4 +2,6 @@
 # Makefile.am for pam/doc/autodocs
 #
 
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled
+
+#include $(top_srcdir)/autodocs-ycp.ami
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-pam.spec.in b/yast2-pam.spec.in
index 10ada1a..4ab84d7 100644
--- a/yast2-pam.spec.in
+++ b/yast2-pam.spec.in
@@ -15,6 +15,8 @@ Obsoletes:	yast2-agent-pam-devel
 
 BuildArchitectures:     noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - PAM Agent
 
 %description
