diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index e109945..26050d7 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -2,6 +2,6 @@ TESTS_ENVIRONMENT = PERL5LIB=$(DESTDIR)$(PERL_VENDORARCH) prefix=$(prefix) libdi
 # test-perl writes to the log :(
 # but so does test-perl-min
 TESTS = test-perl-script test-yast# test-perl
-EXTRA_DIST = $(TESTS) test-perl-min tests/*.ycp tests/*.pm tests/A/*.pm tests/*.out tests/*.err tests.pl/*.pl tests.pl/*.ycp tests.pl/*.out tests.pl/*.err
+EXTRA_DIST = $(TESTS) test-perl-min tests/*.rb tests/*.pm tests/A/*.pm tests/*.out tests/*.err tests.pl/*.pl tests.pl/*.rb tests.pl/*.out tests.pl/*.err
 clean-local:
 	rm -f tmp.out.* tmp.err.*
diff --git a/testsuite/test-yast b/testsuite/test-yast
index 1324577..23e3467 100755
--- a/testsuite/test-yast
+++ b/testsuite/test-yast
@@ -28,8 +28,8 @@ ln -s $PLUGDIR $Y2DIR/plugin
 trap "rm -f $Y2DIR/modules $Y2DIR/plugin" EXIT TERM
 
 RET=0
-for SCRIPT in $TDIR/*.ycp; do
-    BASE=${SCRIPT%.ycp}
+for SCRIPT in $TDIR/*.rb; do
+    BASE=${SCRIPT%.rb}
     OUT=$BASE.out
     ERR=$BASE.err
     BASE=${BASE#$TDIR/}
