diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index e109945..26050d7 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -2,6 +2,6 @@ TESTS_ENVIRONMENT = PERL5LIB=$(DESTDIR)$(PERL_VENDORARCH) prefix=$(prefix) libdi
 # test-perl writes to the log :(
 # but so does test-perl-min
 TESTS = test-perl-script test-yast# test-perl
-EXTRA_DIST = $(TESTS) test-perl-min tests/*.ycp tests/*.pm tests/A/*.pm tests/*.out tests/*.err tests.pl/*.pl tests.pl/*.ycp tests.pl/*.out tests.pl/*.err
+EXTRA_DIST = $(TESTS) test-perl-min tests/*.rb tests/*.pm tests/A/*.pm tests/*.out tests/*.err tests.pl/*.pl tests.pl/*.rb tests.pl/*.out tests.pl/*.err
 clean-local:
 	rm -f tmp.out.* tmp.err.*
diff --git a/testsuite/test-yast b/testsuite/test-yast
index c478299..09696db 100755
--- a/testsuite/test-yast
+++ b/testsuite/test-yast
@@ -28,8 +28,8 @@ ln -s $PLUGDIR $Y2DIR/plugin
 trap "rm -f $Y2DIR/modules $Y2DIR/plugin" EXIT TERM
 
 RET=0
-for SCRIPT in $TDIR/*.ycp; do
-    BASE=${SCRIPT%.ycp}
+for SCRIPT in $TDIR/*.rb; do
+    BASE=${SCRIPT%.rb}
     OUT=$BASE.out
     ERR=$BASE.err
     BASE=${BASE#$TDIR/}
@@ -45,6 +45,8 @@ for SCRIPT in $TDIR/*.ycp; do
     sed --in-place -e 's/\(\[[^]]*\] [^: ]*\):[0-9]\+/\1:XXX/' tmp.err.$BASE
     # filter out uninteresting components
     sed --in-place -e '/^<[0-9]> \[\(liby2\|wfm\|ui-component\)\] /d' tmp.err.$BASE
+    # client call contains absolute path, ignore the line
+    sed --in-place '/^<1> \[Ruby\] yast\/wfm.rb:XXX Call client /d' tmp.err.$BASE
     diff -Nu $ERR tmp.err.$BASE || CASEOK=false
     diff -Nu $OUT tmp.out.$BASE || CASEOK=false
 
diff --git a/testsuite/tests/Long.err b/testsuite/tests/Long.err
index bb143d8..a758756 100644
--- a/testsuite/tests/Long.err
+++ b/testsuite/tests/Long.err
@@ -1,25 +1,25 @@
 <1> [Y2Perl] Y2PerlComponent.cc(Y2PerlComponent):XXX Creating Y2PerlComponent
-<1> [YCP] tests/Long.ycp:XXX 3*2**30: 3221225472
-<1> [YCP] tests/Long.ycp:XXX 3*2**30: 3221225472 (class)
-<1> [YCP] tests/Long.ycp:XXX 3*2**40: 3298534883328
-<1> [YCP] tests/Long.ycp:XXX 3*2**40: 3298534883328 (class)
-<1> [YCP] tests/Long.ycp:XXX -7 * 2**30: -7516192768
-<1> [YCP] tests/Long.ycp:XXX -7 * 2**30: -7516192768 class
-<1> [YCP] tests/Long.ycp:XXX -1 * 2**30: -1073741824
-<1> [YCP] tests/Long.ycp:XXX -1 * 2**30: -1073741824 class
-<1> [YCP] tests/Long.ycp:XXX 1 * 2**30: 1073741824
-<1> [YCP] tests/Long.ycp:XXX 1 * 2**30: 1073741824 class
-<1> [YCP] tests/Long.ycp:XXX 3 * 2**30: 3221225472
-<1> [YCP] tests/Long.ycp:XXX 3 * 2**30: 3221225472 class
-<1> [YCP] tests/Long.ycp:XXX 7 * 2**30: 7516192768
-<1> [YCP] tests/Long.ycp:XXX 7 * 2**30: 7516192768 class
-<1> [YCP] tests/Long.ycp:XXX loop -7 * 2**30: -7516192768
-<1> [YCP] tests/Long.ycp:XXX loop -7 * 2**30: -7516192768 class
-<1> [YCP] tests/Long.ycp:XXX loop -1 * 2**30: -1073741824
-<1> [YCP] tests/Long.ycp:XXX loop -1 * 2**30: -1073741824 class
-<1> [YCP] tests/Long.ycp:XXX loop 1 * 2**30: 1073741824
-<1> [YCP] tests/Long.ycp:XXX loop 1 * 2**30: 1073741824 class
-<1> [YCP] tests/Long.ycp:XXX loop 3 * 2**30: 3221225472
-<1> [YCP] tests/Long.ycp:XXX loop 3 * 2**30: 3221225472 class
-<1> [YCP] tests/Long.ycp:XXX loop 7 * 2**30: 7516192768
-<1> [YCP] tests/Long.ycp:XXX loop 7 * 2**30: 7516192768 class
+<1> [Ruby] tests/Long.rb:XXX 3*2**30: 3221225472
+<1> [Ruby] tests/Long.rb:XXX 3*2**30: 3221225472 (class)
+<1> [Ruby] tests/Long.rb:XXX 3*2**40: 3298534883328
+<1> [Ruby] tests/Long.rb:XXX 3*2**40: 3298534883328 (class)
+<1> [Ruby] tests/Long.rb:XXX -7 * 2**30: -7516192768
+<1> [Ruby] tests/Long.rb:XXX -7 * 2**30: -7516192768 class
+<1> [Ruby] tests/Long.rb:XXX -1 * 2**30: -1073741824
+<1> [Ruby] tests/Long.rb:XXX -1 * 2**30: -1073741824 class
+<1> [Ruby] tests/Long.rb:XXX 1 * 2**30: 1073741824
+<1> [Ruby] tests/Long.rb:XXX 1 * 2**30: 1073741824 class
+<1> [Ruby] tests/Long.rb:XXX 3 * 2**30: 3221225472
+<1> [Ruby] tests/Long.rb:XXX 3 * 2**30: 3221225472 class
+<1> [Ruby] tests/Long.rb:XXX 7 * 2**30: 7516192768
+<1> [Ruby] tests/Long.rb:XXX 7 * 2**30: 7516192768 class
+<1> [Ruby] tests/Long.rb:XXX loop -7 * 2**30: -7516192768
+<1> [Ruby] tests/Long.rb:XXX loop -7 * 2**30: -7516192768 class
+<1> [Ruby] tests/Long.rb:XXX loop -1 * 2**30: -1073741824
+<1> [Ruby] tests/Long.rb:XXX loop -1 * 2**30: -1073741824 class
+<1> [Ruby] tests/Long.rb:XXX loop 1 * 2**30: 1073741824
+<1> [Ruby] tests/Long.rb:XXX loop 1 * 2**30: 1073741824 class
+<1> [Ruby] tests/Long.rb:XXX loop 3 * 2**30: 3221225472
+<1> [Ruby] tests/Long.rb:XXX loop 3 * 2**30: 3221225472 class
+<1> [Ruby] tests/Long.rb:XXX loop 7 * 2**30: 7516192768
+<1> [Ruby] tests/Long.rb:XXX loop 7 * 2**30: 7516192768 class
diff --git a/testsuite/tests/Simple1.err b/testsuite/tests/Simple1.err
index fb04f65..334cdcc 100644
--- a/testsuite/tests/Simple1.err
+++ b/testsuite/tests/Simple1.err
@@ -1,2 +1,2 @@
 <1> [Y2Perl] Y2PerlComponent.cc(Y2PerlComponent):XXX Creating Y2PerlComponent
-<1> [YCP] tests/Simple1.ycp:XXX Hello, world
+<1> [Ruby] tests/Simple1.rb:XXX Hello, world
diff --git a/testsuite/tests/Testpfunc1.err b/testsuite/tests/Testpfunc1.err
index a06b58a..100b0fe 100644
--- a/testsuite/tests/Testpfunc1.err
+++ b/testsuite/tests/Testpfunc1.err
@@ -1,5 +1,5 @@
 <1> [Y2Perl] Y2PerlComponent.cc(Y2PerlComponent):XXX Creating Y2PerlComponent
-<1> [YCP] tests/Testpfunc1.ycp:XXX rxmatch (abracadabra, [a-d]*): true
-<1> [YCP] tests/Testpfunc1.ycp:XXX rxmatch (abracadabra, ^[a-d]*$): false
-<1> [YCP] tests/Testpfunc1.ycp:XXX lengths (["one", "two", "three"]): [3, 3, 5]
-<1> [YCP] tests/Testpfunc1.ycp:XXX amap ($["one":"two"]): $["a":"1", "b":"two"]
+<1> [Ruby] tests/Testpfunc1.rb:XXX rxmatch (abracadabra, [a-d]*): true
+<1> [Ruby] tests/Testpfunc1.rb:XXX rxmatch (abracadabra, ^[a-d]*$): false
+<1> [Ruby] tests/Testpfunc1.rb:XXX lengths (["one", "two", "three"]): [3, 3, 5]
+<1> [Ruby] tests/Testpfunc1.rb:XXX amap ($["one":"two"]): $["a":"1", "b":"two"]
diff --git a/testsuite/tests/Testpfunc2.err b/testsuite/tests/Testpfunc2.err
index 4260f21..ebd2e38 100644
--- a/testsuite/tests/Testpfunc2.err
+++ b/testsuite/tests/Testpfunc2.err
@@ -1,3 +1,3 @@
 <1> [Y2Perl] Y2PerlComponent.cc(Y2PerlComponent):XXX Creating Y2PerlComponent
-<1> [YCP] tests/Testpfunc2.ycp:XXX nested: Hello, world
+<1> [Ruby] tests/Testpfunc2.rb:XXX nested: Hello, world
 <1> [Y2Perl] YPerl.cc(destroy):XXX Shutting down embedded Perl interpreter.
diff --git a/testsuite/tests/Types.err b/testsuite/tests/Types.err
index 672aa00..d8cc5b4 100644
--- a/testsuite/tests/Types.err
+++ b/testsuite/tests/Types.err
@@ -1,3 +1,3 @@
 <1> [Y2Perl] Y2PerlComponent.cc(Y2PerlComponent):XXX Creating Y2PerlComponent
-<1> [YCP] tests/Types.ycp:XXX bool1: false
-<1> [YCP] tests/Types.ycp:XXX bool2: false
+<1> [Ruby] tests/Types.rb:XXX bool1: false
+<1> [Ruby] tests/Types.rb:XXX bool2: false
diff --git a/testsuite/tests/Types2.err b/testsuite/tests/Types2.err
index 24f6780..61b139d 100644
--- a/testsuite/tests/Types2.err
+++ b/testsuite/tests/Types2.err
@@ -1,4 +1,4 @@
 <1> [Y2Perl] Y2PerlComponent.cc(Y2PerlComponent):XXX Creating Y2PerlComponent
-<1> [YCP] tests/Types2.ycp:XXX termloop: `MyTerm ("Hi", "42")
-<1> [YCP] tests/Types2.ycp:XXX termloop nt: `NestedTerm (`id ("42"), `MyTerm ("Hi", "42"))
-<1> [YCP] tests/Types2.ycp:XXX termreverse: `mreTyM ("42", "Hi")
+<1> [Ruby] tests/Types2.rb:XXX termloop: `MyTerm ("Hi", "42")
+<1> [Ruby] tests/Types2.rb:XXX termloop nt: `NestedTerm (`id ("42"), `MyTerm ("Hi", "42"))
+<1> [Ruby] tests/Types2.rb:XXX termreverse: `mreTyM ("42", "Hi")
diff --git a/yast2-perl-bindings.spec.in b/yast2-perl-bindings.spec.in
index bb09b7b..e747aae 100644
--- a/yast2-perl-bindings.spec.in
+++ b/yast2-perl-bindings.spec.in
@@ -4,6 +4,7 @@
 Group:	System/YaST
 License: GPL-2.0+
 BuildRequires:	gcc-c++ yast2-core-devel yast2-devtools yast2-ycp-ui-bindings-devel libtool
+BuildRequires:  yast2-ruby-bindings >= 1.0.0
 
 # YCPValue::valuetype_str()
 Requires:	yast2-core >= 2.16.37
