diff --git a/agents/Makefile.am b/agents/Makefile.am
index 7522808..80fce89 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -2,10 +2,3 @@
 # Makefile.am for devtools/devtools/skeletons/config/agents
 #
 
-agent_SCRIPTS =
-
-scrconf_DATA =				\
-	etc_pm-profiler.scr		\
-	etc_pm-profiler_profiles.scr
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 417f306..637c3e6 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,5 +1,4 @@
 # Makefile.am for devtools/devtools/skeletons/config/doc/autodocs
 
-AUTODOCS_YCP = $(top_srcdir)/src/*.ycp
 include $(top_srcdir)/autodocs-ycp.ami
 
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-power-management.spec.in b/yast2-power-management.spec.in
index 145f877..48eb387 100644
--- a/yast2-power-management.spec.in
+++ b/yast2-power-management.spec.in
@@ -10,6 +10,8 @@ BuildArchitectures: noarch
 # .etc.policykit agent
 Requires:	yast2 >= 2.14.7
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Power Management Configuration
 
 %description
@@ -28,8 +30,8 @@ configuration.
 %defattr(-,root,root)
 %dir @yncludedir@/power-management
 @yncludedir@/power-management/*
-@clientdir@/power-management.ycp
-@clientdir@/power-management_*.ycp
+@clientdir@/power-management.rb
+@clientdir@/power-management_*.rb
 @moduledir@/PowerManagement.*
 @desktopdir@/power-management.desktop
 @scrconfdir@/*
