diff --git a/doc/Makefile.am b/doc/Makefile.am
index 7ee8d36..091061f 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -1,10 +1,10 @@
 # Makefile.am for printer/doc
 
-SUBDIRS = autodocs
+# SUBDIRS = autodocs
 
-htmldir = $(docdir)
+# htmldir = $(docdir)
 
-doc_DATA =
-html_DATA =
+# doc_DATA =
+# html_DATA =
 
-EXTRA_DIST = $(doc_DATA) $(html_DATA)
+# EXTRA_DIST = $(doc_DATA) $(html_DATA)
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index f7e5b2d..23dcf72 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,4 @@
 # Makefile.am for printer/doc/autodocs
-
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled, enable back when 
+#             autodocs support is implemented
+# include $(top_srcdir)/autodocs-ycp.ami
diff --git a/src/include/printer/connectionwizard.ycp b/src/include/printer/connectionwizard.ycp
index 253eff4..dc1bfd2 100644
--- a/src/include/printer/connectionwizard.ycp
+++ b/src/include/printer/connectionwizard.ycp
@@ -590,9 +590,9 @@ void changeSettingsDialog( symbol selected )
                           ( _("The RPM package hplip is not installed.") )
                         )
                       );
-            break;
           }
-          content = getContentFromBackend("hp");
+          else
+          { content = getContentFromBackend("hp"); }
           break;
     case(`serial):
           current_device_uri = getCurrentDeviceURI();
@@ -986,9 +986,10 @@ void changeSettingsDialog( symbol selected )
                             ( _("The RPM package bluez-cups is not installed.") )
                           )
                         );
-              break;
             }
           }
+          else
+          {
           // Fallback message what the user may run manually when it fails
           // to generate a valid list of bluetooth device IDs:
           string  bluetooth_device_list = _("It seems there are no bluetooth device IDs.\nRun 'hcitool scan' to get the bluetooth device IDs.\nEnter the ID without colons like '1A2B3C4D5E6F'.");
@@ -1098,6 +1099,7 @@ void changeSettingsDialog( symbol selected )
                       ),
                       model_content
                     );
+          }
           break;
     // Access network printer or printserver box via
     case(`tcp):
@@ -1264,9 +1266,10 @@ void changeSettingsDialog( symbol selected )
                             ( _("The RPM package samba-client is not installed.") )
                           )
                         );
-              break;
             }
           }
+          else
+          {
           hostname = "";
           domain = "";
           printer = "";
@@ -1433,6 +1436,7 @@ void changeSettingsDialog( symbol selected )
                       ),
                       model_content
                     );
+          }
           break;
     case(`lpr):
           hostname = "";
@@ -1616,9 +1620,10 @@ void changeSettingsDialog( symbol selected )
                             ( _("The RPM package ncpfs is not installed.") )
                           )
                         );
-              break;
             }
           }
+          else
+          {
           hostname = "";
           queue = "";
           user = "";
@@ -1702,6 +1707,7 @@ void changeSettingsDialog( symbol selected )
                       ),
                       model_content
                     );
+          }
           break;
     // Special
     case(`uri):
@@ -1783,9 +1789,10 @@ void changeSettingsDialog( symbol selected )
                             ( _("The RPM package cups-backends is not installed.") )
                           )
                         );
-              break;
             }
           }
+          else
+          {
           uri = "";
           current_device_uri = getCurrentDeviceURI();
           if( "pipe:/" == substring( current_device_uri, 0, size( "pipe:/" ) ) )
@@ -1809,6 +1816,7 @@ void changeSettingsDialog( symbol selected )
                       ),
                       model_content
                     );
+          }
           break;
     case(`beh):
           uri = "";
@@ -2166,22 +2174,22 @@ boolean validateAndMakeURI( symbol selected )
                                // because there is no connection available to be selected:
                                _("If no connection is shown here, it is not possible\nto access the device via this type of connection.\nWas the printer connected and switched on all the time?")
                              );
-            break;
-          }
-          if( selected_connection_index < 0 )
+          } else if ( selected_connection_index < 0 )
           { Popup::AnyMessage( // Header of a Popup::AnyMessage when an invalid connection was selected:
                                _("Select a valid connection"),
                                // Body of a Popup::AnyMessage when an invalid connection was selected
                                // because the current connection is no longer valid:
                                _("When the current connection is no longer valid,\nit does no longer work to access the device via this connection.\nIs the printer still connected and switched on?")
                              );
-            break;
           }
+          else
+          {
           connection_uri = Printer::connections[selected_connection_index,"uri"]:"";
           connection_model = Printer::connections[selected_connection_index,"model"]:"Unknown";
           if( "" != connection_uri )
           { valid = true;
           }
+          }
           break;
     case(`serial):
           string serial_device_node = (string)UI::QueryWidget( `serial_device_node, `Value );
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-printer.spec.in b/yast2-printer.spec.in
index 27b37f2..30f908c 100644
--- a/yast2-printer.spec.in
+++ b/yast2-printer.spec.in
@@ -18,6 +18,8 @@ Requires:       yast2 >= 2.16.12
 # which are pulled in by Autoreqprov because of the basicadd_displaytest tool:
 %define my_requires /tmp/my-requires
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:        YaST2 - Printer Configuration
 License:        GPL-2.0
 Group:          Documentation/SUSE
@@ -55,7 +57,7 @@ chmod 755 %{my_requires}
 %defattr(-,root,root)
 %dir @yncludedir@/printer
 @desktopdir@/printer.desktop
-@moduledir@/*.y*
+@moduledir@/*.rb
 @clientdir@/printer*
 @yncludedir@/printer/*
 @schemadir@/autoyast/rnc/printer.rnc
