diff --git a/src/clients/proxy.ycp b/src/clients/proxy.ycp
index 5b3470c..d5925e8 100644
--- a/src/clients/proxy.ycp
+++ b/src/clients/proxy.ycp
@@ -23,6 +23,8 @@ import "Label";
 import "Proxy";
 import "Wizard";
 
+include "proxy/dialogs.ycp";
+
 /**
  * Return a modification status
  * @return true if data was modified
@@ -32,8 +34,6 @@ define boolean Modified() {
 }
 
 any ProxyGUI() {
-    include "proxy/dialogs.ycp";
-
     Proxy::Read ();
 
     Wizard::CreateDialog();
diff --git a/yast2-proxy.spec.in b/yast2-proxy.spec.in
index 09b83c7..53d134d 100644
--- a/yast2-proxy.spec.in
+++ b/yast2-proxy.spec.in
@@ -33,9 +33,8 @@ This package contains the YaST2 component for proxy configuration.
 
 %files
 %defattr(-,root,root)
-@clientdir@/*.ycp
-@moduledir@/*.ycp
-@moduledir@/*.ybc
+@clientdir@/*.rb
+@moduledir@/*.rb
 @yncludedir@/proxy
 @desktopdir@/*.desktop
 @scrconfdir@/*.scr
