diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-proxy.spec.in b/yast2-proxy.spec.in
index 09b83c7..b751e68 100644
--- a/yast2-proxy.spec.in
+++ b/yast2-proxy.spec.in
@@ -17,6 +17,8 @@ Conflicts:      yast2-network < 2.22.6
 PreReq:         /bin/rm
 BuildArch:      noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:        YaST2 - Proxy Configuration
 Url:            http://en.opensuse.org/Portal:YaST
 
@@ -33,9 +35,8 @@ This package contains the YaST2 component for proxy configuration.
 
 %files
 %defattr(-,root,root)
-@clientdir@/*.ycp
-@moduledir@/*.ycp
-@moduledir@/*.ybc
+@clientdir@/*.rb
+@moduledir@/*.rb
 @yncludedir@/proxy
 @desktopdir@/*.desktop
 @scrconfdir@/*.scr
