diff --git a/src/clients/proxy.ycp b/src/clients/proxy.ycp
index 5b3470c..d5925e8 100644
--- a/src/clients/proxy.ycp
+++ b/src/clients/proxy.ycp
@@ -23,6 +23,8 @@ import "Label";
 import "Proxy";
 import "Wizard";
 
+include "proxy/dialogs.ycp";
+
 /**
  * Return a modification status
  * @return true if data was modified
@@ -32,8 +34,6 @@ define boolean Modified() {
 }
 
 any ProxyGUI() {
-    include "proxy/dialogs.ycp";
-
     Proxy::Read ();
 
     Wizard::CreateDialog();
diff --git a/yast2-proxy.spec.in b/yast2-proxy.spec.in
index 09b83c7..c2c852e 100644
--- a/yast2-proxy.spec.in
+++ b/yast2-proxy.spec.in
@@ -17,6 +17,8 @@ Conflicts:      yast2-network < 2.22.6
 PreReq:         /bin/rm
 BuildArch:      noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:        YaST2 - Proxy Configuration
 Url:            http://en.opensuse.org/Portal:YaST
 
@@ -33,9 +35,8 @@ This package contains the YaST2 component for proxy configuration.
 
 %files
 %defattr(-,root,root)
-@clientdir@/*.ycp
-@moduledir@/*.ycp
-@moduledir@/*.ybc
+@clientdir@/*.rb
+@moduledir@/*.rb
 @yncludedir@/proxy
 @desktopdir@/*.desktop
 @scrconfdir@/*.scr
