diff --git a/SUBDIRS b/SUBDIRS
index 8d9a14f..3508cf2 100644
--- a/SUBDIRS
+++ b/SUBDIRS
@@ -1 +1 @@
-agents doc src
+agents doc src yastbrowser
diff --git a/agents/Makefile.am b/agents/Makefile.am
index 0fcd3e0..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,7 +0,0 @@
-# Makefile.am for registration/agents
-
-agent_SCRIPTS =
-
-scrconf_DATA = $(wildcard *.scr)
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/yast2-registration.spec.in b/yast2-registration.spec.in
index c4bce6a..bb37fa1 100644
--- a/yast2-registration.spec.in
+++ b/yast2-registration.spec.in
@@ -26,6 +26,7 @@ Requires:       suseRegister
 Requires:       perl-camgm perl-TimeDate
 Requires:       yast2-pkg-bindings >= 2.17.20
 Requires:       yast2-registration-branding
+Requires:       yast2-ruby-bindings >= 1.0.0
 Requires(post): sed grep
 ## do never require "mozilla-xulrunner190" (bnc#436900)
 PreReq:         %fillup_prereq
@@ -110,12 +111,11 @@ fi
 %files
 %defattr(-,root,root)
 %doc @docdir@
-@clientdir@/*.ycp
+@clientdir@/*.rb
 @moduledir@/*.pm
-@moduledir@/*.ycp
-@moduledir@/*.ybc
+@moduledir@/*.rb
 %dir @yncludedir@/registration
-@yncludedir@/registration/*.ycp
+@yncludedir@/registration/*.rb
 @schemadir@/autoyast/rnc/*.rnc
 /usr/share/YaST2/yastbrowser
 # agents
