diff --git a/agents/Makefile.am b/agents/Makefile.am
index c1898cd..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,7 +0,0 @@
-# Makefile.am for reipl/agents
-
-agent_SCRIPTS =
-
-scrconf_DATA =
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 26d35b7..4709d9d 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,5 @@
 # Makefile.am for reipl/doc/autodocs
 
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporary disabled
+
+#include $(top_srcdir)/autodocs-ycp.ami
diff --git a/src/clients/reipl.ycp b/src/clients/reipl.ycp
index be5d9d8..c72d261 100644
--- a/src/clients/reipl.ycp
+++ b/src/clients/reipl.ycp
@@ -56,14 +56,14 @@ map cmdline_description = $[
     "initialize"        : Reipl::Read,
     "finish"            : Reipl::Write,
     "actions"           : $[
-	// FIXME TODO: fill the functionality description here
-    ],
+
+    ],	// FIXME TODO: fill the functionality description here
     "options"		: $[
-	// FIXME TODO: fill the option descriptions here
-    ],
+
+    ],	// FIXME TODO: fill the option descriptions here
     "mappings"		: $[
-	// FIXME TODO: fill the mappings of actions and options here
-    ]
+
+    ]	// FIXME TODO: fill the mappings of actions and options here
 ];
 
 /* is this proposal or not? */
diff --git a/src/include/reipl/dialogs.ycp b/src/include/reipl/dialogs.ycp
index 83bbbd9..464185d 100644
--- a/src/include/reipl/dialogs.ycp
+++ b/src/include/reipl/dialogs.ycp
@@ -36,6 +36,7 @@ import "Wizard";
 import "Reipl";
 
 include "reipl/helps.ycp";
+include "reipl/complex.ycp";
 
 /**
  * Configure dialog
diff --git a/yast2-reipl.spec.in b/yast2-reipl.spec.in
index 490276d..9f6c67a 100644
--- a/yast2-reipl.spec.in
+++ b/yast2-reipl.spec.in
@@ -21,14 +21,12 @@ PreReq:         %fillup_prereq
 
 BuildArch:      noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:        YaST2 - IPL loader
 License:        GPL-2.0
 Group:          System/YaST
 
-PreReq:         %fillup_prereq
-
-Summary:        YaST2 - IPL loader
-Group:          System/YaST
 
 %description
 Module for loading IPL from running system on S/390
@@ -48,9 +46,8 @@ Module for loading IPL from running system on S/390
 
 %files
 %defattr(-,root,root)
-@clientdir@/*.ycp
-@moduledir@/*.ycp
-@moduledir@/*.ybc
+@clientdir@/*.rb
+@moduledir@/*.rb
 @desktopdir@/*.desktop
 %dir @yncludedir@/reipl
 @yncludedir@/reipl/*
