diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-runlevel.spec.in b/yast2-runlevel.spec.in
index d8fdbd1..fe38048 100644
--- a/yast2-runlevel.spec.in
+++ b/yast2-runlevel.spec.in
@@ -16,6 +16,8 @@ Obsoletes:	yast2-config-runlevel
 Provides:	yast2-trans-runlevel
 Obsoletes:	yast2-trans-runlevel
 BuildArchitectures:     noarch
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Runlevel Editor
 
 %description
@@ -37,11 +39,10 @@ boot.
 %dir @scrconfdir@
 @scrconfdir@/*.scr
 %dir @clientdir@
-@clientdir@/*.ycp
+@clientdir@/*.rb
 %dir @yncludedir@/runlevel
-@yncludedir@/runlevel/*.ycp
+@yncludedir@/runlevel/*.rb
 %dir @moduledir@
-@moduledir@/*.ycp
-@moduledir@/*.ybc
+@moduledir@/*.rb
 @schemadir@/autoyast/rnc/*.rnc
 %doc @docdir@
