diff --git a/src/include/runlevel/ui.ycp b/src/include/runlevel/ui.ycp
index 0c65120..c214b64 100644
--- a/src/include/runlevel/ui.ycp
+++ b/src/include/runlevel/ui.ycp
@@ -1574,6 +1574,15 @@ name or the name specified is already in use."));
     }
 
     /**
+     * Sets columns 0-S (runlevels) in table so they are synchronized with checkboxes.
+     */
+    define void refreshTableLine2 (map service) ``{
+        updateRlColumns (current_service, service, nil);
+        UI::ChangeWidget (`id (`table), `Item (current_service, c_dirty),
+                          service["dirty"]:false ? UI::Glyph(`CheckMark) : " ");
+    }
+
+    /**
      * Main dialog for changing services.
      * @return symbol for wizard sequencer
      */
@@ -1582,15 +1591,6 @@ name or the name specified is already in use."));
 	// currently selected service we are working with
 	map service = $[];
 
-	/**
-	 * Sets columns 0-S (runlevels) in table so they are synchronized with checkboxes.
-	 */
-	define void refreshTableLine2 () ``{
-	    updateRlColumns (current_service, service, nil);
-	    UI::ChangeWidget (`id (`table), `Item (current_service, c_dirty),
-			      service["dirty"]:false ? UI::Glyph(`CheckMark) : " ");
-	}
-
 	// headers in table
 	term header = `header (
 	    // table header
@@ -1706,7 +1706,7 @@ disabled?\n"), nfs_adj)))
 		{
 		    setServiceToDefault (current_service);
 		    service = RunlevelEd::services[current_service]:$[];
-		    refreshTableLine2 ();
+		    refreshTableLine2(service);
 		    ret = `table;
 		}
 		else if (`clear == ret)
@@ -1714,7 +1714,7 @@ disabled?\n"), nfs_adj)))
 		    // re-read from SCR
 		    service = Service::Info (current_service);
 		    RunlevelEd::services[current_service] = service;
-		    refreshTableLine2 ();
+		    refreshTableLine2(service);
 		    ret = `table;
 		}
 		else if (`clear_all == ret)
