diff --git a/agent-pam_mount/Makefile.am b/agent-pam_mount/Makefile.am
index 1d6b63f..e69de29 100644
--- a/agent-pam_mount/Makefile.am
+++ b/agent-pam_mount/Makefile.am
@@ -1,10 +0,0 @@
-#
-# Makefile.am for agent-pam_mount
-#
-
-agentdir = @execcompdir@/servers_non_y2
-
-agent_SCRIPTS = ag_*
-scrconf_DATA = *.scr
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/agents/Makefile.am b/agents/Makefile.am
index 4b88cc4..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,7 +0,0 @@
-#
-# Makefile.am for devtools/devtools/skeletons/config/agents
-#
-
-scrconf_DATA = cfg_smbconf.scr etc_security_winbind.scr
-
-EXTRA_DIST = $(scrconf_DATA)
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d63b19b..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-samba-client.spec.in b/yast2-samba-client.spec.in
index 0dc5bb4..a212d16 100644
--- a/yast2-samba-client.spec.in
+++ b/yast2-samba-client.spec.in
@@ -15,6 +15,8 @@ Requires:       yast2 >= 2.21.22
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Samba Client Configuration
 
 %description
@@ -33,11 +35,10 @@ workgroup/domain and authentication against an SMB domain.
 %defattr(-,root,root)
 %dir @yncludedir@/samba-client
 @yncludedir@/samba-client/*
-@clientdir@/samba-client.ycp
-@clientdir@/samba-client_*.ycp
+@clientdir@/samba-client.rb
+@clientdir@/samba-client_*.rb
 @moduledir@/Samba*.pm
-@moduledir@/Samba.ycp
-@moduledir@/Samba.ybc
+@moduledir@/Samba.rb
 @desktopdir@/samba-client.desktop
 @scrconfdir@/*.scr
 @agentdir@/ag_pam_mount
