diff --git a/src/include/samba-client/dialogs.ycp b/src/include/samba-client/dialogs.ycp
index 572531d..8255a98 100644
--- a/src/include/samba-client/dialogs.ycp
+++ b/src/include/samba-client/dialogs.ycp
@@ -134,6 +134,17 @@ map MountEntryPopup (map volume) {
     return volume;
 }
 
+list<term> build_mount_items (list<term> & mount_items, map<integer,map> mount_map) {
+
+    mount_items	= maplist (integer id, map volume, mount_map, {
+	return `item (`id (id),
+	    volume["server"]:"", volume["path"]:"",
+	    volume["mountpoint"]:"", volume["user"]:"",
+	    volume["options"]:"");
+    });
+    return mount_items;
+}
+
 /**
  * dialog for setting expert settings, like winbind uid/gid keys (F301518)
  */
@@ -183,16 +194,6 @@ symbol ExpertSettingsDialog (boolean use_winbind) {
 	i		= i + 1;
     });
 
-    list<term> build_mount_items () {
-
-	mount_items	= maplist (integer id, map volume, mount_map, {
-	    return `item (`id (id),
-		volume["server"]:"", volume["path"]:"",
-		volume["mountpoint"]:"", volume["user"]:"",
-		volume["options"]:"");
-	});
-	return mount_items;
-    }
     list<term> kerberos_methods	= maplist (string method,
 	[ "secrets only", "system keytab", "dedicated keytab", "secrets and keytab"], {
 	return `item (`id (method), method, method == kerberos_method);
@@ -238,7 +239,7 @@ symbol ExpertSettingsDialog (boolean use_winbind) {
 		_("User Name"),
 		// table header
 		_("Options")),
-		build_mount_items ()
+		build_mount_items (mount_items, mount_map)
 	    ),
 	    `HBox (
 		`PushButton (`id (`add), Label::AddButton ()),
@@ -273,7 +274,7 @@ symbol ExpertSettingsDialog (boolean use_winbind) {
 	if (ret == `delete)
 	{
 	    mount_map	= remove (mount_map, selected);
-	    UI::ChangeWidget (`id (`table), `Items, build_mount_items ());
+	    UI::ChangeWidget (`id (`table), `Items, build_mount_items (mount_items, mount_map));
 	    foreach (symbol s, [`edit, `delete], {
 		UI::ChangeWidget (`id (s), `Enabled, size (mount_items) > 0);
 	    });
@@ -288,7 +289,7 @@ symbol ExpertSettingsDialog (boolean use_winbind) {
 		volume["fstype"]	= "cifs";
 	    integer id		= ret == `edit ? selected : size (mount_map);
 	    mount_map[id]	= volume;
-	    UI::ChangeWidget (`id (`table), `Items, build_mount_items ());
+	    UI::ChangeWidget (`id (`table), `Items, build_mount_items (mount_items, mount_map));
 	    foreach (symbol s, [ `edit, `delete], {
 		UI::ChangeWidget (`id (s), `Enabled, size (mount_items) > 0);
 	    });
@@ -331,6 +332,40 @@ larger than maximum one.\n"));
     return (symbol) ret;
 }
 
+// internal function: update the status line
+void check_domain_membership (string domain) {
+
+    Samba::SetWorkgroup(domain);
+
+    if (Mode::config ())
+	return;
+
+    // busy popup text
+    Popup::ShowFeedback ("",_("Verifying AD domain membership..."));
+    SambaAD::ReadADS (domain);
+    if (SambaAD::ADS () != "")
+    {
+	domain = SambaAD::GetWorkgroup (domain);
+	Samba::SetWorkgroup(domain);
+	SambaAD::ReadRealm ();
+    }
+    Popup::ClearFeedback ();
+
+    term leave_button	= (SambaAD::ADS () == "") ? `Empty () :
+	// push button label
+	`PushButton (`id (`leave), _("&Leave"));
+    UI::ReplaceWidget (`id(`rpstatus),
+	(Stage::cont () || CheckWorkgroup (domain) != `joined_domain) ?
+	`Empty() :
+	`HBox (
+	    // status label
+	    `Left (`Label (_("Currently a member of this domain"))),
+	    `HStretch (),
+	    leave_button
+	)
+    );
+}
+
 /**
  * Samba memberhip dialog
  * @return dialog result
@@ -354,40 +389,6 @@ symbol MembershipDialog () {
     map pw_data	= Samba::password_data;
     string left_domain	= "";
 
-    // internal function: update the status line
-    void check_domain_membership (string domain) {
-
-	Samba::SetWorkgroup(domain);
-
-	if (Mode::config ())
-	    return;
-
-	// busy popup text
-	Popup::ShowFeedback ("",_("Verifying AD domain membership..."));
-	SambaAD::ReadADS (domain);
-	if (SambaAD::ADS () != "")
-	{
-	    domain = SambaAD::GetWorkgroup (domain);
-	    Samba::SetWorkgroup(domain);
-	    SambaAD::ReadRealm ();
-	}
-	Popup::ClearFeedback ();
-
-	term leave_button	= (SambaAD::ADS () == "") ? `Empty () :
-	    // push button label
-	    `PushButton (`id (`leave), _("&Leave"));
-	UI::ReplaceWidget (`id(`rpstatus),
-	    (Stage::cont () || CheckWorkgroup (domain) != `joined_domain) ?
-	    `Empty() :
-	    `HBox (
-		// status label
-		`Left (`Label (_("Currently a member of this domain"))),
-		`HStretch (),
-		leave_button
-	    )
-	);
-    }
-
     // winbind enabled on start
     boolean was_winbind	= Samba::GetWinbind();
 
