diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-scanner.spec.in b/yast2-scanner.spec.in
index 625ac90..0fcf859 100644
--- a/yast2-scanner.spec.in
+++ b/yast2-scanner.spec.in
@@ -8,6 +8,8 @@ Requires:	yast2
 # which are pulled in by Autoreqprov because of the displaytest tool:
 %define my_requires /tmp/my-requires
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Scanner Configuration
 License:        GPL-2.0
 
@@ -42,7 +44,7 @@ chmod 755 %{my_requires}
 %defattr(-,root,root)
 %dir @yncludedir@/scanner
 @yncludedir@/scanner/*
-@clientdir@/scanner.ycp
+@clientdir@/scanner.rb
 @moduledir@/Scanner.*
 @desktopdir@/scanner.desktop
 @ybindir@/*scanner*
