diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-security.spec.in b/yast2-security.spec.in
index a60270e..5d7f9f3 100644
--- a/yast2-security.spec.in
+++ b/yast2-security.spec.in
@@ -19,6 +19,8 @@ Obsoletes:	yast2-trans-security y2t_sec
 
 BuildArchitectures: noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Security Configuration
 
 %description
@@ -43,8 +45,8 @@ fi
 %dir @yncludedir@/security
 @yncludedir@/security/*
 @desktopdir@/security.desktop
-@clientdir@/security*.ycp
-@moduledir@/Security.y*
+@clientdir@/security*.rb
+@moduledir@/Security.rb
 @scrconfdir@/*.scr
 @schemadir@/autoyast/rnc/security.rnc
 %doc @docdir@
