TODO: update BuildRequires upstream

diff --git a/agents/Makefile.am b/agents/Makefile.am
index fffe01f..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,7 +0,0 @@
-# Makefile.am for slp-server/agents
-
-agent_SCRIPTS =
-
-scrconf_DATA = slp_conf.scr slp_reg.scr
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 5e4329f..783afb4 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,5 @@
 # Makefile.am for slp-server/doc/autodocs
 
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled
+
+#include $(top_srcdir)/autodocs-ycp.ami
diff --git a/src/include/slp-server/dialogs.ycp b/src/include/slp-server/dialogs.ycp
index 5004c0f..cc955c9 100644
--- a/src/include/slp-server/dialogs.ycp
+++ b/src/include/slp-server/dialogs.ycp
@@ -23,7 +23,9 @@ import "LogView";
 import "IP";
 import "String";
 import "Address";
+import "Popup";
 include "slp-server/helps.ycp";
+include "slp-server/complex.ycp";
 
 void initServerSettings(string key);
 symbol handleServerSettings(string key, map event );
diff --git a/yast2-slp-server.spec.in b/yast2-slp-server.spec.in
index 9d75923..5699f61 100644
--- a/yast2-slp-server.spec.in
+++ b/yast2-slp-server.spec.in
@@ -4,7 +4,7 @@
 @HEADER@
 Group:	System/YaST
 License: GPL-2.0+
-BuildRequires:	docbook-xsl-stylesheets doxygen libxslt perl-XML-Writer popt-devel sgml-skel update-desktop-files yast2 yast2-devtools yast2-packagemanager-devel yast2-perl-bindings yast2-testsuite
+BuildRequires:	docbook-xsl-stylesheets doxygen libxslt perl-XML-Writer popt-devel sgml-skel update-desktop-files yast2 yast2-devtools yast2-testsuite
 
 # network needs Wizard::OpenCancelOKDialog()
 #  function from yast2-2.18.2
@@ -13,6 +13,8 @@ Requires:       yast2 >= 2.21.22
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 SLP Daemon Server Configuration
 
 %description
@@ -31,8 +33,8 @@ SLP daemon.
 %defattr(-,root,root)
 %dir @yncludedir@/slp-server
 @yncludedir@/slp-server/*
-@clientdir@/slp-server.ycp
-@clientdir@/slp-server_*.ycp
+@clientdir@/slp-server.rb
+@clientdir@/slp-server_*.rb
 @moduledir@/SlpServer.*
 @desktopdir@/slp-server.desktop
 @scrconfdir@/slp*.scr
