diff --git a/agent-snapper/conf/Makefile.am b/agent-snapper/conf/Makefile.am
index 40c39fb..e69de29 100644
--- a/agent-snapper/conf/Makefile.am
+++ b/agent-snapper/conf/Makefile.am
@@ -1,7 +0,0 @@
-#
-# Makefile.am for .../agent-snapper/conf
-#
-
-scrconf_DATA = snapper.scr
-
-EXTRA_DIST = $(scrconf_DATA)
diff --git a/agent-snapper/doc/autodocs/Makefile.am b/agent-snapper/doc/autodocs/Makefile.am
index 3ca9c78..a599bae 100644
--- a/agent-snapper/doc/autodocs/Makefile.am
+++ b/agent-snapper/doc/autodocs/Makefile.am
@@ -2,5 +2,7 @@
 # Makefile.am for .../agent-snapper/doc/autodocs
 #
 
-AUTODOCS_SUBDIR=agent-snapper
-include $(top_srcdir)/autodocs-cc.ami
+# TODO FIXME: temporarily disabled
+
+#AUTODOCS_SUBDIR=agent-snapper
+#include $(top_srcdir)/autodocs-cc.ami
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 2cbc75e..17633f0 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,5 @@
 # Makefile.am for snapper/doc/autodocs
 
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled
+
+#include $(top_srcdir)/autodocs-ycp.ami
diff --git a/yast2-snapper.spec.in b/yast2-snapper.spec.in
index c91664f..c91c855 100644
--- a/yast2-snapper.spec.in
+++ b/yast2-snapper.spec.in
@@ -8,6 +8,8 @@ BuildRequires:	libsnapper-devel >= 0.0.11
 BuildRequires:	yast2-core-devel >= 2.23.1
 BuildRequires:	libtool doxygen gcc-c++ perl-XML-Writer
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST - file system snapshots review
 
 %description
@@ -25,7 +27,7 @@ YaST module for accessing and managing btrfs system snapshots
 %defattr(-,root,root)
 %dir @yncludedir@/snapper
 @yncludedir@/snapper/*
-@clientdir@/snapper.ycp
+@clientdir@/snapper.rb
 @moduledir@/Snapper.*
 @desktopdir@/snapper.desktop
 @scrconfdir@/*.scr
