diff --git a/data/Makefile.am b/data/Makefile.am
index 7d66112..f7efc11 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -6,7 +6,7 @@
 
 SUBDIRS = doc
 
-ydata_DATA = alsa_packages.ycp sndcards.ycp
+ydata_DATA = sndcards.ycp
 
 sound_db_generator = alsa_drivers.rb data_cards.rb data_joystick.rb data_mixer.rb data_extra_ids.rb generate_soundcard_db.rb sound_db_generator.rb ycp_format.rb
 
@@ -17,4 +17,4 @@ sndcards.ycp: $(sound_db_generator)
 
 CLEANFILES = sndcards.ycp
 
-EXTRA_DIST = alsa_packages.ycp $(sound_db_generator)
+EXTRA_DIST = $(sound_db_generator)
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 2a6f678..19b5419 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,3 @@
 # Makefile.am for YCP module .../doc/autodocs
 
-include $(top_srcdir)/autodocs-ycp.ami
+# include $(top_srcdir)/autodocs-ycp.ami
diff --git a/sound/Makefile.am b/sound/Makefile.am
index 0764d69..9ef8dbb 100644
--- a/sound/Makefile.am
+++ b/sound/Makefile.am
@@ -1,3 +1,3 @@
 #
 # Makefile.am for sound/sound
-SUBDIRS = doc src testsuite agents config
+SUBDIRS =
diff --git a/src/include/sound/complex.ycp b/src/include/sound/complex.ycp
index 93d457e..fe46c42 100644
--- a/src/include/sound/complex.ycp
+++ b/src/include/sound/complex.ycp
@@ -627,4 +627,4 @@ define map sound_complex() ``{
 //    return ComplexDialog ();
 }
 
-}//EOF
+}
diff --git a/src/include/sound/mixer.ycp b/src/include/sound/mixer.ycp
index 7074342..78239f5 100644
--- a/src/include/sound/mixer.ycp
+++ b/src/include/sound/mixer.ycp
@@ -365,4 +365,4 @@ Press <B>Next</B> to save your volume settings, press <B>Back</B> to restore the
 
 }
 
-}//EOF
+}
diff --git a/src/include/sound/options.ycp b/src/include/sound/options.ycp
index 3a5b6d9..78f75ed 100644
--- a/src/include/sound/options.ycp
+++ b/src/include/sound/options.ycp
@@ -699,4 +701,4 @@
 	return $["ui" : result["ui"]:`cancel, "return" : save_entry];
     }
 
-}//EOF
+}
diff --git a/src/include/sound/routines.ycp b/src/include/sound/routines.ycp
index 54655b0..e6b10fc 100644
--- a/src/include/sound/routines.ycp
+++ b/src/include/sound/routines.ycp
@@ -1076,4 +1076,4 @@ could make your system unstable.
 	});
     }
 
-}//EOF
+}
diff --git a/src/include/sound/texts.ycp b/src/include/sound/texts.ycp
index 0fd570b..bce9eb3 100644
--- a/src/include/sound/texts.ycp
+++ b/src/include/sound/texts.ycp
@@ -326,5 +326,5 @@ Use <b>Add sound card</b> to add a sound card.
 	return ret;
     }
 
-}//EOF
+}
 
diff --git a/src/include/sound/ui.ycp b/src/include/sound/ui.ycp
index 7a09398..e300816 100644
--- a/src/include/sound/ui.ycp
+++ b/src/include/sound/ui.ycp
@@ -159,4 +159,4 @@
 	return $["ui" : output];
     }
 
-}//EOF
+}
diff --git a/src/include/sound/volume.ycp b/src/include/sound/volume.ycp
index 39e6ea1..a6260e3 100644
--- a/src/include/sound/volume.ycp
+++ b/src/include/sound/volume.ycp
@@ -399,4 +399,4 @@ define map sound_volume (map save_entry, integer card_id, boolean finish,
     return $["ui" : ui, "return" : size (err_msg) == 0];
 }
 
-}//EOF
+}
diff --git a/src/include/sound/volume_routines.ycp b/src/include/sound/volume_routines.ycp
index 6fefcc8..edcf496 100644
--- a/src/include/sound/volume_routines.ycp
+++ b/src/include/sound/volume_routines.ycp
@@ -283,4 +283,4 @@
 	return true;
     }
 
-}//EOF
+}
diff --git a/yast2-sound.spec.in b/yast2-sound.spec.in
index 1b9ba31..3bce60f 100644
--- a/yast2-sound.spec.in
+++ b/yast2-sound.spec.in
@@ -17,6 +17,8 @@ Obsoletes:	yast2-trans-sound yast2-trans-soundd y2c_snd y2t_snd y2t_sndd
 Provides:	y2c_sparc y2c_sprc yast2-db-sound y2d_snd
 Obsoletes:	y2c_sparc y2c_sprc yast2-db-sound y2d_snd
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Sound Configuration
 
 %description
@@ -42,9 +44,9 @@ fi
 
 # sound
 %dir @yncludedir@/sound
-@yncludedir@/sound/*.ycp
-@clientdir@/sound*.ycp
-@clientdir@/joystick.ycp
+@yncludedir@/sound/*.rb
+@clientdir@/sound*.rb
+@clientdir@/joystick.rb
 @desktopdir@/sound.desktop
 @desktopdir@/joystick.desktop
 @ybindir@/copyfonts
