diff --git a/doc/Makefile.am b/doc/Makefile.am
index f85d6ff..ed7906a 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -1,6 +1,7 @@
 # Makefile.am for sshd/doc
 
-SUBDIRS = autodocs
+# TODO not supported ruby yet
+#SUBDIRS = autodocs
 
 htmldir = $(docdir)
 
diff --git a/src/clients/sshd_auto.ycp b/src/clients/sshd_auto.ycp
index 28e831e..af311ca 100644
--- a/src/clients/sshd_auto.ycp
+++ b/src/clients/sshd_auto.ycp
@@ -115,8 +115,8 @@ switch (func) {
 	ret = Sshd::Write();
 	Progress::set (orig2);
 	break;
-    
-    y2error("Unknown function: %1, Params %2", func, param);
+    default:  
+        y2error("Unknown function: %1, Params %2", func, param);
 }
 
 /* Finish */
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-sshd.spec.in b/yast2-sshd.spec.in
index 3cd820b..f7fd564 100644
--- a/yast2-sshd.spec.in
+++ b/yast2-sshd.spec.in
@@ -9,6 +9,8 @@ BuildRequires:	perl-XML-Writer update-desktop-files yast2 yast2-devtools yast2-t
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - SSH Server Configuration
 
 %description
@@ -26,10 +28,9 @@ This package contains the YaST2 component for SSH server configuration.
 %defattr(-,root,root)
 %dir @yncludedir@/sshd
 @yncludedir@/sshd/*
-@clientdir@/sshd.ycp
-@clientdir@/sshd_auto.ycp
-@moduledir@/*.ybc
-@moduledir@/*.ycp
+@clientdir@/sshd.rb
+@clientdir@/sshd_auto.rb
+@moduledir@/*.rb
 @desktopdir@/sshd.desktop
 @schemadir@/autoyast/rnc/sshd_config.rnc
 %doc @docdir@
