diff --git a/doc/Makefile.am b/doc/Makefile.am
index f85d6ff..ed7906a 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -1,6 +1,7 @@
 # Makefile.am for sshd/doc
 
-SUBDIRS = autodocs
+# TODO not supported ruby yet
+#SUBDIRS = autodocs
 
 htmldir = $(docdir)
 
diff --git a/src/clients/sshd_auto.ycp b/src/clients/sshd_auto.ycp
index 28e831e..af311ca 100644
--- a/src/clients/sshd_auto.ycp
+++ b/src/clients/sshd_auto.ycp
@@ -115,8 +115,8 @@ switch (func) {
 	ret = Sshd::Write();
 	Progress::set (orig2);
 	break;
-    
-    y2error("Unknown function: %1, Params %2", func, param);
+    default:  
+        y2error("Unknown function: %1, Params %2", func, param);
 }
 
 /* Finish */
diff --git a/src/include/sshd/helps.ycp b/src/include/sshd/helps.ycp
index a91b9d0..5301061 100644
--- a/src/include/sshd/helps.ycp
+++ b/src/include/sshd/helps.ycp
@@ -52,29 +52,29 @@ Configure SSHD here.<br></p>"),
     /* Login Settings dialog help 1 */
     "login_settings" : _("<p><b><big>Login Settings</big></b><br>
 Here you can configure the SSHD login and authentication settings.
-Some apply to a particular protocol version only.</p>")
+Some apply to a particular protocol version only.</p>") +
 
     /* Login Settings dialog help 2, text taken from sshd_config manpage */
-    + _("<p><b>Print Message of the Day After Login</b>
+    _("<p><b>Print Message of the Day After Login</b>
 Specifies whether sshd should print /etc/motd when
-a user logs in interactively.</p>")
+a user logs in interactively.</p>") +
 
     /* Login Settings dialog help 3, text taken from sshd_config manpage */
-    + _("<p><b>Permit Root Login</b><br>
-Specifies whether root can log in using ssh.</p>")
+    _("<p><b>Permit Root Login</b><br>
+Specifies whether root can log in using ssh.</p>") +
 
     /* Login Settings dialog help 4, text taken from sshd_config manpage */
-    + _("<p><b>Maximum Authentication Tries</b><br>
+    _("<p><b>Maximum Authentication Tries</b><br>
 Specifies the maximum number of authentication attempts permitted per connection.
-Once the number of failures reaches half this value, additional failures are logged.</p>")
+Once the number of failures reaches half this value, additional failures are logged.</p>") +
 
     /* Login Settings dialog help 5, text taken from sshd_config manpage */
-    + _("<p><b>RSA Authentication</b><br>
+    _("<p><b>RSA Authentication</b><br>
 Specifies whether pure RSA authentication is allowed.
-This option applies to protocol version 1 only.</p>")
+This option applies to protocol version 1 only.</p>") +
 
     /* Login Settings dialog help 6, text taken from sshd_config manpage */
-    + _("<p><b>Public Key Authentication</b><br>
+    _("<p><b>Public Key Authentication</b><br>
 Specifies whether public key authentication is allowed.
 This option applies to protocol version 2 only.</p>"),
 
diff --git a/yast2-sshd.spec.in b/yast2-sshd.spec.in
index 3cd820b..f7fd564 100644
--- a/yast2-sshd.spec.in
+++ b/yast2-sshd.spec.in
@@ -9,6 +9,8 @@ BuildRequires:	perl-XML-Writer update-desktop-files yast2 yast2-devtools yast2-t
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - SSH Server Configuration
 
 %description
@@ -26,10 +28,9 @@ This package contains the YaST2 component for SSH server configuration.
 %defattr(-,root,root)
 %dir @yncludedir@/sshd
 @yncludedir@/sshd/*
-@clientdir@/sshd.ycp
-@clientdir@/sshd_auto.ycp
-@moduledir@/*.ybc
-@moduledir@/*.ycp
+@clientdir@/sshd.rb
+@clientdir@/sshd_auto.rb
+@moduledir@/*.rb
 @desktopdir@/sshd.desktop
 @schemadir@/autoyast/rnc/sshd_config.rnc
 %doc @docdir@
