diff --git a/agents-scr/Makefile.am b/agents-scr/Makefile.am
deleted file mode 100644
index a2c2946..0000000
--- a/agents-scr/Makefile.am
+++ /dev/null
@@ -1,12 +0,0 @@
-#
-# Makefile.am for storage/agents-scr
-#
-
-scrconf_DATA = etc_cryptotab.scr	\
-	       etc_mtab.scr		\
-	       proc_dasddev.scr		\
-	       proc_partitions.scr	\
-	       proc_swaps.scr		\
-	       sysconfig_storage.scr
-
-EXTRA_DIST = $(scrconf_DATA)
diff --git a/bindings/ycp/Makefile.am b/bindings/ycp/Makefile.am
index 7968bc0..a5f0232 100644
--- a/bindings/ycp/Makefile.am
+++ b/bindings/ycp/Makefile.am
@@ -11,4 +11,4 @@ PLUGLIB_DEPEND =					\
 
 include $(top_srcdir)/pluglib-bindings.ami
 
-EXTRA_DIST = LibStorage.i example.ycp
+EXTRA_DIST = LibStorage.i example.rb callback.rb
diff --git a/data/Makefile.am b/data/Makefile.am
deleted file mode 100644
index 1e7e401..0000000
--- a/data/Makefile.am
+++ /dev/null
@@ -1,9 +0,0 @@
-#
-# Makefile.am for storage/data
-#
-
-fillup_DATA = sysconfig.storage-yast2-storage
-
-EXTRA_DIST = $(fillup_DATA)
-
-include $(top_srcdir)/Makefile.am.common
diff --git a/disk/Makefile.am b/disk/Makefile.am
deleted file mode 100644
index a04d172..0000000
--- a/disk/Makefile.am
+++ /dev/null
@@ -1,5 +0,0 @@
-#
-# Makefile.am for storage/disk
-#
-
-SUBDIRS = src
diff --git a/disk/src/Makefile.am b/disk/src/Makefile.am
deleted file mode 100644
index 69ea25b..0000000
--- a/disk/src/Makefile.am
+++ /dev/null
@@ -1,12 +0,0 @@
-#
-# Makefile.am for storage/disk/src
-#
-
-client_DATA =			\
-	storage.ycp		\
-	disk.ycp		\
-	disk_worker.ycp
-
-desktop_DATA = disk.desktop
-
-EXTRA_DIST = $(client_DATA) $(desktop_DATA)
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 9749703..99d2f05 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -2,10 +2,5 @@
 # Makefile.am for storage/storage/doc/autodocs
 #
 
-htmldir = @docdir@/autodocs
+include $(top_srcdir)/autodocs-ycp.ami
 
-html_DATA = index.html $(filter-out index.html, $(wildcard *.html))
-CLEANFILES = $(html_DATA)
-
-index.html: $(srcdir)/../../src/*.ycp $(srcdir)/../../src/include/*.ycp $(srcdir)/../../src/modules/*.ycp
-	@ybindir@/ycpdoc -d. $^
diff --git a/multipath-simple/Makefile.am b/multipath-simple/Makefile.am
deleted file mode 100644
index 77cf4ba..0000000
--- a/multipath-simple/Makefile.am
+++ /dev/null
@@ -1,5 +0,0 @@
-#
-# Makefile.am for storage/multipath-simple
-#
-
-SUBDIRS = src
diff --git a/multipath-simple/src/Makefile.am b/multipath-simple/src/Makefile.am
deleted file mode 100644
index 7585ae1..0000000
--- a/multipath-simple/src/Makefile.am
+++ /dev/null
@@ -1,8 +0,0 @@
-#
-# Makefile.am for storage/multipath-simple/src
-#
-
-client_DATA =			\
-	multipath-simple.ycp
-
-EXTRA_DIST = $(client_DATA)
diff --git a/scripts/Makefile.am b/scripts/Makefile.am
deleted file mode 100644
index 3795ef4..0000000
--- a/scripts/Makefile.am
+++ /dev/null
@@ -1,8 +0,0 @@
-#
-# Makefile.am for storage/scripts
-#
-
-ybin_SCRIPTS =			\
-	check.boot
-
-EXTRA_DIST = $(ybin_SCRIPTS)
diff --git a/storage/Makefile.am b/storage/Makefile.am
deleted file mode 100644
index 3e4cfac..0000000
--- a/storage/Makefile.am
+++ /dev/null
@@ -1,5 +0,0 @@
-#
-# Makefile.am for storage/storage
-#
-
-SUBDIRS = data doc src
diff --git a/storage/src/Makefile.am b/storage/src/Makefile.am
deleted file mode 100644
index 3720ffa..0000000
--- a/storage/src/Makefile.am
+++ /dev/null
@@ -1,25 +0,0 @@
-#
-# Makefile.am for storage/storage/src
-#
-
-SUBDIRS = include modules proposal testsuite
-
-client_DATA =				\
-	inst_custom_part.ycp		\
-	inst_disk.ycp			\
-	inst_disk_proposal.ycp		\
-	inst_prepdisk.ycp		\
-	inst_resize_ui.ycp		\
-	inst_resize_dialog.ycp		\
-	inst_target_part.ycp		\
-	inst_target_selection.ycp	\
-	storage_finish.ycp
-
-EXTRA_DIST = $(client_DATA)
-
-partitioning:
-	ln -s include partitioning
-
-ycpchook = partitioning
-
-include $(top_srcdir)/Makefile.am.common
diff --git a/storage/src/include/Makefile.am b/storage/src/include/Makefile.am
deleted file mode 100644
index e2a0123..0000000
--- a/storage/src/include/Makefile.am
+++ /dev/null
@@ -1,9 +0,0 @@
-#
-# Makefile.am for storage/storage/src/include
-#
-
-yncludedir = @yncludedir@/partitioning
-
-ynclude_HEADERS = $(wildcard *.ycp)
-
-EXTRA_DIST = $(ynclude_HEADERS)
diff --git a/storage/src/modules/Makefile.am b/storage/src/modules/Makefile.am
deleted file mode 100644
index eae6439..0000000
--- a/storage/src/modules/Makefile.am
+++ /dev/null
@@ -1,33 +0,0 @@
-#
-# Makefile.am for storage/storage/src/modules
-#
-
-YCPCFLAGS = -M $(top_builddir)/bindings/ycp
-
-YCPC_LD_LIBRARY_PATH = $(top_builddir)/bindings/src/.libs
-YCPC_Y2DIR = $(top_builddir)/bindings/src/.libs
-
-module_DATA =				\
-	StorageInit.ycp      		\
-	Partitions.ycp			\
-	FileSystems.ycp			\
-	Storage.ycp			\
-	StorageClients.ycp      	\
-	StorageControllers.ycp  	\
-	StorageDevices.ycp		\
-	StorageFields.ycp		\
-	StorageIcons.ycp		\
-	StorageProposal.ycp		\
-	StorageSettings.ycp		\
-	StorageUpdate.ycp		\
-	TreePanel.ycp			\
-	TabPanel.ycp			\
-	MiniWorkflow.ycp		\
-	Greasemonkey.ycp		\
-	DevicesSelectionBox.ycp		\
-	DualMultiSelectionBox.ycp	\
-	Region.ycp
-
-EXTRA_DIST = $(module_DATA)
-
-include $(top_srcdir)/Makefile.am.common
diff --git a/storage/src/proposal/Makefile.am b/storage/src/proposal/Makefile.am
deleted file mode 100644
index 2659b2f..0000000
--- a/storage/src/proposal/Makefile.am
+++ /dev/null
@@ -1,8 +0,0 @@
-#
-# Makefile.am for storage/storage/src/proposal
-#
-
-client_DATA = partitions_proposal.ycp
-EXTRA_DIST = $(client_DATA)
-
-include $(top_srcdir)/Makefile.am.common
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index 8191677..d5a583e 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -7,9 +7,9 @@
 SUBDIRS = data
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
-Y2BASEFLAGS = -M $(top_builddir)/bindings/ycp
+Y2BASEFLAGS = -M $(top_builddir)/bindings/ycp -I tests
 export Y2BASEFLAGS
 
 Y2BASE_LD_LIBRARY_PATH = $(top_builddir)/bindings/src/.libs
diff --git a/testsuite/tests/empty-efi1.ycp b/testsuite/tests/empty-efi1.ycp
index 4c5345d..ad1d0ee 100644
--- a/testsuite/tests/empty-efi1.ycp
+++ b/testsuite/tests/empty-efi1.ycp
@@ -1,10 +1,10 @@
 // testedfiles: helper1b.yh
 {
-    include "./tests/setup-system.yh";
+    include "setup-system.yh";
 
     setup_system("empty-efi");
 
-    include "./tests/helper1a.yh";
+    include "helper1a.yh";
 
     import "ProductFeatures";
 
@@ -13,5 +13,5 @@
     ProductFeatures::SetStringFeature("partitioning", "vm_desired_size", "30 GB");
     ProductFeatures::SetStringFeature("partitioning", "root_base_size", "20 GB");
 
-    include "./tests/helper1b.yh";
+    include "helper1b.yh";
 }
diff --git a/testsuite/tests/empty-efi2.ycp b/testsuite/tests/empty-efi2.ycp
index 92de13a..3acd4f2 100644
--- a/testsuite/tests/empty-efi2.ycp
+++ b/testsuite/tests/empty-efi2.ycp
@@ -1,10 +1,10 @@
 // testedfiles: helper1b.yh
 {
-    include "./tests/setup-system.yh";
+    include "setup-system.yh";
 
     setup_system("empty-efi");
 
-    include "./tests/helper1a.yh";
+    include "helper1a.yh";
 
     import "ProductFeatures";
 
@@ -13,5 +13,5 @@
     ProductFeatures::SetStringFeature("partitioning", "vm_desired_size", "30 GB");
     ProductFeatures::SetStringFeature("partitioning", "root_base_size", "20 GB");
 
-    include "./tests/helper1b.yh";
+    include "helper1b.yh";
 }
diff --git a/testsuite/tests/empty1.ycp b/testsuite/tests/empty1.ycp
index 827e006..915cd3c 100644
--- a/testsuite/tests/empty1.ycp
+++ b/testsuite/tests/empty1.ycp
@@ -1,10 +1,10 @@
 // testedfiles: helper1b.yh
 {
-    include "./tests/setup-system.yh";
+    include "setup-system.yh";
 
     setup_system("empty");
 
-    include "./tests/helper1a.yh";
+    include "helper1a.yh";
 
     import "ProductFeatures";
 
@@ -13,5 +13,5 @@
     ProductFeatures::SetStringFeature("partitioning", "vm_desired_size", "30 GB");
     ProductFeatures::SetStringFeature("partitioning", "root_base_size", "20 GB");
 
-    include "./tests/helper1b.yh";
+    include "helper1b.yh";
 }
diff --git a/testsuite/tests/empty2.ycp b/testsuite/tests/empty2.ycp
index c688058..b6e23a5 100644
--- a/testsuite/tests/empty2.ycp
+++ b/testsuite/tests/empty2.ycp
@@ -1,10 +1,10 @@
 // testedfiles: helper1b.yh
 {
-    include "./tests/setup-system.yh";
+    include "setup-system.yh";
 
     setup_system("empty");
 
-    include "./tests/helper1a.yh";
+    include "helper1a.yh";
 
     import "ProductFeatures";
 
@@ -13,5 +13,5 @@
     ProductFeatures::SetStringFeature("partitioning", "vm_desired_size", "30 GB");
     ProductFeatures::SetStringFeature("partitioning", "root_base_size", "20 GB");
 
-    include "./tests/helper1b.yh";
+    include "helper1b.yh";
 }
diff --git a/testsuite/tests/empty3.ycp b/testsuite/tests/empty3.ycp
index ca57fd0..46ad816 100644
--- a/testsuite/tests/empty3.ycp
+++ b/testsuite/tests/empty3.ycp
@@ -1,10 +1,10 @@
 // testedfiles: helper1b.yh
 {
-    include "./tests/setup-system.yh";
+    include "setup-system.yh";
 
     setup_system("empty");
 
-    include "./tests/helper1a.yh";
+    include "helper1a.yh";
 
     import "ProductFeatures";
 
@@ -14,5 +14,5 @@
     ProductFeatures::SetStringFeature("partitioning", "vm_desired_size", "30 GB");
     ProductFeatures::SetStringFeature("partitioning", "root_base_size", "20 GB");
 
-    include "./tests/helper1b.yh";
+    include "helper1b.yh";
 }
diff --git a/testsuite/tests/empty4.ycp b/testsuite/tests/empty4.ycp
index 2943925..c4eb640 100644
--- a/testsuite/tests/empty4.ycp
+++ b/testsuite/tests/empty4.ycp
@@ -1,10 +1,10 @@
 // testedfiles: helper1b.yh
 {
-    include "./tests/setup-system.yh";
+    include "setup-system.yh";
 
     setup_system("empty");
 
-    include "./tests/helper1a.yh";
+    include "helper1a.yh";
 
     import "ProductFeatures";
 
@@ -14,5 +14,5 @@
     ProductFeatures::SetStringFeature("partitioning", "vm_desired_size", "30 GB");
     ProductFeatures::SetStringFeature("partitioning", "root_base_size", "20 GB");
 
-    include "./tests/helper1b.yh";
+    include "helper1b.yh";
 }
diff --git a/testsuite/tests/empty5.ycp b/testsuite/tests/empty5.ycp
index 96e11c1..e88faf0 100644
--- a/testsuite/tests/empty5.ycp
+++ b/testsuite/tests/empty5.ycp
@@ -1,10 +1,10 @@
 // testedfiles: helper1b.yh
 {
-    include "./tests/setup-system.yh";
+    include "setup-system.yh";
 
     setup_system("empty");
 
-    include "./tests/helper1a.yh";
+    include "helper1a.yh";
 
     import "ProductFeatures";
 
@@ -13,5 +13,5 @@
     ProductFeatures::SetStringFeature("partitioning", "vm_desired_size", "30 GB");
     ProductFeatures::SetStringFeature("partitioning", "root_base_size", "20 GB");
 
-    include "./tests/helper1b.yh";
+    include "helper1b.yh";
 }
diff --git a/testsuite/tests/empty6.ycp b/testsuite/tests/empty6.ycp
index 53d2154..0046892 100644
--- a/testsuite/tests/empty6.ycp
+++ b/testsuite/tests/empty6.ycp
@@ -1,10 +1,10 @@
 // testedfiles: helper1b.yh
 {
-    include "./tests/setup-system.yh";
+    include "setup-system.yh";
 
     setup_system("empty");
 
-    include "./tests/helper1a.yh";
+    include "helper1a.yh";
 
     import "ProductFeatures";
 
@@ -13,5 +13,5 @@
     ProductFeatures::SetStringFeature("partitioning", "vm_desired_size", "30 GB");
     ProductFeatures::SetStringFeature("partitioning", "root_base_size", "20 GB");
 
-    include "./tests/helper1b.yh";
+    include "helper1b.yh";
 }
diff --git a/testsuite/tests/half-empty1.ycp b/testsuite/tests/half-empty1.ycp
index 2bbcf02..4af256b 100644
--- a/testsuite/tests/half-empty1.ycp
+++ b/testsuite/tests/half-empty1.ycp
@@ -1,10 +1,10 @@
 // testedfiles: helper1b.yh
 {
-    include "./tests/setup-system.yh";
+    include "setup-system.yh";
 
     setup_system("half-empty");
 
-    include "./tests/helper1a.yh";
+    include "helper1a.yh";
 
     import "ProductFeatures";
 
@@ -12,5 +12,5 @@
     ProductFeatures::SetBooleanFeature("partitioning", "proposal_lvm", false);
     ProductFeatures::SetStringFeature("partitioning", "root_base_size", "20 GB");
 
-    include "./tests/helper1b.yh";
+    include "helper1b.yh";
 }
diff --git a/testsuite/tests/helper1b.yh b/testsuite/tests/helper1b.yh
index 969f8c6..e42852b 100644
--- a/testsuite/tests/helper1b.yh
+++ b/testsuite/tests/helper1b.yh
@@ -1,6 +1,7 @@
 {
     import "Storage";
     import "StorageProposal";
+    import "Testsuite";
 
     Storage::InitLibstorage(false);
 
diff --git a/testsuite/tests/reuse-home1.ycp b/testsuite/tests/reuse-home1.ycp
index 481de09..7622a00 100644
--- a/testsuite/tests/reuse-home1.ycp
+++ b/testsuite/tests/reuse-home1.ycp
@@ -1,10 +1,10 @@
 // testedfiles: helper1b.yh
 {
-    include "./tests/setup-system.yh";
+    include "setup-system.yh";
 
     setup_system("reuse-home");
 
-    include "./tests/helper1a.yh";
+    include "helper1a.yh";
 
     import "ProductFeatures";
 
@@ -13,5 +13,5 @@
     ProductFeatures::SetStringFeature("partitioning", "root_max_size", "20 GB");
     ProductFeatures::SetStringFeature("partitioning", "root_base_size", "15 GB");
 
-    include "./tests/helper1b.yh";
+    include "helper1b.yh";
 }
diff --git a/testsuite/tests/windows-only-efi1.ycp b/testsuite/tests/windows-only-efi1.ycp
index d222d91..24093a5 100644
--- a/testsuite/tests/windows-only-efi1.ycp
+++ b/testsuite/tests/windows-only-efi1.ycp
@@ -1,10 +1,10 @@
 // testedfiles: helper1b.yh
 {
-    include "./tests/setup-system.yh";
+    include "setup-system.yh";
 
     setup_system("windows-only-efi");
 
-    include "./tests/helper1a.yh";
+    include "helper1a.yh";
 
     import "ProductFeatures";
 
@@ -13,5 +13,5 @@
     ProductFeatures::SetStringFeature("partitioning", "root_max_size", "20 GB");
     ProductFeatures::SetStringFeature("partitioning", "root_base_size", "15 GB");
 
-    include "./tests/helper1b.yh";
+    include "helper1b.yh";
 }
diff --git a/testsuite/tests/windows-only1.ycp b/testsuite/tests/windows-only1.ycp
index 1ef97a5..06cf6a3 100644
--- a/testsuite/tests/windows-only1.ycp
+++ b/testsuite/tests/windows-only1.ycp
@@ -1,10 +1,10 @@
 // testedfiles: helper1b.yh
 {
-    include "./tests/setup-system.yh";
+    include "setup-system.yh";
 
     setup_system("windows-only");
 
-    include "./tests/helper1a.yh";
+    include "helper1a.yh";
 
     import "ProductFeatures";
 
@@ -13,5 +13,5 @@
     ProductFeatures::SetStringFeature("partitioning", "root_max_size", "20 GB");
     ProductFeatures::SetStringFeature("partitioning", "root_base_size", "15 GB");
 
-    include "./tests/helper1b.yh";
+    include "helper1b.yh";
 }
diff --git a/testsuite/tests/windows1.ycp b/testsuite/tests/windows1.ycp
index 39df374..c751218 100644
--- a/testsuite/tests/windows1.ycp
+++ b/testsuite/tests/windows1.ycp
@@ -1,10 +1,10 @@
 // testedfiles: helper1b.yh
 {
-    include "./tests/setup-system.yh";
+    include "setup-system.yh";
 
     setup_system("windows");
 
-    include "./tests/helper1a.yh";
+    include "helper1a.yh";
 
     import "ProductFeatures";
 
@@ -13,5 +13,5 @@
     ProductFeatures::SetStringFeature("partitioning", "root_max_size", "20 GB");
     ProductFeatures::SetStringFeature("partitioning", "root_base_size", "15 GB");
 
-    include "./tests/helper1b.yh";
+    include "helper1b.yh";
 }
diff --git a/yast2-storage.spec.in b/yast2-storage.spec.in
index 8ae9071..cad9820 100644
--- a/yast2-storage.spec.in
+++ b/yast2-storage.spec.in
@@ -23,6 +23,8 @@ Provides:	yast2-trans-inst-partitioning
 Obsoletes:	yast2-trans-inst-partitioning
 Provides:	y2t_inst-partitioning
 Obsoletes:	y2t_inst-partitioning
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Storage Configuration
 
 %description
@@ -47,21 +49,21 @@ rm -f $RPM_BUILD_ROOT/@plugindir@/libpy2StorageCallbacks.so
 
 # storage
 %dir @yncludedir@/partitioning
-@yncludedir@/partitioning/*.ycp
-@clientdir@/inst_custom_part.ycp
-@clientdir@/inst_resize_ui.ycp
-@clientdir@/inst_resize_dialog.ycp
-@clientdir@/inst_disk.ycp
-@clientdir@/inst_target_part.ycp
-@clientdir@/inst_disk_proposal.ycp
-@clientdir@/inst_target_selection.ycp
-@clientdir@/inst_prepdisk.ycp
-@clientdir@/storage_finish.ycp
-@clientdir@/partitions_proposal.ycp
-@clientdir@/storage.ycp
-@clientdir@/disk.ycp
-@clientdir@/disk_worker.ycp
-@clientdir@/multipath-simple.ycp
+@yncludedir@/partitioning/*.rb
+@clientdir@/inst_custom_part.rb
+@clientdir@/inst_resize_ui.rb
+@clientdir@/inst_resize_dialog.rb
+@clientdir@/inst_disk.rb
+@clientdir@/inst_target_part.rb
+@clientdir@/inst_disk_proposal.rb
+@clientdir@/inst_target_selection.rb
+@clientdir@/inst_prepdisk.rb
+@clientdir@/storage_finish.rb
+@clientdir@/partitions_proposal.rb
+@clientdir@/storage.rb
+@clientdir@/disk.rb
+@clientdir@/disk_worker.rb
+@clientdir@/multipath-simple.rb
 @moduledir@/*
 @fillupdir@/sysconfig.storage-yast2-storage
 
@@ -91,6 +93,7 @@ rm -f $RPM_BUILD_ROOT/@plugindir@/libpy2StorageCallbacks.so
 %package devel
 Requires:	libstorage-devel = %(echo `rpm -q --queryformat '%{VERSION}' libstorage-devel`)
 Requires:       libstdc++-devel yast2-storage = %version
+
 Summary:        YaST2 - Storage Library Headers and Documentation
 Group:          Development/Libraries/YaST
 
diff --git a/src/include/partitioning/ep-unused.ycp b/src/include/partitioning/ep-unused.ycp
index 0ab4427..d7bb75e 100644
--- a/src/include/partitioning/ep-unused.ycp
+++ b/src/include/partitioning/ep-unused.ycp
@@ -119,6 +136,7 @@ all current changes. Really rescan unused devices?")))
 		    UpdateNavigationTree(nil);
 		    TreePanel::Create();
 		}
+    break;
 	}
     }
 }
