diff --git a/agents/Makefile.am b/agents/Makefile.am
index 28b8611..31180cb 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,7 +1,7 @@
 # Makefile.am for sudo/agents
 
-agent_SCRIPTS = ag_etc_sudoers
+agent_SCRIPTS =
 
-scrconf_DATA = sudo.scr
+scrconf_DATA =
 
 EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/doc/Makefile.am b/doc/Makefile.am
index 2a519b9..4668e9b 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -1,6 +1,7 @@
 # Makefile.am for sudo/doc
 
-SUBDIRS = autodocs
+# TODO autodocs not yet support ruby
+# SUBDIRS = autodocs
 
 htmldir = $(docdir)
 
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-sudo.spec.in b/yast2-sudo.spec.in
index 26b8cd0..c06766f 100644
--- a/yast2-sudo.spec.in
+++ b/yast2-sudo.spec.in
@@ -14,6 +14,8 @@ BuildRequires:	perl-XML-Writer update-desktop-files yast2 yast2-devtools yast2-t
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - sudo configuration
 
 %description
@@ -32,7 +34,7 @@ of users to run commands as root or other user.
 %defattr(-,root,root)
 %dir @yncludedir@/sudo
 @yncludedir@/sudo/*
-@clientdir@/sudo.ycp
+@clientdir@/sudo.rb
 @moduledir@/Sudo.*
 @desktopdir@/sudo.desktop
 @scrconfdir@/sudo.scr
