diff --git a/agents/Makefile.am b/agents/Makefile.am
index cc3cb74..d223884 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -2,6 +2,6 @@
 
 agent_SCRIPTS =
 
-scrconf_DATA = supportconfig.scr
+scrconf_DATA =
 
 EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/doc/Makefile.am b/doc/Makefile.am
index 942c239..33acb79 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -1,6 +1,7 @@
 # Makefile.am for support/doc
 
-SUBDIRS = autodocs
+# TODO autodocs is not yet supported for ruby
+SUBDIRS =# autodocs
 
 htmldir = $(docdir)
 
diff --git a/src/include/support/dialogs.ycp b/src/include/support/dialogs.ycp
index 105e952..a3adbed 100644
--- a/src/include/support/dialogs.ycp
+++ b/src/include/support/dialogs.ycp
@@ -32,12 +32,14 @@
 textdomain "support";
 
 import "Label";
+import "Popup";
 import "Report";
 import "Wizard";
 import "Support";
 import "Map";
 include "support/helps.ycp";
 include "support/supportconfig_conf.ycp";
+include "support/complex.ycp";
 
 any OverviewDialog(){
 
diff --git a/yast2-support.spec.in b/yast2-support.spec.in
index 41404e7..2e15828 100644
--- a/yast2-support.spec.in
+++ b/yast2-support.spec.in
@@ -8,6 +8,8 @@ BuildRequires:	perl-XML-Writer update-desktop-files yast2 yast2-devtools yast2-t
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Support Inquiries
 
 %description
@@ -26,8 +28,8 @@ support in a standardized format.
 %defattr(-,root,root)
 %dir @yncludedir@/support
 @yncludedir@/support/*
-@clientdir@/support.ycp
-@clientdir@/support_*.ycp
+@clientdir@/support.rb
+@clientdir@/support_*.rb
 @moduledir@/Support.*
 @desktopdir@/support.desktop
 @scrconfdir@/*.scr
