diff --git a/doc/Makefile.am b/doc/Makefile.am
index fbc0358..e11ea30 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -2,7 +2,8 @@
 # Makefile.am for devtools/devtools/skeletons/config/doc
 #
 
-SUBDIRS = autodocs
+# TODO autodocs now works only with ycp and perl files
+#SUBDIRS = autodocs
 
 htmldir = $(docdir)
 
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-sysconfig.spec.in b/yast2-sysconfig.spec.in
index 88bf07f..2fba954 100644
--- a/yast2-sysconfig.spec.in
+++ b/yast2-sysconfig.spec.in
@@ -15,6 +15,8 @@ Obsoletes:	y2c_rc_config yast2-config-rcconfig yast2-config-sysconfig
 Provides:	yast2-trans-sysconfig yast2-trans-rcconfig y2t_rc_config
 Obsoletes:	yast2-trans-sysconfig yast2-trans-rcconfig y2t_rc_config
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Sysconfig Editor
 
 %description
@@ -33,10 +35,9 @@ information.
 %defattr(-,root,root)
 %dir @yncludedir@/sysconfig
 @yncludedir@/sysconfig/*
-@clientdir@/sysconfig.ycp
-@clientdir@/sysconfig_*.ycp
-@moduledir@/Sysconfig.ycp
-@moduledir@/Sysconfig.ybc
+@clientdir@/sysconfig.rb
+@clientdir@/sysconfig_*.rb
+@moduledir@/Sysconfig.rb
 @desktopdir@/sysconfig.desktop
 @ybindir@/parse_configs.pl
 @ydatadir@/sysedit.agent
