diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 2a6f678..4cb582c 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,5 @@
 # Makefile.am for YCP module .../doc/autodocs
 
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled
+
+# include $(top_srcdir)/autodocs-ycp.ami
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9277fe..6a1163a 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -6,13 +6,13 @@ AUTOMAKE_OPTIONS =
 
 TESTFLAGS=-I $(top_srcdir)/src -M $(top_srcdir)/src
 check-local:
-	../skel/runtest.sh ${TESTFLAGS} ./test.ycp tmp.out tmp.err
+	../skel/runtest.sh ${TESTFLAGS} ./test.rb tmp.out tmp.err
 	diff -u test.err tmp.err
 	diff -u test.out tmp.out
 
 clean-local:
 	$(RM) tmp.*
 
-EXTRA_DIST= dump_file test.ycp test.out test.err
+EXTRA_DIST= dump_file test.rb test.out test.err
 
 # EOF
diff --git a/testsuite/test.ycp b/testsuite/test.ycp
index 5596f82..f86f329 100644
--- a/testsuite/test.ycp
+++ b/testsuite/test.ycp
@@ -16,7 +16,7 @@
 
 {
 
-include "./../src/testsuite.ycp";
+include "testsuite.ycp";
 
 map READ = $[ "read" : $[ "path" : "read_result" ] ];
 map WRITE = $[ "write" : $[ "path" : true ] ];
diff --git a/yast2-testsuite.spec.in b/yast2-testsuite.spec.in
index 9c5c8b6..83e4fae 100644
--- a/yast2-testsuite.spec.in
+++ b/yast2-testsuite.spec.in
@@ -7,6 +7,8 @@ BuildRequires:	openslp-devel perl-XML-Writer popt-devel yast2-core-devel yast2-d
 Requires:	expect dejagnu
 # y2base -I includepath -M modulepath
 Requires:	yast2-core >= 2.19.0
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Testsuite
 
 BuildArchitectures: noarch
