diff --git a/skel/Makefile_am b/skel/Makefile_am
index d9b12b0..de7f5c5 100644
--- a/skel/Makefile_am
+++ b/skel/Makefile_am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9277fe..6a1163a 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -6,13 +6,13 @@ AUTOMAKE_OPTIONS =
 
 TESTFLAGS=-I $(top_srcdir)/src -M $(top_srcdir)/src
 check-local:
-	../skel/runtest.sh ${TESTFLAGS} ./test.ycp tmp.out tmp.err
+	../skel/runtest.sh ${TESTFLAGS} ./test.rb tmp.out tmp.err
 	diff -u test.err tmp.err
 	diff -u test.out tmp.out
 
 clean-local:
 	$(RM) tmp.*
 
-EXTRA_DIST= dump_file test.ycp test.out test.err
+EXTRA_DIST= dump_file test.rb test.out test.err
 
 # EOF
diff --git a/testsuite/test.ycp b/testsuite/test.ycp
index 5596f82..f86f329 100644
--- a/testsuite/test.ycp
+++ b/testsuite/test.ycp
@@ -16,7 +16,7 @@
 
 {
 
-include "./../src/testsuite.ycp";
+include "testsuite.ycp";
 
 map READ = $[ "read" : $[ "path" : "read_result" ] ];
 map WRITE = $[ "write" : $[ "path" : true ] ];
diff --git a/yast2-testsuite.spec.in b/yast2-testsuite.spec.in
index 27aa3af..64dd90b 100644
--- a/yast2-testsuite.spec.in
+++ b/yast2-testsuite.spec.in
@@ -4,9 +4,12 @@
 Group:	System/YaST
 License: GPL-2.0+
 BuildRequires:	yast2-core-devel yast2-devtools yast2-ycp-ui-bindings
+BuildRequires:  yast2-ruby-bindings >= 1.0.0
 Requires:	expect dejagnu
 # y2base -I includepath -M modulepath
 Requires:	yast2-core >= 2.19.0
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Testsuite
 
 BuildArchitectures: noarch
