diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index d9b12b0..e85fbc8 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh)
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb)
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-tftp-server.spec.in b/yast2-tftp-server.spec.in
index 96e62f9..3522501 100644
--- a/yast2-tftp-server.spec.in
+++ b/yast2-tftp-server.spec.in
@@ -14,6 +14,8 @@ Requires:	lsof
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - TFTP Server Configuration
 
 %description
@@ -33,7 +35,7 @@ network.
 %defattr(-,root,root)
 %dir @yncludedir@/tftp-server
 @yncludedir@/tftp-server/*
-@clientdir@/tftp-server*.ycp
+@clientdir@/tftp-server*.rb
 @moduledir@/TftpServer.*
 @desktopdir@/tftp-server.desktop
 @scrconfdir@/etc_xinetd_d_tftp.scr
