diff --git a/hwinfo/.cvsignore b/hwinfo/.cvsignore
index 282522d..e69de29 100644
--- a/hwinfo/.cvsignore
+++ b/hwinfo/.cvsignore
@@ -1,2 +0,0 @@
-Makefile
-Makefile.in
diff --git a/hwinfo/Makefile.am b/hwinfo/Makefile.am
index ff70b35..e69de29 100644
--- a/hwinfo/Makefile.am
+++ b/hwinfo/Makefile.am
@@ -1,4 +0,0 @@
-#
-# Makefile.am for tune/hwinfo
-
-SUBDIRS = doc src scrconf
diff --git a/hwinfo/src/.cvsignore b/hwinfo/src/.cvsignore
index cb59cbd..e69de29 100644
--- a/hwinfo/src/.cvsignore
+++ b/hwinfo/src/.cvsignore
@@ -1,4 +0,0 @@
-Makefile
-Makefile.in
-*.ybc
-.dep
diff --git a/hwinfo/src/Makefile.am b/hwinfo/src/Makefile.am
index 204e23b..e69de29 100644
--- a/hwinfo/src/Makefile.am
+++ b/hwinfo/src/Makefile.am
@@ -1,23 +0,0 @@
-#
-# Makefile.am for tune/hwinfo/src
-#
-
-client_DATA = hwinfo.ycp hwinfo_proposal.ycp inst_hwinfo.ycp system_settings.ycp system_settings_finish.ycp
-
-desktop_DATA = hwinfo.desktop system_settings.desktop
-
-yncludedir = @yncludedir@/hwinfo
-ynclude_DATA = routines.ycp newid.ycp system_settings_dialogs.ycp system_settings_ui.ycp
-
-module_DATA = InitHWinfo.ycp NewID.ycp SystemSettings.ycp
-
-hwinfo:
-	ln -sf . hwinfo
-
-ycpchook = hwinfo
-
-YCPCFLAGS=-I.
-
-EXTRA_DIST = $(client_DATA) $(ynclude_DATA) $(desktop_DATA) $(module_DATA)
-
-include $(top_srcdir)/Makefile.am.common
diff --git a/yast2-tune.spec.in b/yast2-tune.spec.in
index d6f2256..a808c2d 100644
--- a/yast2-tune.spec.in
+++ b/yast2-tune.spec.in
@@ -14,6 +14,8 @@ Obsoletes:	yast2-config-hwinfo yast2-tune-idedma yast2-trans-tune
 Provides:	yast2-trans-hwinfo yast2-trans-idedma y2c_tune y2t_tune yast2-config-tune
 Obsoletes:	yast2-trans-hwinfo yast2-trans-idedma y2c_tune y2t_tune yast2-config-tune
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Hardware Tuning
 
 %description
@@ -37,10 +39,9 @@ fi
 %defattr(-,root,root)
 
 @yncludedir@/hwinfo/*
-@clientdir@/*.ycp
+@clientdir@/*.rb
 @desktopdir@/hwinfo.desktop
 @desktopdir@/system_settings.desktop
-@moduledir@/*.ycp
-@moduledir@/*.ybc
+@moduledir@/*.rb
 @scrconfdir@/*.scr
 %doc @docdir@
