diff --git a/control/Makefile.am b/control/Makefile.am
index f5a7abb..2f1bc4a 100644
--- a/control/Makefile.am
+++ b/control/Makefile.am
@@ -1,14 +1,12 @@
 #
-# Makefile.am for wagon/src/config
+# Makefile.am for control
 #
 
 controldir = $(yast2dir)/control
 
 control_DATA = $(wildcard *.xml)
 
-desktop_DATA = $(wildcard *.desktop)
-
-EXTRA_DIST = $(desktop_DATA) $(control_DATA)
+EXTRA_DIST = $(control_DATA)
 
 check-local:
 	xmllint --relaxng $(controldir)/control.rng --noout $(control_DATA)
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 9a2350a..b464f78 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -2,5 +2,7 @@
 # Makefile.am for .../doc/autodocs
 #
 
-AUTODOCS_YCP=$(top_srcdir)/src/*/*.ycp
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled
+
+#AUTODOCS_YCP=$(top_srcdir)/src/*/*.ycp
+#include $(top_srcdir)/autodocs-ycp.ami
diff --git a/kernel-update.ycp b/kernel-update.ycp
index 8385f46..428d367 100644
--- a/kernel-update.ycp
+++ b/kernel-update.ycp
@@ -463,14 +463,14 @@
 
 	y2milestone("Handling update script...");
 	while (
-	    // While the script is running or there is still something to be parsed (stdout / stderr)
 	    (
+	    // While the script is running or there is still something to be parsed (stdout / stderr)
 		(boolean) SCR::Read(.background.isrunning)
 		|| (integer) SCR::Read(.background.newlines) > 0
 		|| (integer) SCR::Read(.background.newlines_err) > 0
-	    )
+	    ) &&
 	    // and it hasn't been aborted or finished
-	    && next_loop
+	    next_loop
 	)  {
 	    if (next_loop)
 		next_loop = HandleSTDERR((list <string>) SCR::Read(.background.newerr));
diff --git a/yast2-update.spec.in b/yast2-update.spec.in
index 07df4cc..faf4488 100644
--- a/yast2-update.spec.in
+++ b/yast2-update.spec.in
@@ -39,6 +39,8 @@ Conflicts:	yast2-pkg-bindings < 2.15.11
 # Storage::DeviceMatchFstab (#244117)
 Conflicts:	yast2-storage < 2.15.4
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Update
 
 %package FACTORY
@@ -50,6 +52,8 @@ Requires:	yast2-update yast2
 # to remove dependency on yast2-storage
 Provides:	yast2-update:/usr/share/YaST2/clients/update.ycp
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Update
 
 %description
@@ -71,19 +75,19 @@ Use this component if you wish to update your system.
 @ybindir@/*
 @moduledir@/*
 
-@clientdir@/inst_rootpart.ycp
-@clientdir@/inst_update.ycp
-@clientdir@/inst_backup.ycp
-@clientdir@/rootpart_proposal.ycp
-@clientdir@/update_proposal.ycp
-@clientdir@/packages_proposal.ycp
-@clientdir@/backup_proposal.ycp
-@clientdir@/inst_update_partition.ycp
-@clientdir@/inst_update_partition_auto.ycp
-@clientdir@/rootpart_check_keyboard.ycp
+@clientdir@/inst_rootpart.rb
+@clientdir@/inst_update.rb
+@clientdir@/inst_backup.rb
+@clientdir@/rootpart_proposal.rb
+@clientdir@/update_proposal.rb
+@clientdir@/packages_proposal.rb
+@clientdir@/backup_proposal.rb
+@clientdir@/inst_update_partition.rb
+@clientdir@/inst_update_partition_auto.rb
+@clientdir@/rootpart_check_keyboard.rb
 
 @yncludedir@/update
-@yncludedir@/update/rootpart.ycp
+@yncludedir@/update/rootpart.rb
 
 %doc @docdir@
 
@@ -92,5 +96,5 @@ Use this component if you wish to update your system.
 @desktopdir@/update.desktop
 %dir /usr/share/YaST2/control
 /usr/share/YaST2/control/update.xml
-@clientdir@/update.ycp
-@clientdir@/run_update.ycp
+@clientdir@/update.rb
+@clientdir@/run_update.rb
