diff --git a/agent-nis/.cvsignore b/agent-nis/.cvsignore
deleted file mode 100644
index 282522d..0000000
--- a/agent-nis/.cvsignore
+++ /dev/null
@@ -1,2 +0,0 @@
-Makefile
-Makefile.in
diff --git a/agent-nis/Makefile.am b/agent-nis/Makefile.am
deleted file mode 100644
index fab125c..0000000
--- a/agent-nis/Makefile.am
+++ /dev/null
@@ -1,10 +0,0 @@
-#
-# Makefile.am for users/agent-nis
-#
-
-agentdir = @execcompdir@/servers_non_y2
-
-agent_SCRIPTS = ag_nis
-scrconf_DATA = nis.scr
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/agent-uid/.cvsignore b/agent-uid/.cvsignore
deleted file mode 100644
index 282522d..0000000
--- a/agent-uid/.cvsignore
+++ /dev/null
@@ -1,2 +0,0 @@
-Makefile
-Makefile.in
diff --git a/agent-uid/Makefile.am b/agent-uid/Makefile.am
deleted file mode 100644
index 78c0fff..0000000
--- a/agent-uid/Makefile.am
+++ /dev/null
@@ -1,10 +0,0 @@
-#
-# Makefile.am for users/agent-uid
-#
-
-agentdir = @execcompdir@/servers_non_y2
-
-agent_SCRIPTS = ag_uid
-scrconf_DATA = uid.scr
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/agents/.cvsignore b/agents/.cvsignore
deleted file mode 100644
index 282522d..0000000
--- a/agents/.cvsignore
+++ /dev/null
@@ -1,2 +0,0 @@
-Makefile
-Makefile.in
diff --git a/agents/Makefile.am b/agents/Makefile.am
deleted file mode 100644
index da36e63..0000000
--- a/agents/Makefile.am
+++ /dev/null
@@ -1,8 +0,0 @@
-#
-# Makefile.am for y2m_inst/ycp/general/agents
-#
-
-
-scrconf_DATA = $(patsubst $(srcdir)/%,%,$(wildcard $(srcdir)/*.scr))
-
-EXTRA_DIST = $(scrconf_DATA)
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 99a5370..72b515e 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,4 +1,4 @@
 # Makefile.am for YCP module .../doc/autodocs
 
-AUTODOCS_PM = $(srcdir)/../../src/*.pm $(srcdir)/../../src/YaPI/*.pm
+AUTODOCS_PM = $(srcdir)/../../src/modules/*.pm $(srcdir)/../../src/modules/YaPI/*.pm
 include $(top_srcdir)/autodocs-ycp.ami
diff --git a/testsuite/Makefile.am b/testsuite/Makefile.am
index 97cf756..85e8a1c 100644
--- a/testsuite/Makefile.am
+++ b/testsuite/Makefile.am
@@ -5,7 +5,7 @@
 #
 
 AUTOMAKE_OPTIONS = dejagnu
-EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.ycp) $(wildcard tests/*.yh) passwd shadow group
+EXTRA_DIST = $(wildcard tests/*.out) $(wildcard tests/*.err) $(wildcard tests/*.rb) passwd shadow group
 
 testsuite_prepare = @ydatadir@/testsuite/Makefile.testsuite
 
diff --git a/yast2-users.spec.in b/yast2-users.spec.in
index dda3d5f..acf6d4c 100644
--- a/yast2-users.spec.in
+++ b/yast2-users.spec.in
@@ -21,6 +21,8 @@ Requires:       yast2 >= 2.23.7
 # cryptsha256, cryptsha516
 Requires:       yast2-core >= 2.21.0
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - User and Group Configuration
 
 %description
@@ -38,10 +40,10 @@ This package provides GUI for maintenance of linux users and groups.
 %defattr(-,root,root)
 %dir @yncludedir@/users
 %dir @moduledir@/YaPI
-@clientdir@/*.ycp
+@clientdir@/*.rb
 @desktopdir@/*.desktop
 @moduledir@/*.pm
-@moduledir@/UsersUI.y*
+@moduledir@/UsersUI.rb
 @moduledir@/YaPI/*.pm
 @yncludedir@/users/*
 @schemadir@/autoyast/rnc/users.rnc
