diff --git a/agent-nis/Makefile.am b/agent-nis/Makefile.am
index fab125c..0d39f54 100644
--- a/agent-nis/Makefile.am
+++ b/agent-nis/Makefile.am
@@ -1,10 +1,10 @@
 #
 # Makefile.am for users/agent-nis
 #
+# moved
+# agentdir = @execcompdir@/servers_non_y2
 
-agentdir = @execcompdir@/servers_non_y2
-
-agent_SCRIPTS = ag_nis
-scrconf_DATA = nis.scr
+agent_SCRIPTS = # ag_nis
+scrconf_DATA = # nis.scr
 
 EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/agent-uid/Makefile.am b/agent-uid/Makefile.am
index 78c0fff..ea980c5 100644
--- a/agent-uid/Makefile.am
+++ b/agent-uid/Makefile.am
@@ -1,10 +1,10 @@
 #
 # Makefile.am for users/agent-uid
 #
+# moved
+# agentdir = @execcompdir@/servers_non_y2
 
-agentdir = @execcompdir@/servers_non_y2
-
-agent_SCRIPTS = ag_uid
-scrconf_DATA = uid.scr
+agent_SCRIPTS = #ag_uid
+scrconf_DATA = #uid.scr
 
 EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/doc/Makefile.am b/doc/Makefile.am
index b6bc8cc..1412807 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -1,8 +1,8 @@
 #
 # Makefile.am for y2c_users/doc
 #
-
-SUBDIRS = autodocs
+# TODO no autodocs for rb yet
+SUBDIRS = #autodocs
 
 htmldir = @docdir@
 
diff --git a/src/include/users/routines.ycp b/src/include/users/routines.ycp
index 84686e4..c5e262b 100644
--- a/src/include/users/routines.ycp
+++ b/src/include/users/routines.ycp
@@ -29,6 +29,7 @@
  */
 {
 import "Mode";
+import "Stage";
 
 textdomain "users";
 
diff --git a/src/include/users/widgets.ycp b/src/include/users/widgets.ycp
index 62bdf33..b2e76b2 100644
--- a/src/include/users/widgets.ycp
+++ b/src/include/users/widgets.ycp
@@ -468,9 +468,9 @@ only obtain users with a username beginning with 'u'.</p>
 		    button = `notnext;
 		    continue;
 		}
-		if (!issubstring (tolower (curr_user_f),tolower(default_user_f))
+		if (!issubstring (tolower (curr_user_f),tolower(default_user_f)) &&
 		    // yes/no popup question
-		    && !Popup::YesNo (_("The new user filter does not contain the default user filter.
+		    !Popup::YesNo (_("The new user filter does not contain the default user filter.
 Really use it?
 ")))
 		{
@@ -2245,4 +2245,4 @@ map <string, map <string, any> > widgets = $[
     ],
 ];
 
-}//EOF
+}
diff --git a/src/include/users/wizards.ycp b/src/include/users/wizards.ycp
index 010cd2b..d552e37 100644
--- a/src/include/users/wizards.ycp
+++ b/src/include/users/wizards.ycp
@@ -276,8 +276,9 @@ define symbol UsersSequence(string start) ``{
 	"read" : $[
 	    `abort	: `abort,
 	    `next	: "main",
-        `nextmodule: `next // this is for skiping users conf during install,
-                           // see Users::CheckHomeMounted() or bug #20365
+        // this is for skiping users conf during install,
+        // see Users::CheckHomeMounted() or bug #20365
+        `nextmodule: `next 
 	],
 	"main" : $[
 	    `abort	: `abort,
diff --git a/yast2-users.spec.in b/yast2-users.spec.in
index dda3d5f..acf6d4c 100644
--- a/yast2-users.spec.in
+++ b/yast2-users.spec.in
@@ -21,6 +21,8 @@ Requires:       yast2 >= 2.23.7
 # cryptsha256, cryptsha516
 Requires:       yast2-core >= 2.21.0
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - User and Group Configuration
 
 %description
@@ -38,10 +40,10 @@ This package provides GUI for maintenance of linux users and groups.
 %defattr(-,root,root)
 %dir @yncludedir@/users
 %dir @moduledir@/YaPI
-@clientdir@/*.ycp
+@clientdir@/*.rb
 @desktopdir@/*.desktop
 @moduledir@/*.pm
-@moduledir@/UsersUI.y*
+@moduledir@/UsersUI.rb
 @moduledir@/YaPI/*.pm
 @yncludedir@/users/*
 @schemadir@/autoyast/rnc/users.rnc
