diff --git a/agents/Makefile.am b/agents/Makefile.am
index badb685..50d0e44 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -2,6 +2,6 @@
 
 agent_SCRIPTS =
 
-scrconf_DATA = xend.scr
+scrconf_DATA =
 
 EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/yast2-vm.spec.in b/yast2-vm.spec.in
index 9ee2661..4d7511a 100644
--- a/yast2-vm.spec.in
+++ b/yast2-vm.spec.in
@@ -7,6 +7,8 @@ BuildRequires:	perl-XML-Writer update-desktop-files yast2 yast2-devtools yast2-t
 License:        GPL-2.0
 Requires:	yast2 >= 2.21.22
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	Configure Hypervisor and Tools for Xen and KVM
 
 %description
@@ -28,10 +30,10 @@ rm -f $RPM_BUILD_ROOT/usr/share/applications/YaST2/relocation-server.desktop
 %defattr(-,root,root)
 %dir @scrconfdir@
 %dir @yncludedir@
-@clientdir@/relocation-server.ycp
-@clientdir@/xen.ycp
-@clientdir@/vm_finish.ycp
-@moduledir@/VM_XEN.y*
+@clientdir@/relocation-server.rb
+@clientdir@/xen.rb
+@clientdir@/vm_finish.rb
+@moduledir@/VM_XEN.rb
 @moduledir@/RelocationServer.*
 @yncludedir@/*
 @scrconfdir@/*
