diff --git a/agents/Makefile.am b/agents/Makefile.am
index badb685..50d0e44 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -2,6 +2,6 @@
 
 agent_SCRIPTS =
 
-scrconf_DATA = xend.scr
+scrconf_DATA =
 
 EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/doc/Makefile.am b/doc/Makefile.am
index 1ec3fcb..6bae5a9 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -2,7 +2,8 @@
 # Makefile.am for devtools/devtools/skeletons/config/doc
 #
 
-SUBDIRS = autodocs
+# TODO add ruby support to autodocs
+#SUBDIRS = autodocs
 
 htmldir = $(docdir)
 
diff --git a/yast2-vm.spec.in b/yast2-vm.spec.in
index 9ee2661..32ef464 100644
--- a/yast2-vm.spec.in
+++ b/yast2-vm.spec.in
@@ -28,10 +28,10 @@ rm -f $RPM_BUILD_ROOT/usr/share/applications/YaST2/relocation-server.desktop
 %defattr(-,root,root)
 %dir @scrconfdir@
 %dir @yncludedir@
-@clientdir@/relocation-server.ycp
-@clientdir@/xen.ycp
-@clientdir@/vm_finish.ycp
-@moduledir@/VM_XEN.y*
+@clientdir@/relocation-server.rb
+@clientdir@/xen.rb
+@clientdir@/vm_finish.rb
+@moduledir@/VM_XEN.rb
 @moduledir@/RelocationServer.*
 @yncludedir@/*
 @scrconfdir@/*
