diff --git a/library/commandline/src/modules/CommandLine.ycp b/library/commandline/src/modules/CommandLine.ycp
index e69a0f4..8ba5163 100644
--- a/library/commandline/src/modules/CommandLine.ycp
+++ b/library/commandline/src/modules/CommandLine.ycp
@@ -220,6 +220,56 @@ you may find current contact information at www.novell.com
     }
 
 
+    list<string> Process(term line)
+    {
+	list<string> ret = [];
+	list<any> anys = argsof(line);
+	foreach(any a, anys, {
+	    if (is(a, string)) {
+		string s = (string) a;
+		ret = add(ret, s);
+	    }
+	    else if (is(a, term)) {
+		term t = (term) a;
+		if (contains([`Left, `Center, `Right], symbolof(t)))
+		    ret = add(ret, argsof(t)[0]:"");
+	    }
+	});
+	return ret;
+    }
+
+    list<symbol> GetAligns(term header)
+    {
+	list<any> anys = argsof(header);
+	list<symbol> ret = [];
+	foreach(integer i, Integer::Range(size(anys)), {
+	    any a = anys[i]:nil;
+	    if (is(a, term))
+	    {
+		term t = (term) a;
+		if (symbolof(t) == `Right)
+		    ret[i] = `right;
+	    }
+	});
+	return ret;
+    }
+
+    void UpdateWidths(list<string> columns, list<integer> & widths)
+    {
+	foreach(integer i, Integer::Range(size(columns)), {
+	    widths[i] = Integer::Max([ widths[i]:0, size(columns[i]:"") ]);
+	});
+    }
+
+    void PrintLine(term line, list<integer> widths, list<symbol> aligns)
+    {
+	list<string> columns = Process(line);
+	foreach(integer i, Integer::Range(size(columns)), {
+	    columns[i] = String::SuperPad(columns[i]:"", widths[i]:0, " ", aligns[i]:`left);
+	});
+	CommandLine::Print(mergestring(columns, " | "));
+    }
+
     /**
      *  @short Print a Table
      *  @descr Print a table using Print(). Format of table is as libyui but not all features
@@ -230,69 +280,20 @@ you may find current contact information at www.novell.com
      */
     global void PrintTable(term header, list<term> content)
     {
-	list<symbol> aligns = [];
 	list<integer> widths = [];
 
-	list<string> Process(term line)
-	{
-	    list<string> ret = [];
-	    list<any> anys = argsof(line);
-	    foreach(any a, anys, {
-		if (is(a, string)) {
-		    string s = (string) a;
-		    ret = add(ret, s);
-		}
-		else if (is(a, term)) {
-		    term t = (term) a;
-		    if (contains([`Left, `Center, `Right], symbolof(t)))
-			ret = add(ret, argsof(t)[0]:"");
-		}
-	    });
-	    return ret;
-	}
-
-	void GetAligns(term header)
-	{
-	    list<any> anys = argsof(header);
-	    foreach(integer i, Integer::Range(size(anys)), {
-		any a = anys[i]:nil;
-		if (is(a, term))
-		{
-		    term t = (term) a;
-		    if (symbolof(t) == `Right)
-			aligns[i] = `right;
-		}
-	    });
-	}
-
-	void UpdateWidths(list<string> columns)
-	{
-	    foreach(integer i, Integer::Range(size(columns)), {
-		widths[i] = Integer::Max([ widths[i]:0, size(columns[i]:"") ]);
-	    });
-	}
-
-	void PrintLine(term line)
-	{
-	    list<string> columns = Process(line);
-	    foreach(integer i, Integer::Range(size(columns)), {
-		columns[i] = String::SuperPad(columns[i]:"", widths[i]:0, " ", aligns[i]:`left);
-	    });
-	    CommandLine::Print(mergestring(columns, " | "));
-	}
-
-	UpdateWidths(Process(header));
-	foreach(term row, content, { UpdateWidths(Process(row)); });
+	UpdateWidths(Process(header), widths);
+	foreach(term row, content, { UpdateWidths(Process(row), widths); });
 
-	PrintLine(header);
+	PrintLine(header, widths, []);
 
-	GetAligns(header);
+	list<symbol> aligns = GetAligns(header);
 
 	CommandLine::Print(mergestring(maplist(integer width, widths, {
 	    return String::Repeat("-", width);
 	}), "-+-"));
 
-	foreach(term row, content, { PrintLine(row); });
+	foreach(term row, content, { PrintLine(row, widths, aligns); });
     }
 
 
diff --git a/library/commandline/testsuite/tests/cmdmap6.ycp b/library/commandline/testsuite/tests/cmdmap6.ycp
index 184a4ed..190f383 100644
--- a/library/commandline/testsuite/tests/cmdmap6.ycp
+++ b/library/commandline/testsuite/tests/cmdmap6.ycp
@@ -33,7 +33,7 @@ you may find current contact information at www.novell.com
 {
 
     include "testsuite.ycp";
-    include "./tests/testsuitedata.yh";
+    include "testsuitedata.yh";
 
     import "TypeRepository";
     import "CommandLine";
diff --git a/library/commandline/testsuite/tests/command1.ycp b/library/commandline/testsuite/tests/command1.ycp
index d0252bc..57d96c0 100644
--- a/library/commandline/testsuite/tests/command1.ycp
+++ b/library/commandline/testsuite/tests/command1.ycp
@@ -33,7 +33,7 @@ you may find current contact information at www.novell.com
 {
 
     include "testsuite.ycp";
-    include "./tests/testsuitedata.yh";
+    include "testsuitedata.yh";
 
     import "TypeRepository";
     import "CommandLine";
diff --git a/library/commandline/testsuite/tests/command2.ycp b/library/commandline/testsuite/tests/command2.ycp
index e0c6823..d1fa598 100644
--- a/library/commandline/testsuite/tests/command2.ycp
+++ b/library/commandline/testsuite/tests/command2.ycp
@@ -33,7 +33,7 @@ you may find current contact information at www.novell.com
 {
 
     include "testsuite.ycp";
-    include "./tests/testsuitedata.yh";
+    include "testsuitedata.yh";
 
     import "TypeRepository";
     import "CommandLine";
diff --git a/library/commandline/testsuite/tests/command3.ycp b/library/commandline/testsuite/tests/command3.ycp
index 6e16b72..d82678c 100644
--- a/library/commandline/testsuite/tests/command3.ycp
+++ b/library/commandline/testsuite/tests/command3.ycp
@@ -33,7 +33,7 @@ you may find current contact information at www.novell.com
 {
 
     include "testsuite.ycp";
-    include "./tests/testsuitedata.yh";
+    include "testsuitedata.yh";
 
     import "TypeRepository";
     import "CommandLine";
diff --git a/library/commandline/testsuite/tests/command4.ycp b/library/commandline/testsuite/tests/command4.ycp
index 8f2bec3..e50cdac 100644
--- a/library/commandline/testsuite/tests/command4.ycp
+++ b/library/commandline/testsuite/tests/command4.ycp
@@ -33,7 +33,7 @@ you may find current contact information at www.novell.com
 {
 
     include "testsuite.ycp";
-    include "./tests/testsuitedata.yh";
+    include "testsuitedata.yh";
 
     import "TypeRepository";
     import "CommandLine";
diff --git a/library/commandline/testsuite/tests/command5.ycp b/library/commandline/testsuite/tests/command5.ycp
index 41eed86..b8f9b29 100644
--- a/library/commandline/testsuite/tests/command5.ycp
+++ b/library/commandline/testsuite/tests/command5.ycp
@@ -33,7 +33,7 @@ you may find current contact information at www.novell.com
 {
 
     include "testsuite.ycp";
-    include "./tests/testsuitedata.yh";
+    include "testsuitedata.yh";
 
     import "CommandLine";
 
diff --git a/library/commandline/testsuite/tests/command6.ycp b/library/commandline/testsuite/tests/command6.ycp
index b784cac..a468295 100644
--- a/library/commandline/testsuite/tests/command6.ycp
+++ b/library/commandline/testsuite/tests/command6.ycp
@@ -33,7 +33,7 @@ you may find current contact information at www.novell.com
 {
 
     include "testsuite.ycp";
-    include "./tests/testsuitedata.yh";
+    include "testsuitedata.yh";
 
     import "CommandLine";
 
diff --git a/library/commandline/testsuite/tests/command7.ycp b/library/commandline/testsuite/tests/command7.ycp
index 7f66dfc..4478c4c 100644
--- a/library/commandline/testsuite/tests/command7.ycp
+++ b/library/commandline/testsuite/tests/command7.ycp
@@ -33,7 +33,7 @@ you may find current contact information at www.novell.com
 {
 
     include "testsuite.ycp";
-    include "./tests/testsuitedata.yh";
+    include "testsuitedata.yh";
 
     import "CommandLine";
 
diff --git a/library/commandline/testsuite/tests/help.ycp b/library/commandline/testsuite/tests/help.ycp
index 29ef271..4ffcf4b 100644
--- a/library/commandline/testsuite/tests/help.ycp
+++ b/library/commandline/testsuite/tests/help.ycp
@@ -33,7 +33,7 @@ you may find current contact information at www.novell.com
 {
 
     include "testsuite.ycp";
-    include "./tests/testsuitedata.yh";
+    include "testsuitedata.yh";
 
     import "TypeRepository";
     import "CommandLine";
diff --git a/library/commandline/testsuite/tests/help1.ycp b/library/commandline/testsuite/tests/help1.ycp
index d291f80..1de1d17 100644
--- a/library/commandline/testsuite/tests/help1.ycp
+++ b/library/commandline/testsuite/tests/help1.ycp
@@ -33,7 +33,7 @@ you may find current contact information at www.novell.com
 {
 
     include "testsuite.ycp";
-    include "./tests/testsuitedata.yh";
+    include "testsuitedata.yh";
 
     import "TypeRepository";
     import "CommandLine";
diff --git a/library/commandline/testsuite/tests/interactive1.ycp b/library/commandline/testsuite/tests/interactive1.ycp
index 2d80566..09f03e5 100644
--- a/library/commandline/testsuite/tests/interactive1.ycp
+++ b/library/commandline/testsuite/tests/interactive1.ycp
@@ -33,7 +33,7 @@ you may find current contact information at www.novell.com
 {
 
     include "testsuite.ycp";
-    include "./tests/testsuitedata.yh";
+    include "testsuitedata.yh";
 
     import "CommandLine";
 
diff --git a/library/commandline/testsuite/tests/interactive2.ycp b/library/commandline/testsuite/tests/interactive2.ycp
index 858004f..dea3f80 100644
--- a/library/commandline/testsuite/tests/interactive2.ycp
+++ b/library/commandline/testsuite/tests/interactive2.ycp
@@ -33,7 +33,7 @@ you may find current contact information at www.novell.com
 {
 
     include "testsuite.ycp";
-    include "./tests/testsuitedata.yh";
+    include "testsuitedata.yh";
 
     import "CommandLine";
 
diff --git a/library/commandline/testsuite/tests/interactive3.ycp b/library/commandline/testsuite/tests/interactive3.ycp
index 6f0c128..fde1506 100644
--- a/library/commandline/testsuite/tests/interactive3.ycp
+++ b/library/commandline/testsuite/tests/interactive3.ycp
@@ -33,7 +33,7 @@ you may find current contact information at www.novell.com
 {
 
     include "testsuite.ycp";
-    include "./tests/testsuitedata.yh";
+    include "testsuitedata.yh";
 
     import "CommandLine";
 
diff --git a/library/commandline/testsuite/tests/interactive4.ycp b/library/commandline/testsuite/tests/interactive4.ycp
index 3d6ad1c..1e8e195 100644
--- a/library/commandline/testsuite/tests/interactive4.ycp
+++ b/library/commandline/testsuite/tests/interactive4.ycp
@@ -33,7 +33,7 @@ you may find current contact information at www.novell.com
 {
 
     include "testsuite.ycp";
-    include "./tests/testsuitedata.yh";
+    include "testsuitedata.yh";
 
     import "CommandLine";
 
diff --git a/library/commandline/testsuite/tests/interactive5.ycp b/library/commandline/testsuite/tests/interactive5.ycp
index 80d1930..797e061 100644
--- a/library/commandline/testsuite/tests/interactive5.ycp
+++ b/library/commandline/testsuite/tests/interactive5.ycp
@@ -33,7 +33,7 @@ you may find current contact information at www.novell.com
 {
 
     include "testsuite.ycp";
-    include "./tests/testsuitedata.yh";
+    include "testsuitedata.yh";
 
     import "CommandLine";
 
diff --git a/library/commandline/testsuite/tests/longhelp.ycp b/library/commandline/testsuite/tests/longhelp.ycp
index 8736e5c..92a7762 100644
--- a/library/commandline/testsuite/tests/longhelp.ycp
+++ b/library/commandline/testsuite/tests/longhelp.ycp
@@ -33,7 +33,7 @@ you may find current contact information at www.novell.com
 {
 
     include "testsuite.ycp";
-    include "./tests/testsuitedata.yh";
+    include "testsuitedata.yh";
 
     import "TypeRepository";
     import "CommandLine";
diff --git a/library/commandline/testsuite/tests/startgui1.ycp b/library/commandline/testsuite/tests/startgui1.ycp
index 6303df6..80d358c 100644
--- a/library/commandline/testsuite/tests/startgui1.ycp
+++ b/library/commandline/testsuite/tests/startgui1.ycp
@@ -33,7 +33,7 @@ you may find current contact information at www.novell.com
 {
 
     include "testsuite.ycp";
-    include "./tests/testsuitedata.yh";
+    include "testsuitedata.yh";
 
     import "TypeRepository";
     import "CommandLine";
diff --git a/library/commandline/testsuite/tests/startgui2.ycp b/library/commandline/testsuite/tests/startgui2.ycp
index a6b9e52..b3bf479 100644
--- a/library/commandline/testsuite/tests/startgui2.ycp
+++ b/library/commandline/testsuite/tests/startgui2.ycp
@@ -33,7 +33,7 @@ you may find current contact information at www.novell.com
 {
 
     include "testsuite.ycp";
-    include "./tests/testsuitedata.yh";
+    include "testsuitedata.yh";
 
     import "TypeRepository";
     import "CommandLine";
diff --git a/library/cwm/testsuite/tests/t1.ycp b/library/cwm/testsuite/tests/t1.ycp
index afa62bb..d41081b 100644
--- a/library/cwm/testsuite/tests/t1.ycp
+++ b/library/cwm/testsuite/tests/t1.ycp
@@ -46,7 +46,7 @@ you may find current contact information at www.novell.com
     Report::display_errors = false;
     Report::log_errors = true;
 
-    include "./tests/testfunc.yh";
+    include "testfunc.yh";
 
 
     map<string, any> functions = $[
diff --git a/library/cwm/testsuite/tests/t2.ycp b/library/cwm/testsuite/tests/t2.ycp
index a86971d..b29f075 100644
--- a/library/cwm/testsuite/tests/t2.ycp
+++ b/library/cwm/testsuite/tests/t2.ycp
@@ -51,7 +51,7 @@ you may find current contact information at www.novell.com
     Report::display_errors = false;
     Report::log_errors = true;
 
-    include "./tests/testfunc.yh";
+    include "testfunc.yh";
     import "Mode";
 
     Mode::SetTest ("testsuite");
diff --git a/library/desktop/src/modules/Desktop.ycp b/library/desktop/src/modules/Desktop.ycp
index 3ced0f7..f57f665 100644
--- a/library/desktop/src/modules/Desktop.ycp
+++ b/library/desktop/src/modules/Desktop.ycp
@@ -123,20 +123,19 @@ void ReadLanguage() {
     y2debug("Language=%1", Language);
 }
 
+string ExtractDesktopFilename( string fullpath )
+{
+    list <string> path_components = splitstring(fullpath, "/");
+    string filename = path_components[ size( path_components) - 1]:"";
+
+    return filename;
+}
+
 /**
  * Read module and group data from desktop files
  * @param Values list of values to be parsed (empty to read all)
  */
 global define void Read(list<string> Values) {
-
-    string ExtractDesktopFilename( string fullpath )
-    {
-	list <string> path_components = splitstring(fullpath, "/");
-	string filename = path_components[ size( path_components) - 1]:"";
-
-	return filename;
-    }
-
     /* read modules */
     map filemap = $[];
     path filepath = nil;
diff --git a/library/sequencer/testsuite/tests/WS_check.ycp b/library/sequencer/testsuite/tests/WS_check.ycp
index 9e11a70..fe8dd72 100644
--- a/library/sequencer/testsuite/tests/WS_check.ycp
+++ b/library/sequencer/testsuite/tests/WS_check.ycp
@@ -73,7 +73,7 @@ TEST(``(Sequencer::WS_check($[], $[ "ws_start" : `ws_finish ] )), [], nil);
 
 list clicks = nil;
 integer cur = -1;
-include "./tests/Wizard.yh";
+include "Wizard.yh";
 TEST(``(Sequencer::WS_check(aliases(), sequence())), [], nil);
 
 }
diff --git a/library/sequencer/testsuite/tests/Wizard0.ycp b/library/sequencer/testsuite/tests/Wizard0.ycp
index 0ceccbb..6a821b2 100644
--- a/library/sequencer/testsuite/tests/Wizard0.ycp
+++ b/library/sequencer/testsuite/tests/Wizard0.ycp
@@ -23,7 +23,7 @@ you may find current contact information at www.novell.com
 
 list clicks = nil;
 integer cur = -1;
-include "./tests/Wizard.yh";
+include "Wizard.yh";
 
 map Aliases = $[
     "0" : ``(next()),
diff --git a/library/sequencer/testsuite/tests/Wizard1.ycp b/library/sequencer/testsuite/tests/Wizard1.ycp
index 794ec22..e7d6d51 100644
--- a/library/sequencer/testsuite/tests/Wizard1.ycp
+++ b/library/sequencer/testsuite/tests/Wizard1.ycp
@@ -23,6 +23,6 @@ you may find current contact information at www.novell.com
 
 integer cur = 0;
 list clicks = [ `next, `next, `finish ];
-include "./tests/Wizard.yh";
+include "Wizard.yh";
 
 }
diff --git a/library/sequencer/testsuite/tests/Wizard2.ycp b/library/sequencer/testsuite/tests/Wizard2.ycp
index 1ff3486..002207d 100644
--- a/library/sequencer/testsuite/tests/Wizard2.ycp
+++ b/library/sequencer/testsuite/tests/Wizard2.ycp
@@ -23,6 +23,6 @@ you may find current contact information at www.novell.com
 
 integer cur = 0;
 list clicks = [ `next, `details, `next, `finish ];
-include "./tests/Wizard.yh";
+include "Wizard.yh";
 
 }
diff --git a/library/sequencer/testsuite/tests/Wizard3.ycp b/library/sequencer/testsuite/tests/Wizard3.ycp
index 84c4c08..43146c9 100644
--- a/library/sequencer/testsuite/tests/Wizard3.ycp
+++ b/library/sequencer/testsuite/tests/Wizard3.ycp
@@ -23,6 +23,6 @@ you may find current contact information at www.novell.com
 
 integer cur = 0;
 list clicks = [ `next, `details, `details, `next, `finish ];
-include "./tests/Wizard.yh";
+include "Wizard.yh";
 
 }
diff --git a/library/sequencer/testsuite/tests/Wizard4.ycp b/library/sequencer/testsuite/tests/Wizard4.ycp
index 00a0679..54afb0b 100644
--- a/library/sequencer/testsuite/tests/Wizard4.ycp
+++ b/library/sequencer/testsuite/tests/Wizard4.ycp
@@ -23,6 +23,6 @@ you may find current contact information at www.novell.com
 
 integer cur = 0;
 list clicks = [ `next, `expert, `next, `next, `finish ];
-include "./tests/Wizard.yh";
+include "Wizard.yh";
 
 }
diff --git a/library/sequencer/testsuite/tests/Wizard5.ycp b/library/sequencer/testsuite/tests/Wizard5.ycp
index 651d5a7..a297f44 100644
--- a/library/sequencer/testsuite/tests/Wizard5.ycp
+++ b/library/sequencer/testsuite/tests/Wizard5.ycp
@@ -23,6 +23,6 @@ you may find current contact information at www.novell.com
 
 integer cur = 0;
 list clicks = [ `back ];
-include "./tests/Wizard.yh";
+include "Wizard.yh";
 
 }
diff --git a/library/sequencer/testsuite/tests/Wizard6.ycp b/library/sequencer/testsuite/tests/Wizard6.ycp
index a4b4a92..53ca86d 100644
--- a/library/sequencer/testsuite/tests/Wizard6.ycp
+++ b/library/sequencer/testsuite/tests/Wizard6.ycp
@@ -23,6 +23,6 @@ you may find current contact information at www.novell.com
 
 integer cur = 0;
 list clicks = [ `next, `back, `back ];
-include "./tests/Wizard.yh";
+include "Wizard.yh";
 
 }
diff --git a/library/sequencer/testsuite/tests/Wizard7.ycp b/library/sequencer/testsuite/tests/Wizard7.ycp
index a5961f7..780cacf 100644
--- a/library/sequencer/testsuite/tests/Wizard7.ycp
+++ b/library/sequencer/testsuite/tests/Wizard7.ycp
@@ -23,6 +23,6 @@ you may find current contact information at www.novell.com
 
 integer cur = 0;
 list clicks = [ `next, `next, `back, `back, `back ];
-include "./tests/Wizard.yh";
+include "Wizard.yh";
 
 }
diff --git a/library/wizard/testsuite/tests/wizard_dialog.ycp b/library/wizard/testsuite/tests/wizard_dialog.ycp
index f8825de..5d58b65 100644
--- a/library/wizard/testsuite/tests/wizard_dialog.ycp
+++ b/library/wizard/testsuite/tests/wizard_dialog.ycp
@@ -23,7 +23,7 @@ you may find current contact information at www.novell.com
   // test wizard_dialog
 
   textdomain "wizard_testsuite";
-  import "./../src/Wizard";
+  import "Wizard";
 
   Wizard::CreateDialog();
   Wizard::SetHelpText( _("<p>This is the example of <tt>CreateWizardDialog()</tt></p>"));
